{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "failed to load logging config file\n"
     ]
    }
   ],
   "source": [
    "import calour as ca\n",
    "import calour_utils as cu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/amnon/miniconda3/envs/calour/lib/python3.7/site-packages/skbio/util/_testing.py:15: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.\n",
      "  import pandas.util.testing as pdt\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import glob\n",
    "import os\n",
    "import pandas as pd\n",
    "import shutil\n",
    "import matplotlib.colors\n",
    "import skbio as skbio"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "ca.set_log_level('INFO')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib notebook"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/Users/amnon/Projects/sheba/metaanalysis/scripts'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Prepare the data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "def join_all_exps(subset_size=23, subset_control=23, keep_less=True,normalize=False, random_seed=None):\n",
    "    '''load all experiemnts\n",
    "    to a single dataset aftert subsampling\n",
    "\n",
    "    Parameters\n",
    "    ----------\n",
    "    subset_size: int or None, optional\n",
    "        if not None, randomly select subset_size samples for case and control groups each\n",
    "    subset_control: int or None, optional\n",
    "        if subset_size is not None, randomly select subset_control control samples. If None, use value of subset_size\n",
    "    keep_less: bool\n",
    "        True to also keep studies with < subset_size samples.\n",
    "        False to throw away studies with < subset_size samples.\n",
    "    normalize: bool, optional\n",
    "        True to noramlize each sample to 10k reads, False (default) to skip normalization\n",
    "    random_seed : int, np.radnom.Generator instance or None, optional, default=None\n",
    "        set the random number generator seed for the random permutations. Used to select subset of samples.\n",
    "        If int, random_seed is the seed used by the random number generator;\n",
    "        If Generator instance, random_seed is set to the random number generator;\n",
    "        If None, then fresh, unpredictable entropy will be pulled from the OS    \n",
    "\n",
    "    Returns\n",
    "    -------\n",
    "    exp: the joined experiment\n",
    "    '''\n",
    "    exp = None\n",
    "    total_ok = 0\n",
    "    total_low = 0\n",
    "    for idx, cname in enumerate(glob.glob('../studies/*')):\n",
    "        # Initialize the random seed for each study, since the order of studies is OS dependent\n",
    "        rng = np.random.default_rng(random_seed)\n",
    "        if os.path.isdir(cname):\n",
    "            print('**********')\n",
    "            print('processing %s (%d)' % (cname, idx))\n",
    "            tables = glob.glob(os.path.join(cname,'all.*biom'))\n",
    "            print(tables)\n",
    "            if len(tables)==0:\n",
    "                print('dir %s does not contain a biom table' % cname)\n",
    "                continue\n",
    "            bt=tables[0]\n",
    "            if normalize is True:\n",
    "                normalize=10000\n",
    "            else:\n",
    "                normalize = None\n",
    "            data=ca.read_amplicon(os.path.join(bt),os.path.join(cname,'up.map.csv'),normalize=normalize,min_reads=1000)\n",
    "            print('-------------')\n",
    "            print(data)\n",
    "            # TODO: if not enough samples, print and continue\n",
    "            # if enough samples, keep only random subet\n",
    "            if subset_size is not None:\n",
    "                tt=data.filter_samples('type','HC')\n",
    "                ns = len(tt.sample_metadata)\n",
    "                if ns < subset_control:\n",
    "                    print('not enough controls (%d). skipping' % ns)\n",
    "                    if not keep_less:\n",
    "                        total_low += 1\n",
    "                        continue\n",
    "                # randomly subsample\n",
    "                rp = rng.permutation(ns)\n",
    "                rp = rp[:subset_size]\n",
    "                new_ids = [tt.sample_metadata['_sample_id'][x] for x in rp]\n",
    "                # similar for disease\n",
    "                tt=data.filter_samples('type','disease')\n",
    "                ns = len(tt.sample_metadata)\n",
    "                if ns < subset_size:\n",
    "                    print('not enough disease (%d). skipping' % ns)\n",
    "                    if not keep_less:\n",
    "                        total_low += 1\n",
    "                        continue\n",
    "                # randomly subsample\n",
    "                rp = np.random.permutation(ns)\n",
    "                rp = rp[:subset_size]\n",
    "                new_ids2 = [tt.sample_metadata['_sample_id'][x] for x in rp]\n",
    "                # join the ids for the sick and controls\n",
    "                all_ids = new_ids + new_ids2\n",
    "                # and finally\n",
    "                data = data.filter_samples('_sample_id',all_ids)\n",
    "                print('keeping %d total samples' % len(data.sample_metadata))\n",
    "            else:\n",
    "                print('filtering for HC and disease')\n",
    "                data = data.filter_samples('type', ['HC', 'disease'])\n",
    "                print(data)\n",
    "            data.description = cname\n",
    "            if exp is None:\n",
    "                exp = data\n",
    "                exp.sample_metadata['exp']=cname\n",
    "            else:\n",
    "                exp = exp.join_experiments(data, field=cname)\n",
    "                exp.sample_metadata.loc[exp.sample_metadata['exp'].isnull(),'exp']=cname\n",
    "            total_ok += 1\n",
    "    print('finished. Total processed %d. Total ok %d. Not enough samples %d' % (idx+1, total_ok, total_low))\n",
    "    return exp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "**********\n",
      "processing ../studies/61 (0)\n",
      "['../studies/61/all.biom']\n",
      "2021-04-05 17:46:18 INFO loaded 41 samples, 2715 features\n",
      "2021-04-05 17:46:18 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:46:18 INFO After filtering, 41 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 41 samples, 2715 features\n",
      "not enough controls (21). skipping\n",
      "not enough disease (20). skipping\n",
      "keeping 41 total samples\n",
      "**********\n",
      "processing ../studies/59 (1)\n",
      "['../studies/59/all.biom']\n",
      "2021-04-05 17:46:18 INFO loaded 33 samples, 2637 features\n",
      "2021-04-05 17:46:18 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:46:18 INFO After filtering, 33 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 33 samples, 2637 features\n",
      "not enough controls (16). skipping\n",
      "not enough disease (17). skipping\n",
      "keeping 33 total samples\n",
      "**********\n",
      "processing ../studies/50 (2)\n",
      "['../studies/50/all.biom']\n",
      "2021-04-05 17:46:18 INFO loaded 58 samples, 959 features\n",
      "2021-04-05 17:46:18 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:46:18 INFO After filtering, 58 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 58 samples, 959 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/57 (3)\n",
      "['../studies/57/all.biom']\n",
      "2021-04-05 17:46:19 INFO loaded 85 samples, 4058 features\n",
      "2021-04-05 17:46:19 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:46:19 INFO After filtering, 85 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 85 samples, 4058 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/32 (4)\n",
      "['../studies/32/all.biom']\n",
      "2021-04-05 17:46:19 INFO loaded 43 samples, 1748 features\n",
      "2021-04-05 17:46:19 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:46:19 INFO After filtering, 43 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 43 samples, 1748 features\n",
      "not enough controls (13). skipping\n",
      "keeping 36 total samples\n",
      "**********\n",
      "processing ../studies/56 (5)\n",
      "['../studies/56/all.biom']\n",
      "2021-04-05 17:46:20 INFO loaded 43 samples, 3045 features\n",
      "2021-04-05 17:46:20 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:46:20 INFO After filtering, 43 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 43 samples, 3045 features\n",
      "not enough controls (22). skipping\n",
      "not enough disease (21). skipping\n",
      "keeping 43 total samples\n",
      "**********\n",
      "processing ../studies/51 (6)\n",
      "['../studies/51/all.biom']\n",
      "2021-04-05 17:46:21 INFO loaded 164 samples, 3240 features\n",
      "2021-04-05 17:46:21 WARNING These have metadata but do not have data - dropped (2): {'SRR10430029', 'SRR10429951'}\n",
      "2021-04-05 17:46:21 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:46:21 INFO After filtering, 164 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 164 samples, 3240 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/58 (8)\n",
      "['../studies/58/all.biom']\n",
      "2021-04-05 17:46:22 INFO loaded 45 samples, 3005 features\n",
      "2021-04-05 17:46:22 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:46:22 INFO After filtering, 45 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 45 samples, 3005 features\n",
      "not enough disease (22). skipping\n",
      "keeping 45 total samples\n",
      "**********\n",
      "processing ../studies/60 (9)\n",
      "['../studies/60/all.biom']\n",
      "2021-04-05 17:46:24 INFO loaded 73 samples, 3771 features\n",
      "2021-04-05 17:46:24 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:46:24 INFO After filtering, 73 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 73 samples, 3771 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/34 (10)\n",
      "['../studies/34/all.biom']\n",
      "2021-04-05 17:46:26 INFO loaded 32 samples, 1234 features\n",
      "2021-04-05 17:46:26 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:46:26 INFO After filtering, 32 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 32 samples, 1234 features\n",
      "not enough controls (14). skipping\n",
      "not enough disease (18). skipping\n",
      "keeping 32 total samples\n",
      "**********\n",
      "processing ../studies/33 (11)\n",
      "['../studies/33/all.biom']\n",
      "2021-04-05 17:46:27 INFO loaded 58 samples, 1250 features\n",
      "2021-04-05 17:46:27 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:46:27 INFO After filtering, 58 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 58 samples, 1250 features\n",
      "not enough disease (19). skipping\n",
      "keeping 42 total samples\n",
      "**********\n",
      "processing ../studies/20 (12)\n",
      "['../studies/20/all.biom']\n",
      "2021-04-05 17:46:29 INFO loaded 174 samples, 1622 features\n",
      "2021-04-05 17:46:29 WARNING These have metadata but do not have data - dropped (1): {'ERR2837004'}\n",
      "2021-04-05 17:46:29 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:46:29 INFO After filtering, 151 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 151 samples, 1622 features\n",
      "not enough disease (20). skipping\n",
      "keeping 43 total samples\n",
      "**********\n",
      "processing ../studies/18 (13)\n",
      "['../studies/18/all.biom']\n",
      "2021-04-05 17:46:31 INFO loaded 84 samples, 1183 features\n",
      "2021-04-05 17:46:31 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:46:31 INFO After filtering, 84 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 84 samples, 1183 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/27 (14)\n",
      "['../studies/27/all.biom']\n",
      "2021-04-05 17:46:33 INFO loaded 251 samples, 1403 features\n",
      "2021-04-05 17:46:33 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:46:33 INFO After filtering, 233 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 233 samples, 1403 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/9 (15)\n",
      "['../studies/9/all.biom']\n",
      "2021-04-05 17:46:35 INFO loaded 498 samples, 3722 features\n",
      "2021-04-05 17:46:35 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:46:35 INFO After filtering, 497 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 497 samples, 3722 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/11 (16)\n",
      "['../studies/11/all.biom']\n",
      "2021-04-05 17:46:38 INFO loaded 109 samples, 2312 features\n",
      "2021-04-05 17:46:38 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:46:38 INFO After filtering, 109 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 109 samples, 2312 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/7 (17)\n",
      "['../studies/7/all.biom']\n",
      "2021-04-05 17:46:41 INFO loaded 441 samples, 4403 features\n",
      "2021-04-05 17:46:41 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:46:41 INFO After filtering, 441 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 441 samples, 4403 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/29 (18)\n",
      "['../studies/29/all.biom']\n",
      "2021-04-05 17:46:45 INFO loaded 595 samples, 3470 features\n",
      "2021-04-05 17:46:45 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:46:45 INFO After filtering, 594 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 594 samples, 3470 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/16 (19)\n",
      "['../studies/16/all.biom']\n",
      "2021-04-05 17:46:49 INFO loaded 120 samples, 2116 features\n",
      "2021-04-05 17:46:49 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:46:49 INFO After filtering, 119 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 119 samples, 2116 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/42 (20)\n",
      "['../studies/42/all.biom']\n",
      "2021-04-05 17:46:54 INFO loaded 99 samples, 3829 features\n",
      "2021-04-05 17:46:54 WARNING These have metadata but do not have data - dropped (6): {'ERR1073898', 'ERR1315870', 'ERR1841805', 'ERR1417474', 'ERR2696902', 'ERR1545879'}\n",
      "2021-04-05 17:46:54 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:46:54 INFO After filtering, 96 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 96 samples, 3829 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/45 (21)\n",
      "['../studies/45/all.biom']\n",
      "2021-04-05 17:47:00 INFO loaded 1095 samples, 14981 features\n",
      "2021-04-05 17:47:00 WARNING These have metadata but do not have data - dropped (66): {'ERR1091564', 'ERR1545831', 'ERR1545996', 'ERR1545778', 'ERR1841761', 'ERR1389884', 'ERR2238631', 'ERR2239656', 'ERR1072642', 'ERR1841758', 'ERR1841817', 'ERR1075171', 'ERR1545722', 'ERR1545752', 'ERR1090995', 'ERR2318028', 'ERR1091321', 'ERR2303979', 'ERR1545981', 'ERR1545967', 'ERR1841716', 'ERR1073742', 'ERR1545883', 'ERR1160652', 'ERR1091606', 'ERR1841810', 'ERR1316257', 'ERR1545905', 'ERR2239754', 'ERR1546017', 'ERR1160500', 'ERR1545998', 'ERR1091145', 'ERR1841815', 'ERR1545894', 'ERR2239633', 'ERR1841814', 'ERR1841492', 'ERR1389937', 'ERR2238647', 'ERR1091364', 'ERR1090798', 'ERR1545957', 'ERR1841804', 'ERR1545906', 'ERR2696527', 'ERR1091654', 'ERR1091156', 'ERR1546023', 'ERR1545847', 'ERR2239699', 'ERR1074008', 'ERR1160588', 'ERR1089803', 'ERR1074155', 'ERR1545935', 'ERR1841512', 'ERR1075099', 'ERR1545921', 'ERR1459053', 'ERR1073096', 'ERR1091590', 'ERR1545915', 'ERR1073328', 'ERR1545994', 'ERR1160532'}\n",
      "2021-04-05 17:47:01 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:47:01 INFO After filtering, 1043 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 1043 samples, 14981 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/6 (22)\n",
      "['../studies/6/all.biom']\n",
      "2021-04-05 17:47:19 INFO loaded 620 samples, 4175 features\n",
      "2021-04-05 17:47:19 WARNING These have metadata but do not have data - dropped (4): {'ERR3519122', 'ERR3520757', 'ERR3520666', 'ERR3520927'}\n",
      "2021-04-05 17:47:20 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:47:20 INFO After filtering, 612 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 612 samples, 4175 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/28 (23)\n",
      "['../studies/28/all.biom']\n",
      "2021-04-05 17:47:37 INFO loaded 142 samples, 1306 features\n",
      "2021-04-05 17:47:37 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:47:37 INFO After filtering, 135 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 135 samples, 1306 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/17 (24)\n",
      "['../studies/17/all.biom']\n",
      "2021-04-05 17:47:54 INFO loaded 50 samples, 2307 features\n",
      "2021-04-05 17:47:54 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:47:54 INFO After filtering, 50 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 50 samples, 2307 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/1 (25)\n",
      "['../studies/1/all.biom']\n",
      "2021-04-05 17:48:12 INFO loaded 38 samples, 1082 features\n",
      "2021-04-05 17:48:12 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:48:12 INFO After filtering, 38 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 38 samples, 1082 features\n",
      "not enough controls (20). skipping\n",
      "not enough disease (18). skipping\n",
      "keeping 38 total samples\n",
      "**********\n",
      "processing ../studies/10 (26)\n",
      "['../studies/10/all.biom']\n",
      "2021-04-05 17:48:31 INFO loaded 174 samples, 3295 features\n",
      "2021-04-05 17:48:31 WARNING These have metadata but do not have data - dropped (2): {'SRR1689253', 'SRR1687461'}\n",
      "2021-04-05 17:48:31 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:48:31 INFO After filtering, 174 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 174 samples, 3295 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/19 (27)\n",
      "['../studies/19/all.biom']\n",
      "2021-04-05 17:48:54 INFO loaded 68 samples, 1401 features\n",
      "2021-04-05 17:48:54 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:48:54 INFO After filtering, 68 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 68 samples, 1401 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/26 (28)\n",
      "['../studies/26/all.biom']\n",
      "2021-04-05 17:49:13 INFO loaded 1388 samples, 11834 features\n",
      "2021-04-05 17:49:13 WARNING These have metadata but do not have data - dropped (41): {'DB3.234', 'DB3.229', 'DB3.223', 'DB3.218', 'DB3.196', 'DB3.228', 'DB2.239', 'DB3.230', 'DB3.269', 'DB3.110', 'DB3.072', 'DB3.036', 'DB3.251', 'DB3.047', 'DB3.071', 'DB3.239', 'DB3.250', 'DB3.102', 'DB3.108', 'DB2.241', 'DB2.135', 'DB3.101', 'DB3.217', 'DB3.109', 'DB3.212', 'DB2.245', 'DB3.078', 'DB3.096', 'DB3.106', 'DB2.242', 'DB2.240', 'DB3.107', 'DB3.103', 'DB3.104', 'DB2.076', 'DB3.231', 'DB3.266', 'DB2.069', 'DB3.265', 'DB3.219', 'DB3.264'}\n",
      "2021-04-05 17:49:13 WARNING These have data but do not have metadata: {'DB7.163', 'DB8.224', 'DB8.075', 'DB4.253', 'DB6.279', 'DB6.145', 'DB7.134', 'DB4.062', 'DB5.281', 'DB8.235', 'DB4.233', 'DB8.071', 'DB8.206', 'DB4.053', 'DB5.167', 'DB8.143', 'DB4.230', 'DB8.027', 'DB7.007', 'DB6.282', 'DB5.289', 'DB7.194', 'DB7.158', 'DB7.202', 'DB7.107', 'DB6.114', 'DB8.174', 'DB4.049', 'DB5.028', 'DB6.101', 'DB8.004', 'DB4.005', 'DB5.115', 'DB7.023', 'DB4.100', 'DB6.246', 'DB5.026', 'DB8.105', 'DB7.079', 'DB4.067', 'DB5.119', 'DB8.001', 'DB5.143', 'DB6.218', 'DB7.196', 'DB7.172', 'DB6.213', 'DB6.115', 'DB8.155', 'DB4.050', 'DB4.202', 'DB6.193', 'DB5.139', 'DB8.248', 'DB8.247', 'DB5.227', 'DB6.078', 'DB8.018', 'DB5.053', 'DB5.070', 'DB7.074', 'DB7.253', 'DB8.045', 'DB8.204', 'DB7.037', 'DB6.230', 'DB5.266', 'DB6.070', 'DB5.252', 'DB6.063', 'DB8.022', 'DB6.236', 'DB6.237', 'DB7.004', 'DB8.280', 'DB7.133', 'DB8.059', 'DB5.032', 'DB4.052', 'DB8.034', 'DB4.262', 'DB7.237', 'DB5.016', 'DB8.077', 'DB7.181', 'DB4.143', 'DB4.108', 'DB4.037', 'DB4.064', 'DB7.033', 'DB8.284', 'DB8.258', 'DB7.173', 'DB8.120', 'DB8.228', 'DB5.292', 'DB8.177', 'DB6.293', 'DB5.047', 'DB8.083', 'DB4.028', 'DB7.063', 'DB4.061', 'DB7.255', 'DB7.247', 'DB4.247', 'DB4.211', 'DB4.222', 'DB4.068', 'DB8.275', 'DB6.172', 'DB6.284', 'DB8.128', 'DB6.054', 'DB8.255', 'DB7.211', 'DB6.182', 'DB5.244', 'DB5.031', 'DB7.136', 'DB6.046', 'DB6.171', 'DB7.081', 'DB7.015', 'DB8.076', 'DB6.009', 'DB6.289', 'DB4.031', 'DB4.131', 'DB6.227', 'DB6.100', 'DB4.110', 'DB7.286', 'DB4.249', 'DB6.075', 'DB4.180', 'DB4.103', 'DB4.013', 'DB6.094', 'DB5.116', 'DB8.064', 'DB5.098', 'DB7.219', 'DB6.280', 'DB8.172', 'DB7.076', 'DB6.260', 'DB4.118', 'DB7.048', 'DB7.252', 'DB7.119', 'DB5.096', 'DB7.212', 'DB5.133', 'DB7.132', 'DB7.201', 'DB5.223', 'DB4.195', 'DB7.014', 'DB5.101', 'DB6.244', 'DB8.300', 'DB6.166', 'DB5.179', 'DB7.002', 'DB5.106', 'DB6.011', 'DB5.158', 'DB8.141', 'DB4.140', 'DB4.137', 'DB8.272', 'DB8.163', 'DB6.098', 'DB6.219', 'DB6.221', 'DB6.110', 'DB8.278', 'DB5.233', 'DB4.162', 'DB8.237', 'DB6.092', 'DB7.010', 'DB6.028', 'DB7.016', 'DB6.240', 'DB8.296', 'DB7.061', 'DB6.027', 'DB5.168', 'DB6.131', 'DB4.009', 'DB5.224', 'DB8.117', 'DB7.151', 'DB7.254', 'DB4.076', 'DB8.256', 'DB4.044', 'DB8.165', 'DB5.160', 'DB7.155', 'DB6.245', 'DB6.090', 'DB6.125', 'DB5.241', 'DB6.064', 'DB7.295', 'DB7.179', 'DB6.169', 'DB4.101', 'DB4.040', 'DB8.024', 'DB5.046', 'DB5.054', 'DB4.003', 'DB6.207', 'DB4.216', 'DB4.243', 'DB8.037', 'DB6.217', 'DB7.266', 'DB6.099', 'DB4.017', 'DB5.264', 'DB8.020', 'DB7.275', 'DB6.038', 'DB6.264', 'DB4.138', 'DB4.289', 'DB7.115', 'DB7.280', 'DB4.105', 'DB7.285', 'DB4.176', 'DB8.097', 'DB7.299', 'DB8.057', 'DB5.221', 'DB7.040', 'DB4.033', 'DB5.254', 'DB5.061', 'DB5.066', 'DB4.169', 'DB4.193', 'DB6.179', 'DB4.035', 'DB8.133', 'DB4.073', 'DB7.199', 'DB6.127', 'DB8.125', 'DB6.147', 'DB4.236', 'DB8.094', 'DB8.282', 'DB4.156', 'DB5.110', 'DB8.087', 'DB7.246', 'DB7.186', 'DB4.088', 'DB8.185', 'DB5.245', 'DB8.151', 'DB4.300', 'DB6.047', 'DB6.037', 'DB4.020', 'DB8.129', 'DB5.041', 'DB5.034', 'DB5.084', 'DB6.160', 'DB6.076', 'DB6.300', 'DB7.274', 'DB6.042', 'DB6.272', 'DB8.259', 'DB5.024', 'DB6.007', 'DB4.297', 'DB6.079', 'DB4.154', 'DB6.285', 'DB5.131', 'DB6.291', 'DB5.023', 'DB8.000', 'DB7.200', 'DB6.001', 'DB7.153', 'DB5.000', 'DB7.145', 'DB6.058', 'DB7.148', 'DB7.143', 'DB8.251', 'DB8.058', 'DB5.193', 'DB4.292', 'DB8.140', 'DB4.121', 'DB7.185', 'DB6.192', 'DB5.025', 'DB7.102', 'DB7.228', 'DB4.194', 'DB7.039', 'DB7.171', 'DB6.111', 'DB4.196', 'DB7.099', 'DB4.203', 'DB8.025', 'DB6.069', 'DB6.137', 'DB8.169', 'DB6.003', 'DB5.009', 'DB6.255', 'DB8.157', 'DB7.216', 'DB8.241', 'DB5.064', 'DB4.161', 'DB5.231', 'DB5.048', 'DB6.107', 'DB6.239', 'DB7.027', 'DB6.299', 'DB7.236', 'DB7.224', 'DB8.144', 'DB6.189', 'DB7.296', 'DB5.099', 'DB7.165', 'DB4.151', 'DB7.187', 'DB6.186', 'DB4.096', 'DB5.213', 'DB7.174', 'DB8.095', 'DB5.120', 'DB5.184', 'DB5.038', 'DB7.150', 'DB7.207', 'DB6.251', 'DB4.144', 'DB6.254', 'DB7.220', 'DB6.258', 'DB5.239', 'DB4.136', 'DB4.070', 'DB4.094', 'DB4.120', 'DB7.231', 'DB6.224', 'DB8.252', 'DB5.092', 'DB8.038', 'DB8.270', 'DB7.067', 'DB7.279', 'DB5.290', 'DB5.279', 'DB7.129', 'DB8.227', 'DB7.051', 'DB5.176', 'DB5.113', 'DB6.140', 'DB5.037', 'DB7.290', 'DB7.025', 'DB4.261', 'DB6.130', 'DB5.192', 'DB7.036', 'DB5.207', 'DB7.111', 'DB5.257', 'DB4.266', 'DB7.125', 'DB5.282', 'DB7.123', 'DB8.290', 'DB6.022', 'DB8.070', 'DB4.256', 'DB4.210', 'DB4.170', 'DB8.061', 'DB6.295', 'DB4.065', 'DB5.152', 'DB4.291', 'DB6.232', 'DB4.212', 'DB4.214', 'DB4.226', 'DB6.122', 'DB6.084', 'DB6.167', 'DB4.229', 'DB6.208', 'DB8.055', 'DB8.244', 'DB7.087', 'DB8.011', 'DB6.175', 'DB6.250', 'DB8.281', 'DB5.200', 'DB8.118', 'DB8.179', 'DB4.191', 'DB7.009', 'DB4.164', 'DB5.253', 'DB5.129', 'DB7.182', 'DB7.092', 'DB7.189', 'DB7.068', 'DB4.079', 'DB6.198', 'DB8.250', 'DB8.229', 'DB5.191', 'DB6.247', 'DB8.226', 'DB8.195', 'DB8.031', 'DB8.049', 'DB4.234', 'DB7.070', 'DB6.201', 'DB5.175', 'DB4.082', 'DB8.014', 'DB5.283', 'DB4.109', 'DB6.129', 'DB7.226', 'DB6.252', 'DB4.123', 'DB5.285', 'DB5.055', 'DB8.192', 'DB5.228', 'DB8.271', 'DB5.261', 'DB4.051', 'DB5.201', 'DB8.217', 'DB4.171', 'DB6.074', 'DB4.205', 'DB6.086', 'DB5.178', 'DB4.147', 'DB8.285', 'DB7.096', 'DB5.049', 'DB5.236', 'DB6.041', 'DB5.260', 'DB6.062', 'DB6.206', 'DB7.241', 'DB8.199', 'DB8.046', 'DB7.082', 'DB4.299', 'DB8.081', 'DB4.198', 'DB8.209', 'DB7.249', 'DB4.127', 'DB7.193', 'DB5.090', 'DB6.273', 'DB8.207', 'DB4.112', 'DB8.291', 'DB5.243', 'DB4.232', 'DB5.174', 'DB4.011', 'DB4.241', 'DB4.027', 'DB8.006', 'DB5.163', 'DB8.021', 'DB5.249', 'DB4.290', 'DB8.205', 'DB5.001', 'DB8.236', 'DB5.205', 'DB7.156', 'DB5.124', 'DB4.201', 'DB4.032', 'DB7.264', 'DB8.113', 'DB6.220', 'DB5.138', 'DB5.198', 'DB7.075', 'DB7.188', 'DB4.115', 'DB6.025', 'DB8.208', 'DB4.219', 'DB6.229', 'DB6.183', 'DB5.076', 'DB5.147', 'DB8.043', 'DB4.263', 'DB4.237', 'DB4.190', 'DB6.017', 'DB8.121', 'DB8.180', 'DB8.074', 'DB7.177', 'DB7.090', 'DB7.124', 'DB5.035', 'DB5.262', 'DB5.259', 'DB5.225', 'DB6.013', 'DB5.150', 'DB5.210', 'DB7.209', 'DB5.075', 'DB5.258', 'DB6.268', 'DB5.222', 'DB6.168', 'DB8.238', 'DB5.203', 'DB5.128', 'DB4.133', 'DB6.014', 'DB4.095', 'DB8.283', 'DB8.138', 'DB6.117', 'DB4.074', 'DB4.153', 'DB8.269', 'DB5.255', 'DB7.138', 'DB5.063', 'DB5.199', 'DB5.107', 'DB5.059', 'DB4.254', 'DB7.284', 'DB6.134', 'DB4.168', 'DB5.219', 'DB4.248', 'DB8.183', 'DB7.271', 'DB5.226', 'DB4.242', 'DB8.298', 'DB6.073', 'DB7.273', 'DB7.242', 'DB5.271', 'DB5.105', 'DB8.044', 'DB8.299', 'DB6.249', 'DB5.132', 'DB7.127', 'DB5.165', 'DB8.262', 'DB6.184', 'DB8.190', 'DB8.010', 'DB6.102', 'DB6.000', 'DB8.215', 'DB8.210', 'DB6.103', 'DB7.259', 'DB4.043', 'DB6.015', 'DB4.148', 'DB8.194', 'DB4.206', 'DB4.130', 'DB4.016', 'DB7.137', 'DB7.001', 'DB5.130', 'DB5.154', 'DB7.144', 'DB6.095', 'DB5.051', 'DB7.233', 'DB8.211', 'DB5.206', 'DB8.134', 'DB7.184', 'DB5.039', 'DB4.284', 'DB4.045', 'DB7.239', 'DB8.253', 'DB4.209', 'DB8.225', 'DB5.218', 'DB5.121', 'DB4.163', 'DB6.077', 'DB5.195', 'DB5.151', 'DB7.072', 'DB7.243', 'DB4.272', 'DB7.250', 'DB8.088', 'DB4.083', 'DB4.113', 'DB5.104', 'DB6.161', 'DB4.058', 'DB5.265', 'DB7.108', 'DB6.044', 'DB8.150', 'DB6.286', 'DB6.066', 'DB5.069', 'DB6.292', 'DB8.297', 'DB5.065', 'DB8.012', 'DB4.250', 'DB6.060', 'DB7.021', 'DB5.278', 'DB5.251', 'DB5.177', 'DB7.269', 'DB8.243', 'DB7.267', 'DB4.124', 'DB6.049', 'DB7.288', 'DB5.045', 'DB6.004', 'DB8.108', 'DB4.220', 'DB7.022', 'DB6.112', 'DB6.096', 'DB7.097', 'DB4.063', 'DB7.175', 'DB5.114', 'DB5.208', 'DB7.225', 'DB7.178', 'DB7.050', 'DB7.168', 'DB6.031', 'DB6.016', 'DB7.008', 'DB7.069', 'DB6.234', 'DB6.203', 'DB5.007', 'DB8.242', 'DB5.242', 'DB6.083', 'DB8.069', 'DB7.109', 'DB5.112', 'DB6.215', 'DB5.126', 'DB6.296', 'DB5.217', 'DB5.182', 'DB4.255', 'DB4.192', 'DB7.161', 'DB4.030', 'DB6.128', 'DB8.193', 'DB7.100', 'DB5.291', 'DB7.130', 'DB6.253', 'DB7.293', 'DB8.168', 'DB4.224', 'DB7.149', 'DB8.294', 'DB5.153', 'DB4.160', 'DB8.007', 'DB6.072', 'DB6.194', 'DB8.170', 'DB7.160', 'DB7.140', 'DB8.106', 'DB7.291', 'DB5.042', 'DB8.261', 'DB6.121', 'DB6.294', 'DB6.012', 'DB7.167', 'DB8.287', 'DB4.204', 'DB7.206', 'DB5.089', 'DB6.222', 'DB4.090', 'DB4.239', 'DB7.062', 'DB6.265', 'DB8.164', 'DB4.207', 'DB8.187', 'DB4.104', 'DB6.132', 'DB5.159', 'DB7.056', 'DB6.108', 'DB6.048', 'DB8.111', 'DB6.209', 'DB5.146', 'DB4.048', 'DB7.227', 'DB5.072', 'DB7.066', 'DB4.111', 'DB8.158', 'DB5.108', 'DB4.054', 'DB6.242', 'DB5.186', 'DB8.127', 'DB7.278', 'DB7.058', 'DB8.107', 'DB4.021', 'DB6.050', 'DB7.159', 'DB8.148', 'DB8.292', 'DB8.232', 'DB5.077', 'DB8.152', 'DB8.085', 'DB8.132', 'DB8.040', 'DB5.238', 'DB4.217', 'DB8.196', 'DB6.235', 'DB6.019', 'DB8.013', 'DB5.220', 'DB8.234', 'DB6.023', 'DB6.040', 'DB4.038', 'DB4.055', 'DB4.098', 'DB7.276', 'DB5.277', 'DB5.202', 'DB5.248', 'DB6.297', 'DB6.212', 'DB6.188', 'DB5.030', 'DB7.024', 'DB7.238', 'DB7.030', 'DB6.071', 'DB7.065', 'DB8.098', 'DB4.141', 'DB7.289', 'DB7.217', 'DB6.024', 'DB6.261', 'DB8.016', 'DB6.097', 'DB7.088', 'DB8.029', 'DB7.230', 'DB8.066', 'DB4.165', 'DB6.170', 'DB5.141', 'DB6.008', 'DB8.279', 'DB6.274', 'DB6.263', 'DB5.190', 'DB4.086', 'DB4.260', 'DB6.005', 'DB6.093', 'DB8.123', 'DB6.199', 'DB8.091', 'DB6.204', 'DB7.283', 'DB6.116', 'DB6.288', 'DB5.022', 'DB7.235', 'DB5.091', 'DB8.188', 'DB6.133', 'DB7.262', 'DB4.275', 'DB6.298', 'DB7.265', 'DB8.182', 'DB7.128', 'DB4.093', 'DB4.149', 'DB7.059', 'DB5.109', 'DB8.186', 'DB8.032', 'DB5.188', 'DB5.274', 'DB8.184', 'DB4.072', 'DB8.104', 'DB4.018', 'DB5.095', 'DB6.080', 'DB6.248', 'DB5.004', 'DB8.122', 'DB8.257', 'DB7.142', 'DB6.139', 'DB7.281', 'DB6.266', 'DB8.079', 'DB7.240', 'DB8.293', 'DB8.268', 'DB7.222', 'DB7.183', 'DB8.212', 'DB7.078', 'DB8.191', 'DB7.154', 'DB8.041', 'DB4.087', 'DB4.159', 'DB5.298', 'DB4.270', 'DB5.118', 'DB4.059', 'DB4.251', 'DB8.052', 'DB7.135', 'DB8.093', 'DB7.094', 'DB4.257', 'DB8.221', 'DB5.117', 'DB6.178', 'DB6.020', 'DB6.197', 'DB4.287', 'DB6.088', 'DB6.053', 'DB4.274', 'DB6.195', 'DB6.267', 'DB5.014', 'DB8.147', 'DB4.252', 'DB7.031', 'DB7.214', 'DB8.136', 'DB6.118', 'DB6.283', 'DB4.273', 'DB7.114', 'DB6.281', 'DB5.056', 'DB6.138', 'DB6.259', 'DB5.269', 'DB5.134', 'DB5.085', 'DB6.241', 'DB7.146', 'DB8.254', 'DB5.288', 'DB5.011', 'DB5.142', 'DB8.048', 'DB5.280', 'DB8.109', 'DB6.055', 'DB4.150', 'DB4.024', 'DB8.100', 'DB7.263', 'DB8.008', 'DB7.060', 'DB8.230', 'DB8.092', 'DB8.009', 'DB8.175', 'DB6.176', 'DB6.006', 'DB8.063', 'DB7.277', 'DB7.270', 'DB4.157', 'DB8.263', 'DB5.212', 'DB5.083', 'DB4.134', 'DB7.260', 'DB7.013', 'DB5.209', 'DB5.156', 'DB8.003', 'DB8.023', 'DB8.176', 'DB8.101', 'DB7.026', 'DB7.234', 'DB6.033', 'DB4.129', 'DB8.135', 'DB5.093', 'DB5.125', 'DB6.262', 'DB4.174', 'DB4.158', 'DB5.194', 'DB6.136', 'DB8.200', 'DB8.084', 'DB7.020', 'DB4.034', 'DB6.238', 'DB7.073', 'DB4.042', 'DB6.200', 'DB8.220', 'DB5.137', 'DB8.153', 'DB7.180', 'DB8.131', 'DB6.010', 'DB6.045', 'DB8.214', 'DB5.029', 'DB7.282', 'DB6.091', 'DB4.269', 'DB7.229', 'DB8.110', 'DB7.197', 'DB5.270', 'DB7.215', 'DB5.062', 'DB8.156', 'DB4.218', 'DB8.161', 'DB4.046', 'DB5.068', 'DB7.292', 'DB7.166', 'DB5.015', 'DB7.122', 'DB4.128', 'DB4.089', 'DB8.086', 'DB8.139', 'DB8.114', 'DB6.190', 'DB5.082', 'DB5.050', 'DB7.203', 'DB7.257', 'DB7.019', 'DB8.015', 'DB7.191', 'DB6.223', 'DB8.160', 'DB5.043', 'DB5.196', 'DB4.175', 'DB8.288', 'DB4.285', 'DB6.233', 'DB8.149', 'DB8.090', 'DB4.238', 'DB8.265', 'DB6.163', 'DB8.028', 'DB8.203', 'DB8.245', 'DB5.102', 'DB4.085', 'DB5.250', 'DB6.119', 'DB4.091', 'DB7.103', 'DB4.227', 'DB7.105', 'DB4.022', 'DB6.018', 'DB7.258', 'DB7.044', 'DB5.204', 'DB4.146', 'DB6.106', 'DB5.275', 'DB7.272', 'DB5.044', 'DB8.216', 'DB5.229', 'DB5.036', 'DB5.155', 'DB5.100', 'DB6.191', 'DB7.162', 'DB5.164', 'DB8.154', 'DB7.261', 'DB7.157', 'DB5.215', 'DB4.023', 'DB4.125', 'DB7.104', 'DB5.060', 'DB5.079', 'DB7.298', 'DB6.205', 'DB7.198', 'DB7.190', 'DB7.045', 'DB7.003', 'DB4.298', 'DB5.211', 'DB8.246', 'DB5.272', 'DB6.162', 'DB4.245', 'DB8.286', 'DB5.123', 'DB8.017', 'DB8.198', 'DB7.064', 'DB5.235', 'DB4.047', 'DB6.181', 'DB4.066', 'DB6.214', 'DB5.187', 'DB4.126', 'DB7.287', 'DB8.124', 'DB8.078', 'DB5.080', 'DB5.149', 'DB7.131', 'DB5.145', 'DB5.169', 'DB7.117', 'DB7.080', 'DB6.165', 'DB4.152', 'DB5.161', 'DB7.121', 'DB7.256', 'DB5.057', 'DB8.273', 'DB8.126', 'DB5.214', 'DB8.239', 'DB7.169', 'DB6.032', 'DB5.197', 'DB5.171', 'DB8.102', 'DB7.170', 'DB8.116', 'DB5.087', 'DB8.178', 'DB5.185', 'DB4.145', 'DB7.084', 'DB4.081', 'DB8.145', 'DB7.126', 'DB8.026', 'DB6.067', 'DB7.120', 'DB8.089', 'DB6.157', 'DB6.216', 'DB6.135', 'DB7.210', 'DB6.270', 'DB7.042', 'DB6.174', 'DB5.263', 'DB4.213', 'DB4.167', 'DB8.080', 'DB8.146', 'DB5.135', 'DB5.027', 'DB7.141', 'DB6.087', 'DB6.187', 'DB4.025', 'DB7.110', 'DB8.030', 'DB6.057', 'DB6.126', 'DB5.127', 'DB5.074', 'DB7.095', 'DB7.038', 'DB8.062', 'DB7.054', 'DB4.099', 'DB6.113', 'DB8.099', 'DB6.089', 'DB7.098', 'DB8.295', 'DB7.195', 'DB6.196', 'DB5.246', 'DB6.034', 'DB5.086', 'DB7.006', 'DB4.026', 'DB4.056', 'DB8.274', 'DB5.094', 'DB6.243', 'DB7.071', 'DB7.251', 'DB7.046', 'DB4.069', 'DB4.155', 'DB7.053', 'DB8.005', 'DB5.078', 'DB4.084', 'DB4.223', 'DB7.047', 'DB7.223', 'DB7.268', 'DB7.018', 'DB6.036', 'DB8.112', 'DB7.049', 'DB8.249', 'DB6.124', 'DB6.257', 'DB8.231', 'DB8.201', 'DB7.000', 'DB4.228', 'DB5.247', 'DB8.019', 'DB4.132', 'DB8.267', 'DB5.256', 'DB8.218', 'DB7.204', 'DB5.286', 'DB7.052', 'DB7.057', 'DB4.004', 'DB7.248', 'DB6.155', 'DB4.078', 'DB7.118', 'DB5.010', 'DB4.267', 'DB7.221', 'DB4.041', 'DB8.213', 'DB4.268', 'DB7.011', 'DB6.211', 'DB5.071', 'DB7.116', 'DB4.057', 'DB5.297', 'DB7.208', 'DB6.173', 'DB6.123', 'DB6.226', 'DB5.111', 'DB8.240', 'DB8.233', 'DB7.232', 'DB6.177', 'DB4.060', 'DB5.136', 'DB4.244', 'DB5.013', 'DB6.225', 'DB7.300', 'DB5.103', 'DB7.106', 'DB8.173', 'DB7.113', 'DB6.051', 'DB8.171', 'DB8.137', 'DB6.105', 'DB8.266', 'DB4.235', 'DB5.033', 'DB8.119', 'DB8.202', 'DB4.107', 'DB4.036', 'DB5.144', 'DB6.002', 'DB5.284', 'DB7.139', 'DB4.281', 'DB5.073', 'DB5.268', 'DB4.221', 'DB5.148', 'DB8.082', 'DB5.008', 'DB4.208', 'DB5.240', 'DB4.001', 'DB5.040', 'DB7.093', 'DB7.213', 'DB6.109', 'DB4.097', 'DB8.103', 'DB5.180', 'DB8.039', 'DB7.055', 'DB7.245', 'DB4.215', 'DB5.273', 'DB7.192', 'DB7.041', 'DB6.202', 'DB4.259', 'DB4.075', 'DB5.097', 'DB8.035', 'DB8.260', 'DB5.232', 'DB8.219', 'DB7.218', 'DB5.181', 'DB4.177', 'DB7.034', 'DB7.244', 'DB6.043', 'DB5.067', 'DB7.152', 'DB7.035', 'DB5.157', 'DB6.056', 'DB8.115', 'DB6.228', 'DB8.181', 'DB7.176', 'DB4.135', 'DB4.172', 'DB7.017', 'DB4.283', 'DB7.205', 'DB5.237', 'DB5.189', 'DB6.120', 'DB5.216', 'DB5.183', 'DB8.042', 'DB5.230', 'DB6.030', 'DB8.002', 'DB8.159', 'DB7.164', 'DB4.142', 'DB6.290', 'DB7.089', 'DB4.077', 'DB6.104', 'DB7.294', 'DB5.287', 'DB4.173', 'DB7.091', 'DB6.068', 'DB6.065', 'DB6.164', 'DB5.276', 'DB7.083', 'DB6.082', 'DB4.225', 'DB7.112', 'DB8.289', 'DB6.231', 'DB5.267', 'DB6.035', 'DB7.147', 'DB5.081', 'DB4.071', 'DB4.122', 'DB6.039', 'DB5.122', 'DB7.005', 'DB7.043', 'DB6.052', 'DB6.185', 'DB5.058', 'DB8.036', 'DB7.101', 'DB4.039', 'DB6.029', 'DB5.052', 'DB8.142'}\n",
      "2021-04-05 17:49:13 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:49:13 INFO After filtering, 1172 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 1172 samples, 11834 features\n",
      "not enough controls (13). skipping\n",
      "not enough disease (12). skipping\n",
      "keeping 25 total samples\n",
      "**********\n",
      "processing ../studies/8 (29)\n",
      "['../studies/8/all.biom']\n",
      "2021-04-05 17:49:53 INFO loaded 33 samples, 780 features\n",
      "2021-04-05 17:49:53 WARNING These have metadata but do not have data - dropped (1): {'ERR315306'}\n",
      "2021-04-05 17:49:53 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:49:53 INFO After filtering, 33 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 33 samples, 780 features\n",
      "not enough controls (11). skipping\n",
      "not enough disease (22). skipping\n",
      "keeping 33 total samples\n",
      "**********\n",
      "processing ../studies/21 (30)\n",
      "['../studies/21/all.biom']\n",
      "2021-04-05 17:50:27 INFO loaded 178 samples, 2597 features\n",
      "2021-04-05 17:50:27 WARNING These have metadata but do not have data - dropped (1): {'SRR6147218'}\n",
      "2021-04-05 17:50:28 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:50:28 INFO After filtering, 178 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 178 samples, 2597 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/44 (31)\n",
      "['../studies/44/all.biom']\n",
      "2021-04-05 17:51:10 INFO loaded 875 samples, 11915 features\n",
      "2021-04-05 17:51:10 WARNING These have metadata but do not have data - dropped (64): {'ERR1545911', 'ERR1160673', 'ERR1545891', 'ERR1090161', 'ERR1545962', 'ERR1545986', 'ERR1545978', 'ERR1091354', 'ERR1089679', 'ERR1545802', 'ERR1074779', 'ERR1091155', 'ERR1545907', 'ERR1841813', 'ERR1075185', 'ERR1545774', 'ERR1074050', 'ERR1074092', 'ERR1545804', 'ERR1074007', 'ERR1160680', 'ERR1546016', 'ERR1072778', 'ERR1841525', 'ERR1597274', 'ERR1073198', 'ERR1089814', 'ERR1545780', 'ERR1074974', 'ERR1075896', 'ERR1073883', 'ERR1073330', 'ERR1075069', 'ERR1842199', 'ERR1545963', 'ERR1075272', 'ERR1546010', 'ERR2239717', 'ERR1074978', 'ERR1545939', 'ERR1545936', 'ERR1074005', 'ERR1072979', 'ERR1091147', 'ERR1545877', 'ERR2242158', 'ERR1080384', 'ERR1545747', 'ERR1545781', 'ERR1160475', 'ERR1074955', 'ERR1389624', 'ERR1545838', 'ERR1072794', 'ERR1545833', 'ERR1545990', 'ERR1545702', 'ERR1545803', 'ERR1546057', 'ERR1597126', 'ERR1389819', 'ERR1075071', 'ERR1072781', 'ERR1075286'}\n",
      "2021-04-05 17:51:11 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:51:11 INFO After filtering, 835 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 835 samples, 11915 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/43 (32)\n",
      "['../studies/43/all.biom']\n",
      "2021-04-05 17:52:14 INFO loaded 91 samples, 3426 features\n",
      "2021-04-05 17:52:14 WARNING These have metadata but do not have data - dropped (9): {'ERR1545914', 'ERR1545813', 'ERR1075067', 'ERR1545849', 'ERR1841752', 'ERR1545876', 'ERR2239660', 'ERR1841803', 'ERR1073734'}\n",
      "2021-04-05 17:52:14 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:52:14 INFO After filtering, 89 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 89 samples, 3426 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/36 (33)\n",
      "['../studies/36/all.biom']\n",
      "2021-04-05 17:53:09 INFO loaded 16 samples, 1191 features\n",
      "2021-04-05 17:53:09 WARNING These have metadata but do not have data - dropped (1): {'ERR1089974'}\n",
      "2021-04-05 17:53:09 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:53:09 INFO After filtering, 15 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 15 samples, 1191 features\n",
      "not enough controls (9). skipping\n",
      "not enough disease (6). skipping\n",
      "keeping 15 total samples\n",
      "**********\n",
      "processing ../studies/31 (34)\n",
      "['../studies/31/all.biom']\n",
      "2021-04-05 17:54:00 INFO loaded 70 samples, 1929 features\n",
      "2021-04-05 17:54:00 WARNING These have metadata but do not have data - dropped (18): {'ERR1331868', 'ERR1331858', 'ERR1331856', 'ERR1331867', 'ERR1331866', 'ERR1331871', 'ERR1331872', 'ERR1331865', 'ERR1331860', 'ERR1331869', 'ERR1331870', 'ERR1331864', 'ERR1331863', 'ERR1331862', 'ERR1331859', 'ERR1331861', 'ERR1331814', 'ERR1331857'}\n",
      "2021-04-05 17:54:00 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:54:00 INFO After filtering, 70 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 70 samples, 1929 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/62 (35)\n",
      "['../studies/62/all.biom']\n",
      "2021-04-05 17:54:47 INFO loaded 196 samples, 5813 features\n",
      "2021-04-05 17:54:47 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:54:47 INFO After filtering, 196 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 196 samples, 5813 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/54 (36)\n",
      "['../studies/54/all.biom']\n",
      "2021-04-05 17:55:45 INFO loaded 63 samples, 3657 features\n",
      "2021-04-05 17:55:45 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:55:45 INFO After filtering, 63 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 63 samples, 3657 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/53 (37)\n",
      "['../studies/53/all.biom']\n",
      "2021-04-05 17:56:47 INFO loaded 115 samples, 4878 features\n",
      "2021-04-05 17:56:48 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:56:48 INFO After filtering, 115 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 115 samples, 4878 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/37 (38)\n",
      "['../studies/37/all.biom']\n",
      "2021-04-05 17:57:43 INFO loaded 757 samples, 12055 features\n",
      "2021-04-05 17:57:43 WARNING These have metadata but do not have data - dropped (63): {'ERR1072865', 'ERR1545954', 'ERR1545899', 'ERR1841696', 'ERR1545922', 'ERR1459193', 'ERR1458945', 'ERR2580081', 'ERR1545999', 'ERR1316091', 'ERR1091311', 'ERR1072872', 'ERR1545923', 'ERR1091687', 'ERR2239311', 'ERR1545837', 'ERR2239796', 'ERR1075799', 'ERR1545979', 'ERR1091094', 'ERR1545863', 'ERR1077624', 'ERR1080064', 'ERR2239648', 'ERR1075259', 'ERR1091073', 'ERR1091610', 'ERR1545856', 'ERR1315992', 'ERR1072691', 'ERR1075115', 'ERR1080269', 'ERR1545860', 'ERR1545721', 'ERR1545995', 'ERR1545931', 'ERR1073095', 'ERR1073219', 'ERR1458825', 'ERR2239672', 'ERR2239737', 'ERR1842276', 'ERR1841719', 'ERR1841811', 'ERR1091136', 'ERR1546066', 'ERR1545835', 'ERR1389679', 'ERR1546000', 'ERR1075296', 'ERR1545811', 'ERR1090142', 'ERR1389829', 'ERR2238737', 'ERR1091704', 'ERR2239190', 'ERR1076845', 'ERR1091456', 'ERR1074977', 'ERR1545934', 'ERR1841820', 'ERR1545779', 'ERR1072933'}\n",
      "2021-04-05 17:57:43 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:57:43 INFO After filtering, 727 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 727 samples, 12055 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/39 (39)\n",
      "['../studies/39/all.biom']\n",
      "2021-04-05 17:58:53 INFO loaded 625 samples, 10422 features\n",
      "2021-04-05 17:58:53 WARNING These have metadata but do not have data - dropped (35): {'ERR2238658', 'ERR1074953', 'ERR1841821', 'ERR1074855', 'ERR2239697', 'ERR1545875', 'ERR1854699', 'ERR2239793', 'ERR1080069', 'ERR2239393', 'ERR1073258', 'ERR2239770', 'ERR1841495', 'ERR1075897', 'ERR2318018', 'ERR1072703', 'ERR1545942', 'ERR1073829', 'ERR1545903', 'ERR1545815', 'ERR1842590', 'ERR2238750', 'ERR1545969', 'ERR1077488', 'ERR1089998', 'ERR1389924', 'ERR1074160', 'ERR1842514', 'ERR2239308', 'ERR1072835', 'ERR1841812', 'ERR1545985', 'ERR1545842', 'ERR1389959', 'ERR1076837'}\n",
      "2021-04-05 17:58:54 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 17:58:54 INFO After filtering, 587 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 587 samples, 10422 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/52 (40)\n",
      "['../studies/52/all.biom']\n",
      "2021-04-05 18:00:07 INFO loaded 30 samples, 2580 features\n",
      "2021-04-05 18:00:07 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 18:00:07 INFO After filtering, 30 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 30 samples, 2580 features\n",
      "not enough controls (17). skipping\n",
      "not enough disease (13). skipping\n",
      "keeping 30 total samples\n",
      "**********\n",
      "processing ../studies/55 (41)\n",
      "['../studies/55/all.biom']\n",
      "2021-04-05 18:01:05 INFO loaded 74 samples, 3939 features\n",
      "2021-04-05 18:01:05 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 18:01:05 INFO After filtering, 74 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 74 samples, 3939 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/46 (42)\n",
      "['../studies/46/all.biom']\n",
      "2021-04-05 18:02:04 INFO loaded 584 samples, 10980 features\n",
      "2021-04-05 18:02:04 WARNING These have metadata but do not have data - dropped (41): {'ERR1545904', 'ERR2238718', 'ERR1075338', 'ERR2239702', 'ERR1073073', 'ERR1545701', 'ERR1077699', 'ERR1546003', 'ERR1089788', 'ERR1545968', 'ERR1459007', 'ERR1546040', 'ERR1854779', 'ERR1075243', 'ERR2304178', 'ERR1841755', 'ERR1074261', 'ERR1074885', 'ERR1841818', 'ERR1074071', 'ERR1080238', 'ERR1545694', 'ERR1841546', 'ERR1089939', 'ERR1075226', 'ERR1545678', 'ERR1545890', 'ERR1545910', 'ERR1073933', 'ERR1545902', 'ERR1545916', 'ERR2239804', 'ERR1073802', 'ERR1546027', 'ERR1073744', 'ERR1545859', 'ERR1389664', 'ERR1545965', 'ERR2242536', 'ERR2239582', 'ERR1545961'}\n",
      "2021-04-05 18:02:04 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 18:02:04 INFO After filtering, 554 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 554 samples, 10980 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/41 (43)\n",
      "['../studies/41/all.biom']\n",
      "2021-04-05 18:03:23 INFO loaded 84 samples, 3557 features\n",
      "2021-04-05 18:03:23 WARNING These have metadata but do not have data - dropped (5): {'ERR1075045', 'ERR2238709', 'ERR1545772', 'ERR1160360', 'ERR2238703'}\n",
      "2021-04-05 18:03:23 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 18:03:23 INFO After filtering, 80 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 80 samples, 3557 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/48 (44)\n",
      "['../studies/48/all.biom']\n",
      "2021-04-05 18:04:21 INFO loaded 293 samples, 6184 features\n",
      "2021-04-05 18:04:21 WARNING These have metadata but do not have data - dropped (21): {'ERR2239290', 'ERR1075875', 'ERR1090826', 'ERR1545775', 'ERR1545869', 'ERR2308577', 'ERR1545878', 'ERR1160560', 'ERR2238646', 'ERR1841718', 'ERR1075817', 'ERR1089851', 'ERR1091689', 'ERR1546021', 'ERR1546022', 'ERR1089874', 'ERR1073280', 'ERR1089880', 'ERR2033565', 'ERR1091390', 'ERR1841798'}\n",
      "2021-04-05 18:04:21 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 18:04:21 INFO After filtering, 280 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 280 samples, 6184 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/24 (45)\n",
      "['../studies/24/all.biom']\n",
      "2021-04-05 18:05:15 INFO loaded 25 samples, 999 features\n",
      "2021-04-05 18:05:15 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 18:05:15 INFO After filtering, 25 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 25 samples, 999 features\n",
      "not enough controls (10). skipping\n",
      "not enough disease (15). skipping\n",
      "keeping 25 total samples\n",
      "**********\n",
      "processing ../studies/23 (46)\n",
      "['../studies/23/all.biom']\n",
      "2021-04-05 18:06:02 INFO loaded 114 samples, 1867 features\n",
      "2021-04-05 18:06:02 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 18:06:02 INFO After filtering, 114 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 114 samples, 1867 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/4 (47)\n",
      "['../studies/4/all.biom']\n",
      "2021-04-05 18:06:54 INFO loaded 83 samples, 1652 features\n",
      "2021-04-05 18:06:54 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 18:06:54 INFO After filtering, 82 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 82 samples, 1652 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/15 (48)\n",
      "['../studies/15/all.biom']\n",
      "2021-04-05 18:07:47 INFO loaded 174 samples, 2103 features\n",
      "2021-04-05 18:07:47 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 18:07:47 INFO After filtering, 162 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 162 samples, 2103 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/3 (49)\n",
      "['../studies/3/all.biom']\n",
      "2021-04-05 18:08:41 INFO loaded 337 samples, 3021 features\n",
      "2021-04-05 18:08:41 WARNING These have metadata but do not have data - dropped (11): {'ERR1513792', 'ERR1513697', 'ERR1513809', 'ERR1513852', 'ERR1513897', 'ERR1513791', 'ERR1513823', 'ERR1513668', 'ERR1513848', 'ERR1513720', 'ERR1513893'}\n",
      "2021-04-05 18:08:41 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 18:08:41 INFO After filtering, 334 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 334 samples, 3021 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/12 (50)\n",
      "['../studies/12/all.biom']\n",
      "2021-04-05 18:09:36 INFO loaded 2664 samples, 13732 features\n",
      "2021-04-05 18:09:36 WARNING These have data but do not have metadata: {'DB7.163', 'DB9.089', 'DB3.063', 'DB3.117', 'DB4.253', 'DB6.279', 'DB10.263', 'DB2.134', 'DB4.062', 'DB8.235', 'DB10.105', 'DB4.233', 'DB3.053', 'DB9.253', 'DB4.053', 'DB8.206', 'DB8.143', 'DB6.282', 'DB9.110', 'DB5.289', 'DB2.092', 'DB3.107', 'DB7.158', 'DB8.174', 'DB4.049', 'DB2.030', 'DB5.028', 'DB1.270', 'DB8.004', 'DB6.246', 'DB10.269', 'DB7.079', 'DB2.122', 'DB4.067', 'DB9.091', 'DB5.003', 'DB8.001', 'DB3.026', 'DB5.143', 'DB6.218', 'DB9.188', 'DB6.213', 'DB9.283', 'DB8.155', 'DB10.014', 'DB4.050', 'DB6.193', 'DB9.291', 'DB2.222', 'DB8.247', 'DB5.227', 'DB2.025', 'DB10.170', 'DB3.004', 'DB8.018', 'DB10.053', 'DB2.009', 'DB9.144', 'DB2.149', 'DB3.116', 'DB7.074', 'DB7.253', 'DB10.132', 'DB6.230', 'DB3.157', 'DB10.282', 'DB6.029', 'DB9.194', 'DB9.154', 'DB3.166', 'DB9.216', 'DB6.063', 'DB6.236', 'DB6.237', 'DB8.280', 'DB7.133', 'DB2.183', 'DB4.052', 'DB9.118', 'DB9.038', 'DB4.037', 'DB2.096', 'DB4.064', 'DB9.117', 'DB9.251', 'DB7.173', 'DB8.120', 'DB6.293', 'DB8.177', 'DB1.129', 'DB8.083', 'DB4.028', 'DB4.061', 'DB3.064', 'DB7.247', 'DB4.247', 'DB3.091', 'DB4.211', 'DB4.068', 'DB4.222', 'DB6.284', 'DB3.122', 'DB6.054', 'DB8.255', 'DB2.044', 'DB5.031', 'DB3.206', 'DB9.102', 'DB7.081', 'DB3.033', 'DB6.289', 'DB4.031', 'DB4.131', 'DB3.209', 'DB2.101', 'DB2.148', 'DB4.110', 'DB6.100', 'DB7.286', 'DB3.013', 'DB3.054', 'DB2.036', 'DB9.098', 'DB4.103', 'DB3.110', 'DB9.095', 'DB4.013', 'DB2.212', 'DB9.059', 'DB2.139', 'DB6.094', 'DB5.116', 'DB5.098', 'DB9.207', 'DB2.067', 'DB9.058', 'DB3.096', 'DB6.260', 'DB9.139', 'DB7.048', 'DB7.252', 'DB5.096', 'DB2.240', 'DB2.068', 'DB7.201', 'DB7.014', 'DB9.056', 'DB2.089', 'DB6.011', 'DB5.158', 'DB4.140', 'DB3.260', 'DB8.141', 'DB9.167', 'DB6.098', 'DB6.219', 'DB6.110', 'DB9.244', 'DB4.162', 'DB8.237', 'DB6.092', 'DB7.010', 'DB7.016', 'DB6.240', 'DB9.074', 'DB3.102', 'DB10.094', 'DB6.131', 'DB8.117', 'DB3.086', 'DB7.151', 'DB7.254', 'DB10.104', 'DB10.069', 'DB4.076', 'DB9.163', 'DB8.256', 'DB9.147', 'DB10.131', 'DB8.165', 'DB2.076', 'DB3.022', 'DB5.160', 'DB2.205', 'DB7.155', 'DB2.028', 'DB5.241', 'DB9.276', 'DB9.195', 'DB6.064', 'DB9.077', 'DB9.048', 'DB10.055', 'DB3.095', 'DB10.286', 'DB6.169', 'DB10.089', 'DB2.186', 'DB4.040', 'DB5.046', 'DB2.180', 'DB2.074', 'DB4.216', 'DB1.299', 'DB6.217', 'DB7.266', 'DB8.020', 'DB9.099', 'DB6.038', 'DB6.264', 'DB9.158', 'DB3.159', 'DB9.015', 'DB4.138', 'DB9.247', 'DB7.115', 'DB7.280', 'DB4.105', 'DB9.019', 'DB2.083', 'DB5.221', 'DB2.191', 'DB5.254', 'DB5.066', 'DB6.179', 'DB4.271', 'DB9.192', 'DB4.035', 'DB8.133', 'DB6.127', 'DB4.236', 'DB8.125', 'DB8.094', 'DB8.282', 'DB1.064', 'DB8.087', 'DB4.088', 'DB8.185', 'DB1.278', 'DB2.174', 'DB8.151', 'DB6.047', 'DB6.037', 'DB1.271', 'DB9.042', 'DB9.006', 'DB4.020', 'DB9.014', 'DB5.041', 'DB5.034', 'DB5.084', 'DB9.222', 'DB6.042', 'DB7.274', 'DB10.020', 'DB1.096', 'DB8.259', 'DB9.219', 'DB3.043', 'DB1.275', 'DB5.024', 'DB6.007', 'DB9.119', 'DB2.201', 'DB5.131', 'DB6.291', 'DB1.078', 'DB9.070', 'DB3.149', 'DB9.023', 'DB10.211', 'DB5.023', 'DB9.069', 'DB2.120', 'DB3.202', 'DB3.097', 'DB2.061', 'DB7.153', 'DB3.008', 'DB9.106', 'DB5.006', 'DB6.058', 'DB7.145', 'DB1.065', 'DB9.295', 'DB3.098', 'DB9.108', 'DB10.035', 'DB2.199', 'DB8.140', 'DB7.185', 'DB9.241', 'DB7.102', 'DB7.228', 'DB3.245', 'DB7.171', 'DB2.192', 'DB9.169', 'DB3.171', 'DB6.137', 'DB8.157', 'DB7.216', 'DB9.109', 'DB10.183', 'DB2.102', 'DB5.064', 'DB8.241', 'DB9.135', 'DB6.107', 'DB7.027', 'DB7.236', 'DB7.224', 'DB8.144', 'DB1.273', 'DB3.267', 'DB9.126', 'DB1.276', 'DB3.153', 'DB2.161', 'DB2.112', 'DB2.095', 'DB7.165', 'DB4.151', 'DB2.170', 'DB6.186', 'DB2.159', 'DB4.096', 'DB10.206', 'DB1.193', 'DB2.220', 'DB8.095', 'DB10.276', 'DB5.184', 'DB5.038', 'DB7.150', 'DB4.144', 'DB3.016', 'DB10.060', 'DB4.070', 'DB5.239', 'DB6.258', 'DB7.220', 'DB3.129', 'DB4.120', 'DB3.135', 'DB7.231', 'DB6.224', 'DB10.191', 'DB8.270', 'DB10.159', 'DB1.150', 'DB7.279', 'DB2.082', 'DB10.057', 'DB8.227', 'DB2.049', 'DB3.228', 'DB9.239', 'DB10.215', 'DB6.140', 'DB5.037', 'DB2.173', 'DB9.181', 'DB10.086', 'DB2.014', 'DB2.077', 'DB3.126', 'DB3.029', 'DB9.094', 'DB6.022', 'DB4.256', 'DB4.210', 'DB10.294', 'DB10.289', 'DB2.225', 'DB3.192', 'DB4.212', 'DB4.226', 'DB2.164', 'DB2.178', 'DB10.037', 'DB6.084', 'DB3.188', 'DB4.229', 'DB10.242', 'DB2.206', 'DB8.244', 'DB7.087', 'DB3.067', 'DB10.180', 'DB6.250', 'DB8.118', 'DB8.179', 'DB9.200', 'DB8.050', 'DB9.004', 'DB10.048', 'DB3.136', 'DB5.253', 'DB7.092', 'DB7.189', 'DB7.068', 'DB3.007', 'DB6.198', 'DB8.250', 'DB3.078', 'DB10.298', 'DB5.191', 'DB3.061', 'DB10.237', 'DB10.220', 'DB8.031', 'DB7.070', 'DB6.201', 'DB2.109', 'DB8.014', 'DB10.277', 'DB3.165', 'DB9.137', 'DB10.107', 'DB3.094', 'DB2.034', 'DB2.023', 'DB8.192', 'DB5.228', 'DB5.261', 'DB4.051', 'DB10.250', 'DB10.139', 'DB10.067', 'DB10.205', 'DB5.201', 'DB4.171', 'DB8.217', 'DB6.074', 'DB2.187', 'DB4.205', 'DB3.115', 'DB2.202', 'DB5.178', 'DB4.147', 'DB7.096', 'DB10.066', 'DB5.236', 'DB10.052', 'DB1.282', 'DB5.260', 'DB7.241', 'DB9.084', 'DB2.210', 'DB8.046', 'DB7.082', 'DB4.299', 'DB8.081', 'DB4.127', 'DB10.003', 'DB3.154', 'DB2.153', 'DB7.193', 'DB4.112', 'DB9.243', 'DB4.011', 'DB2.163', 'DB9.124', 'DB4.241', 'DB9.196', 'DB8.006', 'DB9.114', 'DB3.263', 'DB10.221', 'DB2.029', 'DB8.205', 'DB8.236', 'DB2.228', 'DB7.156', 'DB9.025', 'DB4.201', 'DB4.032', 'DB3.197', 'DB8.113', 'DB6.220', 'DB5.138', 'DB5.198', 'DB10.054', 'DB6.025', 'DB10.200', 'DB2.003', 'DB2.105', 'DB5.076', 'DB9.231', 'DB4.263', 'DB8.043', 'DB9.017', 'DB3.241', 'DB4.190', 'DB9.278', 'DB6.017', 'DB8.121', 'DB9.087', 'DB7.177', 'DB7.124', 'DB9.249', 'DB6.013', 'DB9.068', 'DB5.150', 'DB10.280', 'DB3.001', 'DB9.285', 'DB3.168', 'DB5.075', 'DB5.258', 'DB6.268', 'DB5.222', 'DB9.010', 'DB9.172', 'DB8.238', 'DB6.014', 'DB10.151', 'DB8.138', 'DB10.295', 'DB9.229', 'DB2.143', 'DB5.063', 'DB7.138', 'DB5.059', 'DB3.243', 'DB2.097', 'DB4.248', 'DB8.183', 'DB5.226', 'DB7.271', 'DB4.242', 'DB10.296', 'DB6.073', 'DB7.273', 'DB9.213', 'DB7.242', 'DB8.044', 'DB2.235', 'DB6.249', 'DB8.299', 'DB5.132', 'DB7.127', 'DB5.165', 'DB8.073', 'DB6.184', 'DB2.080', 'DB3.123', 'DB9.300', 'DB6.000', 'DB9.073', 'DB9.179', 'DB6.103', 'DB10.077', 'DB10.266', 'DB8.194', 'DB10.059', 'DB4.130', 'DB2.001', 'DB7.137', 'DB7.001', 'DB5.130', 'DB5.154', 'DB9.046', 'DB5.051', 'DB5.206', 'DB7.233', 'DB8.211', 'DB10.091', 'DB7.184', 'DB8.134', 'DB9.013', 'DB7.239', 'DB10.051', 'DB9.125', 'DB6.077', 'DB2.004', 'DB5.151', 'DB7.243', 'DB4.272', 'DB1.152', 'DB2.078', 'DB9.237', 'DB10.265', 'DB8.088', 'DB4.113', 'DB9.036', 'DB9.208', 'DB5.265', 'DB7.108', 'DB6.044', 'DB6.286', 'DB6.066', 'DB3.031', 'DB5.069', 'DB6.292', 'DB10.006', 'DB10.026', 'DB1.269', 'DB8.297', 'DB2.019', 'DB3.169', 'DB3.089', 'DB8.012', 'DB10.071', 'DB4.250', 'DB9.142', 'DB1.074', 'DB10.134', 'DB7.021', 'DB5.278', 'DB9.078', 'DB3.208', 'DB9.178', 'DB8.243', 'DB9.293', 'DB5.045', 'DB8.108', 'DB2.211', 'DB9.141', 'DB7.097', 'DB3.182', 'DB2.175', 'DB10.192', 'DB4.063', 'DB5.114', 'DB5.208', 'DB10.074', 'DB7.050', 'DB7.168', 'DB1.190', 'DB9.088', 'DB6.031', 'DB10.152', 'DB7.069', 'DB6.203', 'DB8.242', 'DB5.242', 'DB2.213', 'DB3.244', 'DB6.083', 'DB2.155', 'DB7.109', 'DB10.162', 'DB9.150', 'DB5.112', 'DB9.133', 'DB9.152', 'DB5.126', 'DB10.082', 'DB4.030', 'DB7.100', 'DB8.193', 'DB3.014', 'DB5.291', 'DB3.266', 'DB7.130', 'DB6.253', 'DB3.265', 'DB10.166', 'DB9.160', 'DB9.153', 'DB3.048', 'DB3.065', 'DB7.149', 'DB3.070', 'DB4.160', 'DB5.153', 'DB2.162', 'DB2.032', 'DB7.160', 'DB2.209', 'DB7.140', 'DB8.106', 'DB2.056', 'DB8.261', 'DB2.145', 'DB6.294', 'DB8.287', 'DB2.166', 'DB4.204', 'DB5.089', 'DB9.248', 'DB2.197', 'DB7.062', 'DB9.204', 'DB4.207', 'DB4.104', 'DB8.187', 'DB10.224', 'DB6.108', 'DB7.056', 'DB8.111', 'DB6.209', 'DB4.048', 'DB5.146', 'DB1.144', 'DB3.199', 'DB5.072', 'DB4.111', 'DB3.158', 'DB5.108', 'DB8.158', 'DB6.242', 'DB9.166', 'DB5.186', 'DB10.033', 'DB7.278', 'DB1.261', 'DB9.203', 'DB10.112', 'DB6.050', 'DB3.146', 'DB8.148', 'DB8.292', 'DB10.081', 'DB10.217', 'DB1.068', 'DB9.045', 'DB8.132', 'DB8.040', 'DB10.120', 'DB8.196', 'DB6.235', 'DB6.019', 'DB5.220', 'DB8.234', 'DB6.023', 'DB3.246', 'DB4.038', 'DB4.055', 'DB7.276', 'DB5.202', 'DB10.238', 'DB2.107', 'DB6.297', 'DB6.212', 'DB5.030', 'DB7.024', 'DB2.118', 'DB10.126', 'DB10.175', 'DB2.198', 'DB3.242', 'DB8.098', 'DB9.155', 'DB6.024', 'DB7.217', 'DB2.142', 'DB3.073', 'DB9.274', 'DB8.016', 'DB6.097', 'DB3.071', 'DB7.088', 'DB10.279', 'DB7.230', 'DB4.165', 'DB6.170', 'DB10.155', 'DB5.141', 'DB6.008', 'DB8.279', 'DB6.263', 'DB3.138', 'DB4.086', 'DB4.260', 'DB6.005', 'DB6.093', 'DB2.156', 'DB7.283', 'DB2.133', 'DB8.188', 'DB9.107', 'DB6.133', 'DB10.096', 'DB9.234', 'DB2.167', 'DB4.275', 'DB9.005', 'DB1.274', 'DB3.148', 'DB8.054', 'DB4.093', 'DB7.128', 'DB4.149', 'DB7.059', 'DB8.186', 'DB5.188', 'DB8.032', 'DB3.035', 'DB8.184', 'DB4.072', 'DB1.077', 'DB8.104', 'DB9.185', 'DB10.016', 'DB5.095', 'DB9.082', 'DB9.275', 'DB8.122', 'DB8.257', 'DB10.267', 'DB7.142', 'DB6.139', 'DB9.044', 'DB2.045', 'DB2.079', 'DB6.266', 'DB10.243', 'DB8.293', 'DB8.268', 'DB10.229', 'DB3.066', 'DB7.222', 'DB10.190', 'DB8.212', 'DB3.024', 'DB7.078', 'DB8.191', 'DB8.041', 'DB4.159', 'DB4.270', 'DB4.251', 'DB2.125', 'DB3.124', 'DB7.094', 'DB6.197', 'DB10.230', 'DB9.193', 'DB3.006', 'DB2.048', 'DB6.267', 'DB9.217', 'DB3.099', 'DB9.273', 'DB4.252', 'DB7.214', 'DB9.258', 'DB6.118', 'DB4.273', 'DB2.236', 'DB5.056', 'DB1.197', 'DB1.226', 'DB9.097', 'DB10.176', 'DB5.269', 'DB10.187', 'DB2.013', 'DB3.174', 'DB10.193', 'DB10.163', 'DB10.271', 'DB5.134', 'DB2.179', 'DB9.051', 'DB6.241', 'DB3.204', 'DB8.254', 'DB1.132', 'DB5.280', 'DB8.109', 'DB4.150', 'DB4.024', 'DB9.202', 'DB7.263', 'DB10.136', 'DB10.218', 'DB8.008', 'DB3.072', 'DB7.060', 'DB2.204', 'DB2.073', 'DB8.230', 'DB3.002', 'DB6.006', 'DB4.157', 'DB7.277', 'DB7.270', 'DB10.178', 'DB8.263', 'DB5.212', 'DB3.111', 'DB10.058', 'DB3.023', 'DB8.003', 'DB8.023', 'DB8.176', 'DB2.237', 'DB7.026', 'DB6.033', 'DB7.234', 'DB3.167', 'DB4.129', 'DB3.088', 'DB5.125', 'DB9.252', 'DB2.035', 'DB3.057', 'DB10.255', 'DB2.031', 'DB4.174', 'DB4.158', 'DB10.099', 'DB9.079', 'DB7.020', 'DB8.084', 'DB8.200', 'DB4.034', 'DB10.185', 'DB10.174', 'DB1.066', 'DB6.238', 'DB2.099', 'DB1.280', 'DB3.090', 'DB8.220', 'DB1.296', 'DB5.137', 'DB1.277', 'DB9.286', 'DB6.045', 'DB5.029', 'DB10.116', 'DB2.151', 'DB7.229', 'DB9.047', 'DB3.180', 'DB9.064', 'DB2.063', 'DB8.161', 'DB5.068', 'DB3.151', 'DB7.166', 'DB2.223', 'DB2.114', 'DB7.122', 'DB3.010', 'DB8.086', 'DB8.139', 'DB10.194', 'DB5.020', 'DB1.265', 'DB6.190', 'DB9.072', 'DB5.082', 'DB9.265', 'DB7.203', 'DB7.257', 'DB8.015', 'DB7.191', 'DB6.223', 'DB8.160', 'DB3.068', 'DB5.196', 'DB4.175', 'DB10.153', 'DB8.149', 'DB8.090', 'DB8.265', 'DB5.102', 'DB8.245', 'DB5.250', 'DB6.119', 'DB3.005', 'DB2.123', 'DB9.037', 'DB9.261', 'DB7.105', 'DB10.144', 'DB6.018', 'DB2.157', 'DB10.142', 'DB5.204', 'DB2.086', 'DB3.083', 'DB9.187', 'DB9.029', 'DB9.171', 'DB5.275', 'DB5.044', 'DB8.216', 'DB5.229', 'DB10.041', 'DB3.042', 'DB10.001', 'DB3.184', 'DB6.191', 'DB9.224', 'DB5.164', 'DB7.162', 'DB8.154', 'DB10.034', 'DB3.181', 'DB9.272', 'DB5.215', 'DB5.060', 'DB7.104', 'DB2.188', 'DB5.079', 'DB2.136', 'DB10.244', 'DB1.266', 'DB5.272', 'DB3.003', 'DB10.119', 'DB10.291', 'DB9.197', 'DB10.248', 'DB8.198', 'DB7.064', 'DB3.170', 'DB4.047', 'DB4.066', 'DB6.181', 'DB10.090', 'DB6.214', 'DB4.126', 'DB10.161', 'DB8.124', 'DB3.201', 'DB5.080', 'DB5.149', 'DB2.091', 'DB4.152', 'DB7.256', 'DB5.214', 'DB6.032', 'DB7.169', 'DB5.197', 'DB7.170', 'DB9.132', 'DB5.087', 'DB3.100', 'DB5.185', 'DB10.288', 'DB4.145', 'DB9.101', 'DB4.081', 'DB10.028', 'DB8.145', 'DB7.126', 'DB6.067', 'DB7.120', 'DB1.251', 'DB9.092', 'DB8.089', 'DB6.135', 'DB2.189', 'DB7.210', 'DB7.042', 'DB10.213', 'DB1.070', 'DB4.213', 'DB4.167', 'DB9.024', 'DB7.141', 'DB4.025', 'DB6.187', 'DB8.030', 'DB6.126', 'DB3.144', 'DB7.095', 'DB9.033', 'DB6.113', 'DB6.089', 'DB10.150', 'DB10.186', 'DB7.098', 'DB7.195', 'DB5.246', 'DB9.190', 'DB10.044', 'DB5.086', 'DB7.006', 'DB4.056', 'DB8.274', 'DB5.094', 'DB1.154', 'DB7.071', 'DB7.046', 'DB10.179', 'DB9.021', 'DB5.078', 'DB4.223', 'DB7.047', 'DB9.052', 'DB10.212', 'DB3.234', 'DB3.164', 'DB3.075', 'DB10.203', 'DB7.268', 'DB6.036', 'DB7.018', 'DB2.058', 'DB10.275', 'DB9.294', 'DB10.157', 'DB8.231', 'DB10.184', 'DB5.247', 'DB9.233', 'DB8.019', 'DB4.132', 'DB9.210', 'DB5.256', 'DB9.138', 'DB7.204', 'DB5.286', 'DB7.052', 'DB7.057', 'DB10.088', 'DB4.004', 'DB7.248', 'DB3.120', 'DB7.118', 'DB5.010', 'DB2.100', 'DB7.221', 'DB7.011', 'DB9.031', 'DB2.071', 'DB10.141', 'DB9.145', 'DB6.226', 'DB5.111', 'DB3.032', 'DB2.141', 'DB4.060', 'DB4.244', 'DB1.288', 'DB5.013', 'DB9.246', 'DB6.225', 'DB9.083', 'DB3.047', 'DB10.245', 'DB2.138', 'DB7.113', 'DB5.033', 'DB10.118', 'DB8.119', 'DB10.201', 'DB8.202', 'DB2.171', 'DB4.107', 'DB4.036', 'DB9.270', 'DB6.002', 'DB3.058', 'DB10.039', 'DB10.253', 'DB3.012', 'DB2.190', 'DB3.045', 'DB5.268', 'DB4.221', 'DB10.029', 'DB9.016', 'DB8.068', 'DB8.082', 'DB10.087', 'DB10.247', 'DB10.098', 'DB4.001', 'DB3.211', 'DB8.039', 'DB9.180', 'DB1.284', 'DB9.225', 'DB3.251', 'DB10.272', 'DB5.273', 'DB10.083', 'DB7.041', 'DB4.075', 'DB10.114', 'DB5.097', 'DB9.146', 'DB5.181', 'DB7.218', 'DB8.219', 'DB3.059', 'DB7.034', 'DB3.077', 'DB5.067', 'DB3.019', 'DB7.152', 'DB7.035', 'DB5.157', 'DB2.150', 'DB10.102', 'DB9.115', 'DB10.073', 'DB9.060', 'DB7.176', 'DB4.172', 'DB9.100', 'DB7.017', 'DB3.050', 'DB4.283', 'DB10.092', 'DB5.189', 'DB5.183', 'DB8.042', 'DB10.084', 'DB8.002', 'DB7.164', 'DB9.080', 'DB10.123', 'DB4.142', 'DB9.131', 'DB2.104', 'DB10.036', 'DB6.104', 'DB4.173', 'DB7.091', 'DB5.276', 'DB6.068', 'DB6.164', 'DB6.065', 'DB4.225', 'DB7.112', 'DB5.267', 'DB3.175', 'DB10.208', 'DB10.021', 'DB3.104', 'DB5.081', 'DB3.080', 'DB4.071', 'DB6.039', 'DB9.227', 'DB7.005', 'DB7.043', 'DB10.121', 'DB6.052', 'DB5.058', 'DB7.101', 'DB2.059', 'DB8.142', 'DB8.224', 'DB3.218', 'DB8.075', 'DB3.040', 'DB9.220', 'DB7.134', 'DB5.281', 'DB9.065', 'DB5.167', 'DB2.040', 'DB7.007', 'DB8.027', 'DB10.017', 'DB3.092', 'DB10.042', 'DB9.076', 'DB7.194', 'DB6.114', 'DB6.101', 'DB7.202', 'DB7.107', 'DB10.135', 'DB4.005', 'DB5.115', 'DB7.023', 'DB4.100', 'DB5.026', 'DB3.021', 'DB8.105', 'DB2.094', 'DB5.119', 'DB9.039', 'DB10.093', 'DB7.196', 'DB6.115', 'DB7.172', 'DB9.184', 'DB1.298', 'DB9.112', 'DB4.202', 'DB5.139', 'DB1.285', 'DB8.248', 'DB10.226', 'DB2.042', 'DB6.078', 'DB5.053', 'DB5.070', 'DB1.223', 'DB9.264', 'DB10.108', 'DB10.169', 'DB8.045', 'DB8.204', 'DB7.037', 'DB10.097', 'DB5.266', 'DB9.182', 'DB6.070', 'DB5.252', 'DB8.022', 'DB10.268', 'DB7.004', 'DB8.034', 'DB4.262', 'DB7.237', 'DB1.158', 'DB3.081', 'DB8.077', 'DB7.181', 'DB2.168', 'DB4.143', 'DB4.108', 'DB10.293', 'DB10.214', 'DB7.033', 'DB3.079', 'DB8.284', 'DB8.258', 'DB8.228', 'DB5.292', 'DB5.047', 'DB7.063', 'DB9.161', 'DB7.255', 'DB9.292', 'DB10.111', 'DB8.275', 'DB9.218', 'DB6.172', 'DB3.052', 'DB9.183', 'DB3.195', 'DB8.128', 'DB10.283', 'DB2.000', 'DB2.087', 'DB7.211', 'DB6.182', 'DB5.244', 'DB9.120', 'DB3.198', 'DB7.136', 'DB9.287', 'DB6.046', 'DB6.171', 'DB7.015', 'DB2.144', 'DB2.113', 'DB6.009', 'DB8.076', 'DB9.000', 'DB10.209', 'DB6.227', 'DB4.249', 'DB6.075', 'DB10.189', 'DB7.219', 'DB6.280', 'DB2.055', 'DB8.172', 'DB10.270', 'DB3.207', 'DB7.076', 'DB10.260', 'DB7.119', 'DB7.212', 'DB5.133', 'DB7.132', 'DB5.223', 'DB4.195', 'DB3.213', 'DB5.101', 'DB9.086', 'DB6.244', 'DB2.015', 'DB6.166', 'DB5.179', 'DB8.300', 'DB3.268', 'DB7.002', 'DB5.106', 'DB4.137', 'DB10.165', 'DB10.101', 'DB8.272', 'DB10.154', 'DB8.163', 'DB6.221', 'DB10.103', 'DB8.278', 'DB5.233', 'DB10.259', 'DB3.084', 'DB6.028', 'DB3.027', 'DB8.296', 'DB7.061', 'DB6.027', 'DB5.168', 'DB4.009', 'DB5.224', 'DB2.051', 'DB10.143', 'DB9.170', 'DB3.189', 'DB9.041', 'DB4.044', 'DB9.257', 'DB2.172', 'DB6.245', 'DB3.132', 'DB6.090', 'DB6.125', 'DB9.259', 'DB7.295', 'DB7.179', 'DB4.101', 'DB8.024', 'DB4.003', 'DB2.098', 'DB5.054', 'DB10.115', 'DB6.207', 'DB2.115', 'DB2.037', 'DB9.148', 'DB4.243', 'DB8.037', 'DB1.163', 'DB4.017', 'DB6.099', 'DB5.264', 'DB10.197', 'DB2.135', 'DB3.217', 'DB7.275', 'DB10.258', 'DB8.072', 'DB10.133', 'DB10.080', 'DB1.279', 'DB7.285', 'DB8.097', 'DB9.177', 'DB3.127', 'DB7.040', 'DB10.137', 'DB4.033', 'DB3.049', 'DB5.061', 'DB9.254', 'DB10.129', 'DB4.169', 'DB4.193', 'DB9.289', 'DB4.073', 'DB1.262', 'DB7.199', 'DB3.009', 'DB4.156', 'DB5.110', 'DB9.140', 'DB9.032', 'DB10.290', 'DB1.250', 'DB7.246', 'DB7.186', 'DB9.277', 'DB5.245', 'DB4.300', 'DB9.043', 'DB2.203', 'DB8.129', 'DB6.076', 'DB6.300', 'DB2.121', 'DB9.209', 'DB2.081', 'DB6.079', 'DB4.154', 'DB6.285', 'DB9.143', 'DB8.000', 'DB7.200', 'DB2.117', 'DB9.297', 'DB3.093', 'DB7.148', 'DB7.143', 'DB8.251', 'DB1.281', 'DB5.193', 'DB4.121', 'DB2.088', 'DB5.025', 'DB6.192', 'DB3.055', 'DB4.194', 'DB7.039', 'DB1.290', 'DB6.111', 'DB9.212', 'DB4.196', 'DB7.099', 'DB4.203', 'DB6.069', 'DB8.025', 'DB6.003', 'DB8.169', 'DB6.255', 'DB2.060', 'DB4.161', 'DB5.231', 'DB5.048', 'DB6.239', 'DB6.299', 'DB8.053', 'DB2.154', 'DB10.156', 'DB6.189', 'DB3.130', 'DB7.296', 'DB3.131', 'DB5.099', 'DB3.186', 'DB7.187', 'DB1.242', 'DB10.284', 'DB5.213', 'DB3.118', 'DB7.174', 'DB9.103', 'DB5.120', 'DB10.222', 'DB7.207', 'DB6.251', 'DB10.297', 'DB9.022', 'DB9.034', 'DB10.138', 'DB3.172', 'DB10.072', 'DB6.254', 'DB4.094', 'DB3.037', 'DB10.046', 'DB8.252', 'DB5.092', 'DB10.070', 'DB8.038', 'DB10.047', 'DB7.067', 'DB5.290', 'DB5.279', 'DB10.124', 'DB7.129', 'DB2.047', 'DB3.194', 'DB7.051', 'DB5.176', 'DB5.113', 'DB10.056', 'DB4.261', 'DB7.290', 'DB7.025', 'DB5.192', 'DB6.130', 'DB10.232', 'DB10.223', 'DB9.151', 'DB10.285', 'DB7.036', 'DB3.101', 'DB5.207', 'DB7.111', 'DB5.257', 'DB4.266', 'DB5.282', 'DB7.125', 'DB7.123', 'DB8.290', 'DB8.070', 'DB3.103', 'DB10.064', 'DB4.170', 'DB9.280', 'DB6.295', 'DB3.114', 'DB4.065', 'DB5.152', 'DB2.130', 'DB3.264', 'DB4.214', 'DB2.129', 'DB3.162', 'DB6.167', 'DB6.208', 'DB2.239', 'DB2.039', 'DB6.175', 'DB8.011', 'DB10.040', 'DB8.281', 'DB5.200', 'DB10.147', 'DB4.191', 'DB7.009', 'DB3.152', 'DB3.250', 'DB4.164', 'DB5.129', 'DB7.182', 'DB2.018', 'DB4.079', 'DB8.229', 'DB3.018', 'DB9.130', 'DB6.247', 'DB10.236', 'DB3.106', 'DB8.226', 'DB9.228', 'DB9.175', 'DB8.195', 'DB3.076', 'DB8.049', 'DB2.195', 'DB4.234', 'DB5.175', 'DB9.168', 'DB10.231', 'DB2.233', 'DB9.268', 'DB4.082', 'DB3.190', 'DB10.261', 'DB5.283', 'DB4.109', 'DB10.100', 'DB6.129', 'DB7.226', 'DB6.252', 'DB4.123', 'DB9.211', 'DB5.285', 'DB2.038', 'DB5.055', 'DB9.198', 'DB2.010', 'DB10.177', 'DB8.271', 'DB3.196', 'DB1.283', 'DB8.051', 'DB6.086', 'DB8.285', 'DB1.067', 'DB3.060', 'DB5.049', 'DB9.127', 'DB2.219', 'DB6.041', 'DB6.062', 'DB6.206', 'DB8.199', 'DB8.209', 'DB7.249', 'DB1.267', 'DB9.290', 'DB9.156', 'DB5.090', 'DB8.207', 'DB8.291', 'DB5.243', 'DB9.018', 'DB9.071', 'DB4.232', 'DB5.174', 'DB9.271', 'DB9.221', 'DB4.027', 'DB5.163', 'DB8.021', 'DB5.018', 'DB5.249', 'DB5.205', 'DB5.124', 'DB10.196', 'DB7.264', 'DB8.060', 'DB2.147', 'DB3.133', 'DB7.075', 'DB7.188', 'DB8.208', 'DB4.219', 'DB10.011', 'DB6.229', 'DB6.183', 'DB9.001', 'DB5.147', 'DB4.237', 'DB3.128', 'DB3.119', 'DB8.180', 'DB10.239', 'DB8.074', 'DB7.090', 'DB5.035', 'DB5.262', 'DB5.259', 'DB5.225', 'DB9.121', 'DB5.210', 'DB7.209', 'DB4.133', 'DB5.203', 'DB5.128', 'DB6.168', 'DB4.095', 'DB8.283', 'DB3.108', 'DB4.074', 'DB6.117', 'DB3.147', 'DB4.153', 'DB8.269', 'DB2.110', 'DB5.255', 'DB5.199', 'DB5.107', 'DB2.006', 'DB4.254', 'DB2.017', 'DB9.066', 'DB7.284', 'DB6.134', 'DB4.168', 'DB5.219', 'DB9.067', 'DB9.096', 'DB10.188', 'DB2.158', 'DB8.298', 'DB5.271', 'DB10.273', 'DB5.105', 'DB9.242', 'DB9.226', 'DB8.262', 'DB9.235', 'DB8.190', 'DB2.221', 'DB2.229', 'DB8.010', 'DB6.102', 'DB3.193', 'DB8.215', 'DB8.210', 'DB7.259', 'DB10.262', 'DB4.043', 'DB6.015', 'DB1.142', 'DB9.191', 'DB4.148', 'DB2.242', 'DB4.206', 'DB4.016', 'DB10.013', 'DB7.144', 'DB6.095', 'DB10.274', 'DB5.039', 'DB4.045', 'DB8.253', 'DB4.209', 'DB3.017', 'DB8.225', 'DB5.218', 'DB4.163', 'DB5.121', 'DB5.195', 'DB9.260', 'DB9.090', 'DB7.072', 'DB10.234', 'DB4.083', 'DB7.250', 'DB9.240', 'DB5.104', 'DB3.062', 'DB6.161', 'DB4.058', 'DB3.230', 'DB8.150', 'DB10.210', 'DB10.117', 'DB10.207', 'DB8.056', 'DB5.065', 'DB2.072', 'DB3.038', 'DB10.204', 'DB6.060', 'DB3.044', 'DB5.251', 'DB5.177', 'DB7.269', 'DB7.267', 'DB4.124', 'DB6.049', 'DB1.076', 'DB7.288', 'DB9.111', 'DB4.220', 'DB6.004', 'DB9.027', 'DB9.262', 'DB3.177', 'DB7.022', 'DB6.112', 'DB6.096', 'DB2.169', 'DB10.085', 'DB3.191', 'DB5.017', 'DB3.112', 'DB7.175', 'DB9.057', 'DB7.225', 'DB7.178', 'DB3.020', 'DB6.016', 'DB1.184', 'DB6.234', 'DB7.008', 'DB9.215', 'DB10.233', 'DB9.162', 'DB6.215', 'DB4.255', 'DB5.217', 'DB3.121', 'DB4.192', 'DB5.182', 'DB6.296', 'DB7.161', 'DB9.003', 'DB6.128', 'DB9.053', 'DB9.230', 'DB2.227', 'DB7.293', 'DB8.168', 'DB4.224', 'DB3.229', 'DB8.294', 'DB8.007', 'DB6.072', 'DB6.194', 'DB8.170', 'DB10.235', 'DB7.291', 'DB5.042', 'DB9.205', 'DB2.116', 'DB2.165', 'DB6.012', 'DB7.167', 'DB1.072', 'DB9.062', 'DB9.116', 'DB6.222', 'DB7.206', 'DB10.008', 'DB2.245', 'DB3.254', 'DB4.090', 'DB10.031', 'DB4.239', 'DB6.265', 'DB3.179', 'DB10.252', 'DB9.238', 'DB9.263', 'DB2.160', 'DB8.164', 'DB10.068', 'DB5.159', 'DB6.132', 'DB6.048', 'DB10.164', 'DB2.043', 'DB9.236', 'DB7.227', 'DB7.066', 'DB10.012', 'DB4.054', 'DB3.028', 'DB7.058', 'DB8.127', 'DB8.107', 'DB3.160', 'DB9.282', 'DB4.021', 'DB10.125', 'DB7.159', 'DB5.077', 'DB8.232', 'DB8.152', 'DB8.085', 'DB10.278', 'DB2.232', 'DB10.241', 'DB10.002', 'DB9.007', 'DB10.167', 'DB10.256', 'DB10.264', 'DB5.238', 'DB4.217', 'DB10.148', 'DB9.281', 'DB8.013', 'DB6.040', 'DB4.098', 'DB5.277', 'DB5.248', 'DB6.188', 'DB7.238', 'DB2.062', 'DB6.071', 'DB7.065', 'DB10.172', 'DB3.025', 'DB2.127', 'DB10.181', 'DB9.214', 'DB4.141', 'DB7.289', 'DB10.219', 'DB2.026', 'DB2.053', 'DB2.052', 'DB6.261', 'DB9.093', 'DB8.029', 'DB10.292', 'DB3.240', 'DB9.176', 'DB2.152', 'DB9.085', 'DB5.190', 'DB8.123', 'DB6.199', 'DB6.204', 'DB8.091', 'DB6.116', 'DB6.288', 'DB7.235', 'DB5.091', 'DB6.298', 'DB7.262', 'DB3.225', 'DB2.215', 'DB7.265', 'DB8.182', 'DB9.054', 'DB1.075', 'DB5.109', 'DB3.269', 'DB9.061', 'DB9.008', 'DB3.155', 'DB5.274', 'DB2.046', 'DB3.015', 'DB4.018', 'DB10.199', 'DB2.241', 'DB6.080', 'DB6.248', 'DB2.093', 'DB10.168', 'DB7.281', 'DB7.240', 'DB8.079', 'DB7.183', 'DB7.154', 'DB9.134', 'DB3.220', 'DB3.113', 'DB4.087', 'DB5.118', 'DB4.059', 'DB9.279', 'DB8.052', 'DB7.135', 'DB4.257', 'DB8.093', 'DB5.117', 'DB8.221', 'DB4.200', 'DB10.109', 'DB6.178', 'DB6.020', 'DB2.024', 'DB3.247', 'DB6.088', 'DB6.053', 'DB1.286', 'DB4.274', 'DB9.011', 'DB6.195', 'DB10.160', 'DB8.147', 'DB10.106', 'DB3.210', 'DB5.019', 'DB8.136', 'DB6.283', 'DB10.216', 'DB9.136', 'DB7.114', 'DB6.281', 'DB6.138', 'DB6.259', 'DB2.007', 'DB5.085', 'DB9.128', 'DB2.132', 'DB2.146', 'DB7.146', 'DB5.288', 'DB2.069', 'DB5.011', 'DB2.090', 'DB5.142', 'DB2.108', 'DB2.070', 'DB8.048', 'DB9.269', 'DB6.055', 'DB8.100', 'DB10.075', 'DB9.020', 'DB10.149', 'DB8.092', 'DB8.009', 'DB8.175', 'DB6.176', 'DB9.255', 'DB10.079', 'DB10.173', 'DB2.106', 'DB4.134', 'DB5.083', 'DB7.260', 'DB9.159', 'DB7.013', 'DB5.209', 'DB5.156', 'DB2.021', 'DB3.105', 'DB9.050', 'DB8.101', 'DB5.093', 'DB8.135', 'DB9.026', 'DB2.124', 'DB9.223', 'DB3.156', 'DB6.262', 'DB5.194', 'DB4.258', 'DB10.128', 'DB9.298', 'DB3.039', 'DB10.249', 'DB10.122', 'DB4.042', 'DB7.073', 'DB9.129', 'DB9.035', 'DB6.200', 'DB8.153', 'DB9.232', 'DB7.180', 'DB8.131', 'DB6.010', 'DB3.109', 'DB6.091', 'DB7.282', 'DB8.214', 'DB4.269', 'DB9.105', 'DB8.110', 'DB5.270', 'DB7.197', 'DB9.049', 'DB7.215', 'DB3.231', 'DB5.062', 'DB8.156', 'DB4.046', 'DB7.292', 'DB3.137', 'DB4.128', 'DB4.089', 'DB8.114', 'DB10.022', 'DB3.051', 'DB10.240', 'DB5.050', 'DB2.131', 'DB7.019', 'DB2.226', 'DB2.176', 'DB2.103', 'DB5.043', 'DB2.011', 'DB10.225', 'DB10.009', 'DB8.288', 'DB10.063', 'DB9.002', 'DB10.127', 'DB2.016', 'DB3.212', 'DB4.238', 'DB6.233', 'DB4.085', 'DB6.163', 'DB8.028', 'DB8.203', 'DB10.198', 'DB4.091', 'DB7.077', 'DB7.103', 'DB4.227', 'DB2.194', 'DB4.022', 'DB7.258', 'DB7.044', 'DB3.183', 'DB9.012', 'DB4.146', 'DB3.219', 'DB6.106', 'DB3.011', 'DB7.272', 'DB9.245', 'DB5.036', 'DB10.158', 'DB5.155', 'DB3.056', 'DB4.246', 'DB5.100', 'DB9.186', 'DB3.036', 'DB7.261', 'DB2.050', 'DB10.032', 'DB7.157', 'DB10.195', 'DB4.023', 'DB4.125', 'DB6.205', 'DB2.181', 'DB1.069', 'DB7.198', 'DB1.272', 'DB7.190', 'DB7.298', 'DB7.045', 'DB7.003', 'DB4.298', 'DB5.211', 'DB8.246', 'DB10.145', 'DB5.021', 'DB6.162', 'DB10.065', 'DB4.245', 'DB8.286', 'DB5.123', 'DB8.017', 'DB2.207', 'DB3.150', 'DB5.235', 'DB10.257', 'DB5.187', 'DB7.287', 'DB10.049', 'DB8.078', 'DB3.074', 'DB3.145', 'DB3.069', 'DB7.131', 'DB3.125', 'DB5.145', 'DB5.169', 'DB7.117', 'DB6.165', 'DB7.080', 'DB5.161', 'DB7.121', 'DB5.057', 'DB8.273', 'DB2.140', 'DB8.126', 'DB2.119', 'DB5.002', 'DB8.239', 'DB5.171', 'DB2.084', 'DB8.102', 'DB8.116', 'DB1.297', 'DB8.178', 'DB10.005', 'DB7.084', 'DB2.111', 'DB10.110', 'DB2.005', 'DB8.026', 'DB6.216', 'DB3.187', 'DB9.149', 'DB1.268', 'DB6.174', 'DB3.030', 'DB5.263', 'DB10.246', 'DB8.146', 'DB5.135', 'DB8.080', 'DB5.027', 'DB10.050', 'DB3.139', 'DB6.087', 'DB7.110', 'DB10.007', 'DB6.057', 'DB8.065', 'DB10.140', 'DB5.127', 'DB9.063', 'DB2.193', 'DB5.074', 'DB9.299', 'DB9.122', 'DB7.038', 'DB2.008', 'DB7.054', 'DB4.099', 'DB8.099', 'DB6.196', 'DB8.295', 'DB6.034', 'DB4.026', 'DB3.185', 'DB9.055', 'DB6.243', 'DB10.146', 'DB3.161', 'DB7.251', 'DB4.069', 'DB4.155', 'DB7.053', 'DB3.046', 'DB3.176', 'DB8.005', 'DB2.064', 'DB10.043', 'DB4.084', 'DB2.085', 'DB7.223', 'DB8.112', 'DB7.049', 'DB8.249', 'DB6.257', 'DB10.076', 'DB8.201', 'DB4.228', 'DB2.126', 'DB9.173', 'DB10.030', 'DB3.087', 'DB8.267', 'DB8.218', 'DB10.171', 'DB9.104', 'DB4.078', 'DB4.041', 'DB8.213', 'DB4.268', 'DB6.211', 'DB5.071', 'DB7.116', 'DB4.057', 'DB2.185', 'DB9.284', 'DB7.208', 'DB6.173', 'DB8.240', 'DB8.233', 'DB10.228', 'DB7.232', 'DB10.202', 'DB5.136', 'DB6.177', 'DB3.178', 'DB5.103', 'DB7.106', 'DB8.173', 'DB9.123', 'DB6.051', 'DB8.171', 'DB3.239', 'DB8.137', 'DB6.105', 'DB8.266', 'DB4.235', 'DB10.023', 'DB9.009', 'DB5.144', 'DB5.284', 'DB7.139', 'DB4.281', 'DB5.073', 'DB5.148', 'DB9.030', 'DB3.041', 'DB4.208', 'DB3.082', 'DB5.240', 'DB9.267', 'DB10.010', 'DB5.040', 'DB6.109', 'DB4.097', 'DB7.093', 'DB7.213', 'DB8.103', 'DB2.234', 'DB9.081', 'DB5.180', 'DB10.004', 'DB3.223', 'DB7.055', 'DB9.189', 'DB7.245', 'DB4.215', 'DB10.254', 'DB7.192', 'DB9.296', 'DB9.174', 'DB9.040', 'DB6.202', 'DB4.259', 'DB10.027', 'DB10.227', 'DB9.250', 'DB8.035', 'DB9.164', 'DB8.260', 'DB5.232', 'DB7.244', 'DB6.043', 'DB9.157', 'DB10.251', 'DB6.056', 'DB6.228', 'DB8.115', 'DB8.181', 'DB4.135', 'DB2.184', 'DB9.266', 'DB7.205', 'DB10.061', 'DB5.237', 'DB1.141', 'DB5.216', 'DB6.120', 'DB5.230', 'DB6.030', 'DB8.159', 'DB3.034', 'DB3.200', 'DB10.018', 'DB6.290', 'DB7.089', 'DB4.077', 'DB2.218', 'DB5.005', 'DB9.075', 'DB2.238', 'DB2.002', 'DB7.294', 'DB5.287', 'DB3.134', 'DB10.045', 'DB10.281', 'DB10.113', 'DB10.095', 'DB7.083', 'DB4.240', 'DB6.082', 'DB2.128', 'DB8.289', 'DB6.231', 'DB6.035', 'DB9.028', 'DB10.130', 'DB7.147', 'DB4.122', 'DB5.122', 'DB1.071', 'DB6.185', 'DB9.206', 'DB1.073', 'DB8.036', 'DB4.039', 'DB5.052'}\n",
      "2021-04-05 18:09:37 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 18:09:37 INFO After filtering, 2373 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 2373 samples, 13732 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/49 (52)\n",
      "['../studies/49/all.biom']\n",
      "2021-04-05 18:10:47 INFO loaded 273 samples, 8112 features\n",
      "2021-04-05 18:10:47 WARNING These have metadata but do not have data - dropped (28): {'ERR1545766', 'ERR1545737', 'ERR1545712', 'ERR1074598', 'ERR1545980', 'ERR1074963', 'ERR1545898', 'ERR1091143', 'ERR1841753', 'ERR1546004', 'ERR1841517', 'ERR2032154', 'ERR1080264', 'ERR1545971', 'ERR1090164', 'ERR1545713', 'ERR1091007', 'ERR1090991', 'ERR2238662', 'ERR1546037', 'ERR1545736', 'ERR1091581', 'ERR1091017', 'ERR1090999', 'ERR1090029', 'ERR1545777', 'ERR1841808', 'ERR1075006'}\n",
      "2021-04-05 18:10:47 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 18:10:47 INFO After filtering, 263 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 263 samples, 8112 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/40 (53)\n",
      "['../studies/40/all.biom']\n",
      "2021-04-05 18:11:53 INFO loaded 86 samples, 3917 features\n",
      "2021-04-05 18:11:53 WARNING These have metadata but do not have data - dropped (7): {'ERR1073218', 'ERR2580024', 'ERR1091444', 'ERR1091053', 'ERR1841756', 'ERR1090007', 'ERR1075021'}\n",
      "2021-04-05 18:11:53 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 18:11:53 INFO After filtering, 85 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 85 samples, 3917 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/47 (54)\n",
      "['../studies/47/all.biom']\n",
      "2021-04-05 18:12:56 INFO loaded 257 samples, 7437 features\n",
      "2021-04-05 18:12:56 WARNING These have metadata but do not have data - dropped (17): {'ERR1074882', 'ERR1545836', 'ERR1545948', 'ERR1545812', 'ERR1841629', 'ERR1091158', 'ERR1073703', 'ERR1458779', 'ERR1160308', 'ERR1841809', 'ERR1545953', 'ERR1073252', 'ERR1091085', 'ERR1091044', 'ERR1841777', 'ERR1841802', 'ERR1546002'}\n",
      "2021-04-05 18:12:56 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 18:12:56 INFO After filtering, 247 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 247 samples, 7437 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/2 (55)\n",
      "['../studies/2/all.biom']\n",
      "2021-04-05 18:14:04 INFO loaded 179 samples, 2528 features\n",
      "2021-04-05 18:14:04 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 18:14:04 INFO After filtering, 179 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 179 samples, 2528 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/13 (56)\n",
      "['../studies/13/all.biom']\n",
      "2021-04-05 18:15:08 INFO loaded 31 samples, 925 features\n",
      "2021-04-05 18:15:08 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 18:15:08 INFO After filtering, 31 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 31 samples, 925 features\n",
      "not enough controls (17). skipping\n",
      "not enough disease (14). skipping\n",
      "keeping 31 total samples\n",
      "**********\n",
      "processing ../studies/5 (57)\n",
      "['../studies/5/all.biom']\n",
      "2021-04-05 18:16:10 INFO loaded 333 samples, 4664 features\n",
      "2021-04-05 18:16:10 WARNING These have metadata but do not have data - dropped (507): {'SRR10911493', 'SRR10911928', 'SRR10911696', 'SRR10911171', 'SRR10911808', 'SRR10911335', 'SRR10911445', 'SRR10911244', 'SRR10911330', 'SRR10911418', 'SRR10911953', 'SRR10911447', 'SRR10911900', 'SRR10911889', 'SRR10911349', 'SRR10911239', 'SRR10911870', 'SRR10911222', 'SRR10911963', 'SRR10911891', 'SRR10911734', 'SRR10911887', 'SRR10911365', 'SRR10911745', 'SRR10911880', 'SRR10911196', 'SRR10911710', 'SRR10911862', 'SRR10911368', 'SRR10911958', 'SRR10911163', 'SRR10911726', 'SRR10911829', 'SRR10911911', 'SRR10911816', 'SRR10911852', 'SRR10911243', 'SRR10911820', 'SRR10911678', 'SRR10911159', 'SRR10911814', 'SRR10911181', 'SRR10911697', 'SRR10911861', 'SRR10911962', 'SRR10911850', 'SRR10911276', 'SRR10911853', 'SRR10911338', 'SRR10911192', 'SRR10911446', 'SRR10911707', 'SRR10911310', 'SRR10911947', 'SRR10911441', 'SRR10911823', 'SRR10911448', 'SRR10911687', 'SRR10911226', 'SRR10911322', 'SRR10911705', 'SRR10911832', 'SRR10911860', 'SRR10911193', 'SRR10911875', 'SRR10911763', 'SRR10911921', 'SRR10911149', 'SRR10911806', 'SRR10911692', 'SRR10911712', 'SRR10911956', 'SRR10911798', 'SRR10911346', 'SRR10911835', 'SRR10911162', 'SRR10911156', 'SRR10911158', 'SRR10911700', 'SRR10911709', 'SRR10911505', 'SRR10911782', 'SRR10911896', 'SRR10911848', 'SRR10911164', 'SRR10911800', 'SRR10911905', 'SRR10911680', 'SRR10911248', 'SRR10911918', 'SRR10911320', 'SRR10911948', 'SRR10911144', 'SRR10911242', 'SRR10911298', 'SRR10911755', 'SRR10911922', 'SRR10911735', 'SRR10911357', 'SRR10911919', 'SRR10911356', 'SRR10911373', 'SRR10911685', 'SRR10911197', 'SRR10911555', 'SRR10911359', 'SRR10911247', 'SRR10911926', 'SRR10911874', 'SRR10911528', 'SRR10911878', 'SRR10911175', 'SRR10911160', 'SRR10911440', 'SRR10911730', 'SRR10911788', 'SRR10911792', 'SRR10911336', 'SRR10911888', 'SRR10911229', 'SRR10911831', 'SRR10911733', 'SRR10911610', 'SRR10911199', 'SRR10911287', 'SRR10911797', 'SRR10911885', 'SRR10911177', 'SRR10911779', 'SRR10911824', 'SRR10911769', 'SRR10911382', 'SRR10911371', 'SRR10911224', 'SRR10911701', 'SRR10911718', 'SRR10911315', 'SRR10911343', 'SRR10911327', 'SRR10911698', 'SRR10911954', 'SRR10911153', 'SRR10911183', 'SRR10911746', 'SRR10911172', 'SRR10911369', 'SRR10911145', 'SRR10911378', 'SRR10911810', 'SRR10911451', 'SRR10911713', 'SRR10911754', 'SRR10911693', 'SRR10911377', 'SRR10911790', 'SRR10911347', 'SRR10911326', 'SRR10911703', 'SRR10911457', 'SRR10911711', 'SRR10911787', 'SRR10911185', 'SRR10911951', 'SRR10911873', 'SRR10911187', 'SRR10911688', 'SRR10911304', 'SRR10911339', 'SRR10911238', 'SRR10911150', 'SRR10911844', 'SRR10911864', 'SRR10911912', 'SRR10911348', 'SRR10911902', 'SRR10911311', 'SRR10911950', 'SRR10911813', 'SRR10911727', 'SRR10911210', 'SRR10911753', 'SRR10911331', 'SRR10911191', 'SRR10911764', 'SRR10911748', 'SRR10911883', 'SRR10911795', 'SRR10911722', 'SRR10911716', 'SRR10911694', 'SRR10911316', 'SRR10911186', 'SRR10911328', 'SRR10911817', 'SRR10911704', 'SRR10911245', 'SRR10911842', 'SRR10911367', 'SRR10911925', 'SRR10911857', 'SRR10911720', 'SRR10911897', 'SRR10911442', 'SRR10911775', 'SRR10911194', 'SRR10911855', 'SRR10911146', 'SRR10911964', 'SRR10911309', 'SRR10911344', 'SRR10911908', 'SRR10911354', 'SRR10911453', 'SRR10911578', 'SRR10911742', 'SRR10911773', 'SRR10911838', 'SRR10911325', 'SRR10911751', 'SRR10911223', 'SRR10911961', 'SRR10911246', 'SRR10911783', 'SRR10911913', 'SRR10911772', 'SRR10911924', 'SRR10911628', 'SRR10911907', 'SRR10911165', 'SRR10911719', 'SRR10911379', 'SRR10911872', 'SRR10911740', 'SRR10911750', 'SRR10911854', 'SRR10911909', 'SRR10911943', 'SRR10911372', 'SRR10911825', 'SRR10911271', 'SRR10911351', 'SRR10911195', 'SRR10911681', 'SRR10911154', 'SRR10911353', 'SRR10911937', 'SRR10911731', 'SRR10911736', 'SRR10911826', 'SRR10911933', 'SRR10911270', 'SRR10911566', 'SRR10911847', 'SRR10911780', 'SRR10911312', 'SRR10911168', 'SRR10911952', 'SRR10911767', 'SRR10911173', 'SRR10911725', 'SRR10911946', 'SRR10911895', 'SRR10911863', 'SRR10911932', 'SRR10911840', 'SRR10911757', 'SRR10911358', 'SRR10911714', 'SRR10911939', 'SRR10911821', 'SRR10911375', 'SRR10911957', 'SRR10911805', 'SRR10911308', 'SRR10911686', 'SRR10911809', 'SRR10911599', 'SRR10911148', 'SRR10911869', 'SRR10911184', 'SRR10911777', 'SRR10911364', 'SRR10911230', 'SRR10911879', 'SRR10911721', 'SRR10911917', 'SRR10911960', 'SRR10911152', 'SRR10911241', 'SRR10911319', 'SRR10911147', 'SRR10911233', 'SRR10911884', 'SRR10911898', 'SRR10911724', 'SRR10911455', 'SRR10911589', 'SRR10911801', 'SRR10911449', 'SRR10911760', 'SRR10911965', 'SRR10911765', 'SRR10911929', 'SRR10911942', 'SRR10911577', 'SRR10911143', 'SRR10911352', 'SRR10911770', 'SRR10911915', 'SRR10911708', 'SRR10911739', 'SRR10911340', 'SRR10911232', 'SRR10911324', 'SRR10911901', 'SRR10911881', 'SRR10911231', 'SRR10911717', 'SRR10911332', 'SRR10911389', 'SRR10911737', 'SRR10911945', 'SRR10911738', 'SRR10911967', 'SRR10911802', 'SRR10911400', 'SRR10911867', 'SRR10911450', 'SRR10911454', 'SRR10911314', 'SRR10911227', 'SRR10911968', 'SRR10911342', 'SRR10911151', 'SRR10911456', 'SRR10911689', 'SRR10911851', 'SRR10911894', 'SRR10911899', 'SRR10911240', 'SRR10911856', 'SRR10911944', 'SRR10911836', 'SRR10911812', 'SRR10911176', 'SRR10911381', 'SRR10911360', 'SRR10911333', 'SRR10911236', 'SRR10911906', 'SRR10911752', 'SRR10911892', 'SRR10911161', 'SRR10911762', 'SRR10911935', 'SRR10911741', 'SRR10911776', 'SRR10911904', 'SRR10911789', 'SRR10911683', 'SRR10911793', 'SRR10911667', 'SRR10911732', 'SRR10911914', 'SRR10911656', 'SRR10911866', 'SRR10911188', 'SRR10911786', 'SRR10911539', 'SRR10911756', 'SRR10911182', 'SRR10911843', 'SRR10911916', 'SRR10911682', 'SRR10911715', 'SRR10911744', 'SRR10911699', 'SRR10911749', 'SRR10911791', 'SRR10911934', 'SRR10911362', 'SRR10911329', 'SRR10911376', 'SRR10911804', 'SRR10911318', 'SRR10911178', 'SRR10911702', 'SRR10911868', 'SRR10911370', 'SRR10911691', 'SRR10911466', 'SRR10911729', 'SRR10911621', 'SRR10911940', 'SRR10911822', 'SRR10911859', 'SRR10911174', 'SRR10911827', 'SRR10911305', 'SRR10911794', 'SRR10911761', 'SRR10911955', 'SRR10911876', 'SRR10911941', 'SRR10911361', 'SRR10911157', 'SRR10911771', 'SRR10911849', 'SRR10911679', 'SRR10911366', 'SRR10911949', 'SRR10911341', 'SRR10911818', 'SRR10911306', 'SRR10911796', 'SRR10911759', 'SRR10911768', 'SRR10911834', 'SRR10911684', 'SRR10911452', 'SRR10911350', 'SRR10911927', 'SRR10911355', 'SRR10911690', 'SRR10911846', 'SRR10911190', 'SRR10911516', 'SRR10911234', 'SRR10911334', 'SRR10911170', 'SRR10911766', 'SRR10911429', 'SRR10911966', 'SRR10911920', 'SRR10911363', 'SRR10911841', 'SRR10911228', 'SRR10911189', 'SRR10911815', 'SRR10911337', 'SRR10911167', 'SRR10911313', 'SRR10911865', 'SRR10911443', 'SRR10911133', 'SRR10911166', 'SRR10911858', 'SRR10911845', 'SRR10911910', 'SRR10911936', 'SRR10911237', 'SRR10911728', 'SRR10911155', 'SRR10911169', 'SRR10911540', 'SRR10911706', 'SRR10911758', 'SRR10911828', 'SRR10911959', 'SRR10911321', 'SRR10911833', 'SRR10911903', 'SRR10911871', 'SRR10911890', 'SRR10911471', 'SRR10911785', 'SRR10911886', 'SRR10911345', 'SRR10911317', 'SRR10911747', 'SRR10911839', 'SRR10911225', 'SRR10911307', 'SRR10911743', 'SRR10911923', 'SRR10911931', 'SRR10911374', 'SRR10911380', 'SRR10911781', 'SRR10911179', 'SRR10911235', 'SRR10911811', 'SRR10911180', 'SRR10911411', 'SRR10911639', 'SRR10911383', 'SRR10911444', 'SRR10911221', 'SRR10911938', 'SRR10911723', 'SRR10911819', 'SRR10911837', 'SRR10911893', 'SRR10911930', 'SRR10911799', 'SRR10911323', 'SRR10911695', 'SRR10911803', 'SRR10911882', 'SRR10911259', 'SRR10911784', 'SRR10911807', 'SRR10911774', 'SRR10911482', 'SRR10911830', 'SRR10911877', 'SRR10911778'}\n",
      "2021-04-05 18:16:10 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 18:16:10 INFO After filtering, 333 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 333 samples, 4664 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/14 (58)\n",
      "['../studies/14/all.biom']\n",
      "2021-04-05 18:17:25 INFO loaded 123 samples, 1271 features\n",
      "2021-04-05 18:17:25 WARNING These have metadata but do not have data - dropped (57): {'SRR6804656', 'SRR6307733', 'SRR6804663', 'SRR6804657', 'SRR6307712', 'SRR6307779', 'SRR6307775', 'SRR6307789', 'SRR6307795', 'SRR6307780', 'SRR6804660', 'SRR6307756', 'SRR6307788', 'SRR6307794', 'SRR6307777', 'SRR6307759', 'SRR6307781', 'SRR6307790', 'SRR6804661', 'SRR6307774', 'SRR6804665', 'SRR6307796', 'SRR6307792', 'SRR6307734', 'SRR6307714', 'SRR6804653', 'SRR6804654', 'SRR6804655', 'SRR6307716', 'SRR6307797', 'SRR6307721', 'SRR6307778', 'SRR6307791', 'SRR6804666', 'SRR6307755', 'SRR6307776', 'SRR6804652', 'SRR6307750', 'SRR6307757', 'SRR6307717', 'SRR6307718', 'SRR6804664', 'SRR6804658', 'SRR6307713', 'SRR6307758', 'SRR6307719', 'SRR6307773', 'SRR6804659', 'SRR6307715', 'SRR6307760', 'SRR6307793', 'SRR6307753', 'SRR6804662', 'SRR6307720', 'SRR6307782', 'SRR6307754', 'SRR6307751'}\n",
      "2021-04-05 18:17:25 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 18:17:25 INFO After filtering, 123 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 123 samples, 1271 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/22 (59)\n",
      "['../studies/22/all.biom']\n",
      "2021-04-05 18:18:34 INFO loaded 144 samples, 2687 features\n",
      "2021-04-05 18:18:34 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 18:18:34 INFO After filtering, 144 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 144 samples, 2687 features\n",
      "keeping 46 total samples\n",
      "**********\n",
      "processing ../studies/25 (61)\n",
      "['../studies/25/all.biom']\n",
      "2021-04-05 18:19:48 INFO loaded 92 samples, 1781 features\n",
      "2021-04-05 18:19:48 WARNING These have metadata but do not have data - dropped (1): {'ERR1776941'}\n",
      "2021-04-05 18:19:48 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 18:19:48 INFO After filtering, 89 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 89 samples, 1781 features\n",
      "not enough disease (20). skipping\n",
      "keeping 43 total samples\n",
      "finished. Total processed 62. Total ok 59. Not enough samples 0\n"
     ]
    }
   ],
   "source": [
    "exp=join_all_exps(random_seed=2020)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AmpliconExperiment (\"merged\") with 2533 samples, 44537 features"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exp.description='merged'\n",
    "exp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>type</th>\n",
       "      <th>disease_cohort</th>\n",
       "      <th>age</th>\n",
       "      <th>_sample_id</th>\n",
       "      <th>_calour_original_abundance</th>\n",
       "      <th>exp</th>\n",
       "      <th>../studies/59</th>\n",
       "      <th>host_subject</th>\n",
       "      <th>title</th>\n",
       "      <th>host_sex</th>\n",
       "      <th>...</th>\n",
       "      <th>../studies/13</th>\n",
       "      <th>Isolation_source</th>\n",
       "      <th>sequencing_lab</th>\n",
       "      <th>../studies/5</th>\n",
       "      <th>../studies/14</th>\n",
       "      <th>../studies/22</th>\n",
       "      <th>Sample</th>\n",
       "      <th>rep</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>../studies/25</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ERR1385491</th>\n",
       "      <td>disease</td>\n",
       "      <td>T2D</td>\n",
       "      <td>adult</td>\n",
       "      <td>ERR1385491</td>\n",
       "      <td>68395.0</td>\n",
       "      <td>../studies/61</td>\n",
       "      <td>exp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>exp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exp</td>\n",
       "      <td>exp</td>\n",
       "      <td>exp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERR1385206</th>\n",
       "      <td>HC</td>\n",
       "      <td>T2D</td>\n",
       "      <td>adult</td>\n",
       "      <td>ERR1385206</td>\n",
       "      <td>95408.0</td>\n",
       "      <td>../studies/61</td>\n",
       "      <td>exp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>exp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exp</td>\n",
       "      <td>exp</td>\n",
       "      <td>exp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERR1383735</th>\n",
       "      <td>disease</td>\n",
       "      <td>T2D</td>\n",
       "      <td>adult</td>\n",
       "      <td>ERR1383735</td>\n",
       "      <td>60137.0</td>\n",
       "      <td>../studies/61</td>\n",
       "      <td>exp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>exp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exp</td>\n",
       "      <td>exp</td>\n",
       "      <td>exp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERR1384980</th>\n",
       "      <td>disease</td>\n",
       "      <td>T2D</td>\n",
       "      <td>adult</td>\n",
       "      <td>ERR1384980</td>\n",
       "      <td>74487.0</td>\n",
       "      <td>../studies/61</td>\n",
       "      <td>exp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>exp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exp</td>\n",
       "      <td>exp</td>\n",
       "      <td>exp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERR1382783</th>\n",
       "      <td>HC</td>\n",
       "      <td>T2D</td>\n",
       "      <td>adult</td>\n",
       "      <td>ERR1382783</td>\n",
       "      <td>48282.0</td>\n",
       "      <td>../studies/61</td>\n",
       "      <td>exp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>exp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exp</td>\n",
       "      <td>exp</td>\n",
       "      <td>exp</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>exp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERR1776911</th>\n",
       "      <td>disease</td>\n",
       "      <td>NaN</td>\n",
       "      <td>child</td>\n",
       "      <td>ERR1776911</td>\n",
       "      <td>4943.0</td>\n",
       "      <td>../studies/25</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S74</td>\n",
       "      <td>a</td>\n",
       "      <td>Crohn_at _start_of_EEN</td>\n",
       "      <td>other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERR1776929</th>\n",
       "      <td>disease</td>\n",
       "      <td>NaN</td>\n",
       "      <td>child</td>\n",
       "      <td>ERR1776929</td>\n",
       "      <td>27155.0</td>\n",
       "      <td>../studies/25</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S80</td>\n",
       "      <td>c</td>\n",
       "      <td>Crohn_at _start_of_EEN</td>\n",
       "      <td>other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERR1776938</th>\n",
       "      <td>HC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>adult</td>\n",
       "      <td>ERR1776938</td>\n",
       "      <td>4227.0</td>\n",
       "      <td>../studies/25</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S84</td>\n",
       "      <td>a</td>\n",
       "      <td>Relative_mother</td>\n",
       "      <td>other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERR1776947</th>\n",
       "      <td>disease</td>\n",
       "      <td>NaN</td>\n",
       "      <td>child</td>\n",
       "      <td>ERR1776947</td>\n",
       "      <td>10203.0</td>\n",
       "      <td>../studies/25</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S88</td>\n",
       "      <td>a</td>\n",
       "      <td>Crohn_at _start_of_EEN</td>\n",
       "      <td>other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERR1776954</th>\n",
       "      <td>disease</td>\n",
       "      <td>NaN</td>\n",
       "      <td>child</td>\n",
       "      <td>ERR1776954</td>\n",
       "      <td>32712.0</td>\n",
       "      <td>../studies/25</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>S90</td>\n",
       "      <td>b</td>\n",
       "      <td>Crohn_at _start_of_EEN</td>\n",
       "      <td>other</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2533 rows  344 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               type disease_cohort    age  _sample_id  \\\n",
       "ERR1385491  disease            T2D  adult  ERR1385491   \n",
       "ERR1385206       HC            T2D  adult  ERR1385206   \n",
       "ERR1383735  disease            T2D  adult  ERR1383735   \n",
       "ERR1384980  disease            T2D  adult  ERR1384980   \n",
       "ERR1382783       HC            T2D  adult  ERR1382783   \n",
       "...             ...            ...    ...         ...   \n",
       "ERR1776911  disease            NaN  child  ERR1776911   \n",
       "ERR1776929  disease            NaN  child  ERR1776929   \n",
       "ERR1776938       HC            NaN  adult  ERR1776938   \n",
       "ERR1776947  disease            NaN  child  ERR1776947   \n",
       "ERR1776954  disease            NaN  child  ERR1776954   \n",
       "\n",
       "            _calour_original_abundance            exp ../studies/59  \\\n",
       "ERR1385491                     68395.0  ../studies/61           exp   \n",
       "ERR1385206                     95408.0  ../studies/61           exp   \n",
       "ERR1383735                     60137.0  ../studies/61           exp   \n",
       "ERR1384980                     74487.0  ../studies/61           exp   \n",
       "ERR1382783                     48282.0  ../studies/61           exp   \n",
       "...                                ...            ...           ...   \n",
       "ERR1776911                      4943.0  ../studies/25           NaN   \n",
       "ERR1776929                     27155.0  ../studies/25           NaN   \n",
       "ERR1776938                      4227.0  ../studies/25           NaN   \n",
       "ERR1776947                     10203.0  ../studies/25           NaN   \n",
       "ERR1776954                     32712.0  ../studies/25           NaN   \n",
       "\n",
       "           host_subject title host_sex  ... ../studies/13 Isolation_source  \\\n",
       "ERR1385491          NaN   NaN      NaN  ...           exp              NaN   \n",
       "ERR1385206          NaN   NaN      NaN  ...           exp              NaN   \n",
       "ERR1383735          NaN   NaN      NaN  ...           exp              NaN   \n",
       "ERR1384980          NaN   NaN      NaN  ...           exp              NaN   \n",
       "ERR1382783          NaN   NaN      NaN  ...           exp              NaN   \n",
       "...                 ...   ...      ...  ...           ...              ...   \n",
       "ERR1776911          NaN   NaN      NaN  ...           NaN              NaN   \n",
       "ERR1776929          NaN   NaN      NaN  ...           NaN              NaN   \n",
       "ERR1776938          NaN   NaN      NaN  ...           NaN              NaN   \n",
       "ERR1776947          NaN   NaN      NaN  ...           NaN              NaN   \n",
       "ERR1776954          NaN   NaN      NaN  ...           NaN              NaN   \n",
       "\n",
       "           sequencing_lab ../studies/5 ../studies/14  ../studies/22  Sample  \\\n",
       "ERR1385491            NaN          exp           exp            exp     NaN   \n",
       "ERR1385206            NaN          exp           exp            exp     NaN   \n",
       "ERR1383735            NaN          exp           exp            exp     NaN   \n",
       "ERR1384980            NaN          exp           exp            exp     NaN   \n",
       "ERR1382783            NaN          exp           exp            exp     NaN   \n",
       "...                   ...          ...           ...            ...     ...   \n",
       "ERR1776911            NaN          NaN           NaN            NaN     S74   \n",
       "ERR1776929            NaN          NaN           NaN            NaN     S80   \n",
       "ERR1776938            NaN          NaN           NaN            NaN     S84   \n",
       "ERR1776947            NaN          NaN           NaN            NaN     S88   \n",
       "ERR1776954            NaN          NaN           NaN            NaN     S90   \n",
       "\n",
       "            rep             sample_type ../studies/25  \n",
       "ERR1385491  NaN                     NaN           exp  \n",
       "ERR1385206  NaN                     NaN           exp  \n",
       "ERR1383735  NaN                     NaN           exp  \n",
       "ERR1384980  NaN                     NaN           exp  \n",
       "ERR1382783  NaN                     NaN           exp  \n",
       "...         ...                     ...           ...  \n",
       "ERR1776911    a  Crohn_at _start_of_EEN         other  \n",
       "ERR1776929    c  Crohn_at _start_of_EEN         other  \n",
       "ERR1776938    a         Relative_mother         other  \n",
       "ERR1776947    a  Crohn_at _start_of_EEN         other  \n",
       "ERR1776954    b  Crohn_at _start_of_EEN         other  \n",
       "\n",
       "[2533 rows x 344 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exp.sample_metadata"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Save the resulting biom table and mapping file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2021-04-05 18:21:04 INFO Metadata field taxonomy not found. Saving biom table without metadata\n"
     ]
    }
   ],
   "source": [
    "exp.save('../allsamples/allsamples')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "exp.save_fasta('../allsamples/allsamples.fa')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AmpliconExperiment (\"merged\") with 2533 samples, 44537 features"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exp"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Also save after filtering for blooiming bacteria"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "exp=exp.filter_by_fasta('../allsamples/blooming-bacteria.fa',negate=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2021-04-05 18:21:10 INFO Metadata field taxonomy not found. Saving biom table without metadata\n"
     ]
    }
   ],
   "source": [
    "exp.save('../allsamples/allsamples-no-bloom')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "exp.save_fasta('../allsamples/allsamples-no-bloom.fa')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AmpliconExperiment (\"merged\") with 2533 samples, 44526 features"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exp"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Now create the single biom table for all samples without subsampling"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "**********\n",
      "processing ../studies/61 (0)\n",
      "['../studies/61/all.biom']\n",
      "2021-04-05 19:56:44 INFO loaded 41 samples, 2715 features\n",
      "2021-04-05 19:56:44 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:56:44 INFO After filtering, 41 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 41 samples, 2715 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 41 samples, 2715 features\n",
      "**********\n",
      "processing ../studies/59 (1)\n",
      "['../studies/59/all.biom']\n",
      "2021-04-05 19:56:44 INFO loaded 33 samples, 2637 features\n",
      "2021-04-05 19:56:44 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:56:44 INFO After filtering, 33 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 33 samples, 2637 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 33 samples, 2637 features\n",
      "**********\n",
      "processing ../studies/50 (2)\n",
      "['../studies/50/all.biom']\n",
      "2021-04-05 19:56:44 INFO loaded 58 samples, 959 features\n",
      "2021-04-05 19:56:44 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:56:44 INFO After filtering, 58 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 58 samples, 959 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 58 samples, 959 features\n",
      "**********\n",
      "processing ../studies/57 (3)\n",
      "['../studies/57/all.biom']\n",
      "2021-04-05 19:56:44 INFO loaded 85 samples, 4058 features\n",
      "2021-04-05 19:56:44 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:56:44 INFO After filtering, 85 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 85 samples, 4058 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 85 samples, 4058 features\n",
      "**********\n",
      "processing ../studies/32 (4)\n",
      "['../studies/32/all.biom']\n",
      "2021-04-05 19:56:45 INFO loaded 43 samples, 1748 features\n",
      "2021-04-05 19:56:45 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:56:45 INFO After filtering, 43 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 43 samples, 1748 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 43 samples, 1748 features\n",
      "**********\n",
      "processing ../studies/56 (5)\n",
      "['../studies/56/all.biom']\n",
      "2021-04-05 19:56:46 INFO loaded 43 samples, 3045 features\n",
      "2021-04-05 19:56:46 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:56:46 INFO After filtering, 43 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 43 samples, 3045 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 43 samples, 3045 features\n",
      "**********\n",
      "processing ../studies/51 (6)\n",
      "['../studies/51/all.biom']\n",
      "2021-04-05 19:56:47 INFO loaded 164 samples, 3240 features\n",
      "2021-04-05 19:56:47 WARNING These have metadata but do not have data - dropped (2): {'SRR10430029', 'SRR10429951'}\n",
      "2021-04-05 19:56:47 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:56:47 INFO After filtering, 164 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 164 samples, 3240 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 164 samples, 3240 features\n",
      "**********\n",
      "processing ../studies/58 (8)\n",
      "['../studies/58/all.biom']\n",
      "2021-04-05 19:56:48 INFO loaded 45 samples, 3005 features\n",
      "2021-04-05 19:56:48 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:56:48 INFO After filtering, 45 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 45 samples, 3005 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 45 samples, 3005 features\n",
      "**********\n",
      "processing ../studies/60 (9)\n",
      "['../studies/60/all.biom']\n",
      "2021-04-05 19:56:49 INFO loaded 73 samples, 3771 features\n",
      "2021-04-05 19:56:49 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:56:49 INFO After filtering, 73 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 73 samples, 3771 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 73 samples, 3771 features\n",
      "**********\n",
      "processing ../studies/34 (10)\n",
      "['../studies/34/all.biom']\n",
      "2021-04-05 19:56:50 INFO loaded 32 samples, 1234 features\n",
      "2021-04-05 19:56:50 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:56:50 INFO After filtering, 32 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 32 samples, 1234 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 32 samples, 1234 features\n",
      "**********\n",
      "processing ../studies/33 (11)\n",
      "['../studies/33/all.biom']\n",
      "2021-04-05 19:56:52 INFO loaded 58 samples, 1250 features\n",
      "2021-04-05 19:56:52 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:56:52 INFO After filtering, 58 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 58 samples, 1250 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 58 samples, 1250 features\n",
      "**********\n",
      "processing ../studies/20 (12)\n",
      "['../studies/20/all.biom']\n",
      "2021-04-05 19:56:53 INFO loaded 174 samples, 1622 features\n",
      "2021-04-05 19:56:53 WARNING These have metadata but do not have data - dropped (1): {'ERR2837004'}\n",
      "2021-04-05 19:56:53 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:56:53 INFO After filtering, 151 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 151 samples, 1622 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 151 samples, 1622 features\n",
      "**********\n",
      "processing ../studies/18 (13)\n",
      "['../studies/18/all.biom']\n",
      "2021-04-05 19:56:55 INFO loaded 84 samples, 1183 features\n",
      "2021-04-05 19:56:55 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:56:55 INFO After filtering, 84 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 84 samples, 1183 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 84 samples, 1183 features\n",
      "**********\n",
      "processing ../studies/27 (14)\n",
      "['../studies/27/all.biom']\n",
      "2021-04-05 19:56:57 INFO loaded 251 samples, 1403 features\n",
      "2021-04-05 19:56:57 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:56:57 INFO After filtering, 233 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 233 samples, 1403 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 233 samples, 1403 features\n",
      "**********\n",
      "processing ../studies/9 (15)\n",
      "['../studies/9/all.biom']\n",
      "2021-04-05 19:56:59 INFO loaded 498 samples, 3722 features\n",
      "2021-04-05 19:56:59 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:56:59 INFO After filtering, 497 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 497 samples, 3722 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 451 samples, 3722 features\n",
      "**********\n",
      "processing ../studies/11 (16)\n",
      "['../studies/11/all.biom']\n",
      "2021-04-05 19:57:01 INFO loaded 109 samples, 2312 features\n",
      "2021-04-05 19:57:01 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:57:01 INFO After filtering, 109 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 109 samples, 2312 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 109 samples, 2312 features\n",
      "**********\n",
      "processing ../studies/7 (17)\n",
      "['../studies/7/all.biom']\n",
      "2021-04-05 19:57:04 INFO loaded 441 samples, 4403 features\n",
      "2021-04-05 19:57:04 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:57:04 INFO After filtering, 441 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 441 samples, 4403 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 441 samples, 4403 features\n",
      "**********\n",
      "processing ../studies/29 (18)\n",
      "['../studies/29/all.biom']\n",
      "2021-04-05 19:57:08 INFO loaded 595 samples, 3470 features\n",
      "2021-04-05 19:57:08 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:57:08 INFO After filtering, 594 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 594 samples, 3470 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 594 samples, 3470 features\n",
      "**********\n",
      "processing ../studies/16 (19)\n",
      "['../studies/16/all.biom']\n",
      "2021-04-05 19:57:12 INFO loaded 120 samples, 2116 features\n",
      "2021-04-05 19:57:12 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:57:12 INFO After filtering, 119 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 119 samples, 2116 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 119 samples, 2116 features\n",
      "**********\n",
      "processing ../studies/42 (20)\n",
      "['../studies/42/all.biom']\n",
      "2021-04-05 19:57:17 INFO loaded 99 samples, 3829 features\n",
      "2021-04-05 19:57:17 WARNING These have metadata but do not have data - dropped (6): {'ERR1073898', 'ERR1315870', 'ERR1841805', 'ERR1417474', 'ERR2696902', 'ERR1545879'}\n",
      "2021-04-05 19:57:17 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:57:17 INFO After filtering, 96 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 96 samples, 3829 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 96 samples, 3829 features\n",
      "**********\n",
      "processing ../studies/45 (21)\n",
      "['../studies/45/all.biom']\n",
      "2021-04-05 19:57:24 INFO loaded 1095 samples, 14981 features\n",
      "2021-04-05 19:57:24 WARNING These have metadata but do not have data - dropped (66): {'ERR1091564', 'ERR1545831', 'ERR1545996', 'ERR1545778', 'ERR1841761', 'ERR1389884', 'ERR2238631', 'ERR2239656', 'ERR1072642', 'ERR1841758', 'ERR1841817', 'ERR1075171', 'ERR1545722', 'ERR1545752', 'ERR1090995', 'ERR2318028', 'ERR1091321', 'ERR2303979', 'ERR1545981', 'ERR1545967', 'ERR1841716', 'ERR1073742', 'ERR1545883', 'ERR1160652', 'ERR1091606', 'ERR1841810', 'ERR1316257', 'ERR1545905', 'ERR2239754', 'ERR1546017', 'ERR1160500', 'ERR1545998', 'ERR1091145', 'ERR1841815', 'ERR1545894', 'ERR2239633', 'ERR1841814', 'ERR1841492', 'ERR1389937', 'ERR2238647', 'ERR1091364', 'ERR1090798', 'ERR1545957', 'ERR1841804', 'ERR1545906', 'ERR2696527', 'ERR1091654', 'ERR1091156', 'ERR1546023', 'ERR1545847', 'ERR2239699', 'ERR1074008', 'ERR1160588', 'ERR1089803', 'ERR1074155', 'ERR1545935', 'ERR1841512', 'ERR1075099', 'ERR1545921', 'ERR1459053', 'ERR1073096', 'ERR1091590', 'ERR1545915', 'ERR1073328', 'ERR1545994', 'ERR1160532'}\n",
      "2021-04-05 19:57:25 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:57:25 INFO After filtering, 1043 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 1043 samples, 14981 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 1043 samples, 14981 features\n",
      "**********\n",
      "processing ../studies/6 (22)\n",
      "['../studies/6/all.biom']\n",
      "2021-04-05 19:57:44 INFO loaded 620 samples, 4175 features\n",
      "2021-04-05 19:57:44 WARNING These have metadata but do not have data - dropped (4): {'ERR3519122', 'ERR3520757', 'ERR3520666', 'ERR3520927'}\n",
      "2021-04-05 19:57:44 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:57:44 INFO After filtering, 612 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 612 samples, 4175 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 612 samples, 4175 features\n",
      "**********\n",
      "processing ../studies/28 (23)\n",
      "['../studies/28/all.biom']\n",
      "2021-04-05 19:58:01 INFO loaded 142 samples, 1306 features\n",
      "2021-04-05 19:58:01 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:58:01 INFO After filtering, 135 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 135 samples, 1306 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 135 samples, 1306 features\n",
      "**********\n",
      "processing ../studies/17 (24)\n",
      "['../studies/17/all.biom']\n",
      "2021-04-05 19:58:17 INFO loaded 50 samples, 2307 features\n",
      "2021-04-05 19:58:17 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:58:17 INFO After filtering, 50 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 50 samples, 2307 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 50 samples, 2307 features\n",
      "**********\n",
      "processing ../studies/1 (25)\n",
      "['../studies/1/all.biom']\n",
      "2021-04-05 19:58:34 INFO loaded 38 samples, 1082 features\n",
      "2021-04-05 19:58:34 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:58:34 INFO After filtering, 38 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 38 samples, 1082 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 38 samples, 1082 features\n",
      "**********\n",
      "processing ../studies/10 (26)\n",
      "['../studies/10/all.biom']\n",
      "2021-04-05 19:58:50 INFO loaded 174 samples, 3295 features\n",
      "2021-04-05 19:58:50 WARNING These have metadata but do not have data - dropped (2): {'SRR1689253', 'SRR1687461'}\n",
      "2021-04-05 19:58:50 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:58:50 INFO After filtering, 174 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 174 samples, 3295 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 174 samples, 3295 features\n",
      "**********\n",
      "processing ../studies/19 (27)\n",
      "['../studies/19/all.biom']\n",
      "2021-04-05 19:59:08 INFO loaded 68 samples, 1401 features\n",
      "2021-04-05 19:59:08 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:59:08 INFO After filtering, 68 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 68 samples, 1401 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 68 samples, 1401 features\n",
      "**********\n",
      "processing ../studies/26 (28)\n",
      "['../studies/26/all.biom']\n",
      "2021-04-05 19:59:25 INFO loaded 1388 samples, 11834 features\n",
      "2021-04-05 19:59:25 WARNING These have metadata but do not have data - dropped (41): {'DB3.234', 'DB3.229', 'DB3.223', 'DB3.218', 'DB3.196', 'DB3.228', 'DB2.239', 'DB3.230', 'DB3.269', 'DB3.110', 'DB3.072', 'DB3.036', 'DB3.251', 'DB3.047', 'DB3.071', 'DB3.239', 'DB3.250', 'DB3.102', 'DB3.108', 'DB2.241', 'DB2.135', 'DB3.101', 'DB3.217', 'DB3.109', 'DB3.212', 'DB2.245', 'DB3.078', 'DB3.096', 'DB3.106', 'DB2.242', 'DB2.240', 'DB3.107', 'DB3.103', 'DB3.104', 'DB2.076', 'DB3.231', 'DB3.266', 'DB2.069', 'DB3.265', 'DB3.219', 'DB3.264'}\n",
      "2021-04-05 19:59:25 WARNING These have data but do not have metadata: {'DB7.163', 'DB8.224', 'DB8.075', 'DB4.253', 'DB6.279', 'DB6.145', 'DB7.134', 'DB4.062', 'DB5.281', 'DB8.235', 'DB4.233', 'DB8.071', 'DB8.206', 'DB4.053', 'DB5.167', 'DB8.143', 'DB4.230', 'DB8.027', 'DB7.007', 'DB6.282', 'DB5.289', 'DB7.194', 'DB7.158', 'DB7.202', 'DB7.107', 'DB6.114', 'DB8.174', 'DB4.049', 'DB5.028', 'DB6.101', 'DB8.004', 'DB4.005', 'DB5.115', 'DB7.023', 'DB4.100', 'DB6.246', 'DB5.026', 'DB8.105', 'DB7.079', 'DB4.067', 'DB5.119', 'DB8.001', 'DB5.143', 'DB6.218', 'DB7.196', 'DB7.172', 'DB6.213', 'DB6.115', 'DB8.155', 'DB4.050', 'DB4.202', 'DB6.193', 'DB5.139', 'DB8.248', 'DB8.247', 'DB5.227', 'DB6.078', 'DB8.018', 'DB5.053', 'DB5.070', 'DB7.074', 'DB7.253', 'DB8.045', 'DB8.204', 'DB7.037', 'DB6.230', 'DB5.266', 'DB6.070', 'DB5.252', 'DB6.063', 'DB8.022', 'DB6.236', 'DB6.237', 'DB7.004', 'DB8.280', 'DB7.133', 'DB8.059', 'DB5.032', 'DB4.052', 'DB8.034', 'DB4.262', 'DB7.237', 'DB5.016', 'DB8.077', 'DB7.181', 'DB4.143', 'DB4.108', 'DB4.037', 'DB4.064', 'DB7.033', 'DB8.284', 'DB8.258', 'DB7.173', 'DB8.120', 'DB8.228', 'DB5.292', 'DB8.177', 'DB6.293', 'DB5.047', 'DB8.083', 'DB4.028', 'DB7.063', 'DB4.061', 'DB7.255', 'DB7.247', 'DB4.247', 'DB4.211', 'DB4.222', 'DB4.068', 'DB8.275', 'DB6.172', 'DB6.284', 'DB8.128', 'DB6.054', 'DB8.255', 'DB7.211', 'DB6.182', 'DB5.244', 'DB5.031', 'DB7.136', 'DB6.046', 'DB6.171', 'DB7.081', 'DB7.015', 'DB8.076', 'DB6.009', 'DB6.289', 'DB4.031', 'DB4.131', 'DB6.227', 'DB6.100', 'DB4.110', 'DB7.286', 'DB4.249', 'DB6.075', 'DB4.180', 'DB4.103', 'DB4.013', 'DB6.094', 'DB5.116', 'DB8.064', 'DB5.098', 'DB7.219', 'DB6.280', 'DB8.172', 'DB7.076', 'DB6.260', 'DB4.118', 'DB7.048', 'DB7.252', 'DB7.119', 'DB5.096', 'DB7.212', 'DB5.133', 'DB7.132', 'DB7.201', 'DB5.223', 'DB4.195', 'DB7.014', 'DB5.101', 'DB6.244', 'DB8.300', 'DB6.166', 'DB5.179', 'DB7.002', 'DB5.106', 'DB6.011', 'DB5.158', 'DB8.141', 'DB4.140', 'DB4.137', 'DB8.272', 'DB8.163', 'DB6.098', 'DB6.219', 'DB6.221', 'DB6.110', 'DB8.278', 'DB5.233', 'DB4.162', 'DB8.237', 'DB6.092', 'DB7.010', 'DB6.028', 'DB7.016', 'DB6.240', 'DB8.296', 'DB7.061', 'DB6.027', 'DB5.168', 'DB6.131', 'DB4.009', 'DB5.224', 'DB8.117', 'DB7.151', 'DB7.254', 'DB4.076', 'DB8.256', 'DB4.044', 'DB8.165', 'DB5.160', 'DB7.155', 'DB6.245', 'DB6.090', 'DB6.125', 'DB5.241', 'DB6.064', 'DB7.295', 'DB7.179', 'DB6.169', 'DB4.101', 'DB4.040', 'DB8.024', 'DB5.046', 'DB5.054', 'DB4.003', 'DB6.207', 'DB4.216', 'DB4.243', 'DB8.037', 'DB6.217', 'DB7.266', 'DB6.099', 'DB4.017', 'DB5.264', 'DB8.020', 'DB7.275', 'DB6.038', 'DB6.264', 'DB4.138', 'DB4.289', 'DB7.115', 'DB7.280', 'DB4.105', 'DB7.285', 'DB4.176', 'DB8.097', 'DB7.299', 'DB8.057', 'DB5.221', 'DB7.040', 'DB4.033', 'DB5.254', 'DB5.061', 'DB5.066', 'DB4.169', 'DB4.193', 'DB6.179', 'DB4.035', 'DB8.133', 'DB4.073', 'DB7.199', 'DB6.127', 'DB8.125', 'DB6.147', 'DB4.236', 'DB8.094', 'DB8.282', 'DB4.156', 'DB5.110', 'DB8.087', 'DB7.246', 'DB7.186', 'DB4.088', 'DB8.185', 'DB5.245', 'DB8.151', 'DB4.300', 'DB6.047', 'DB6.037', 'DB4.020', 'DB8.129', 'DB5.041', 'DB5.034', 'DB5.084', 'DB6.160', 'DB6.076', 'DB6.300', 'DB7.274', 'DB6.042', 'DB6.272', 'DB8.259', 'DB5.024', 'DB6.007', 'DB4.297', 'DB6.079', 'DB4.154', 'DB6.285', 'DB5.131', 'DB6.291', 'DB5.023', 'DB8.000', 'DB7.200', 'DB6.001', 'DB7.153', 'DB5.000', 'DB7.145', 'DB6.058', 'DB7.148', 'DB7.143', 'DB8.251', 'DB8.058', 'DB5.193', 'DB4.292', 'DB8.140', 'DB4.121', 'DB7.185', 'DB6.192', 'DB5.025', 'DB7.102', 'DB7.228', 'DB4.194', 'DB7.039', 'DB7.171', 'DB6.111', 'DB4.196', 'DB7.099', 'DB4.203', 'DB8.025', 'DB6.069', 'DB6.137', 'DB8.169', 'DB6.003', 'DB5.009', 'DB6.255', 'DB8.157', 'DB7.216', 'DB8.241', 'DB5.064', 'DB4.161', 'DB5.231', 'DB5.048', 'DB6.107', 'DB6.239', 'DB7.027', 'DB6.299', 'DB7.236', 'DB7.224', 'DB8.144', 'DB6.189', 'DB7.296', 'DB5.099', 'DB7.165', 'DB4.151', 'DB7.187', 'DB6.186', 'DB4.096', 'DB5.213', 'DB7.174', 'DB8.095', 'DB5.120', 'DB5.184', 'DB5.038', 'DB7.150', 'DB7.207', 'DB6.251', 'DB4.144', 'DB6.254', 'DB7.220', 'DB6.258', 'DB5.239', 'DB4.136', 'DB4.070', 'DB4.094', 'DB4.120', 'DB7.231', 'DB6.224', 'DB8.252', 'DB5.092', 'DB8.038', 'DB8.270', 'DB7.067', 'DB7.279', 'DB5.290', 'DB5.279', 'DB7.129', 'DB8.227', 'DB7.051', 'DB5.176', 'DB5.113', 'DB6.140', 'DB5.037', 'DB7.290', 'DB7.025', 'DB4.261', 'DB6.130', 'DB5.192', 'DB7.036', 'DB5.207', 'DB7.111', 'DB5.257', 'DB4.266', 'DB7.125', 'DB5.282', 'DB7.123', 'DB8.290', 'DB6.022', 'DB8.070', 'DB4.256', 'DB4.210', 'DB4.170', 'DB8.061', 'DB6.295', 'DB4.065', 'DB5.152', 'DB4.291', 'DB6.232', 'DB4.212', 'DB4.214', 'DB4.226', 'DB6.122', 'DB6.084', 'DB6.167', 'DB4.229', 'DB6.208', 'DB8.055', 'DB8.244', 'DB7.087', 'DB8.011', 'DB6.175', 'DB6.250', 'DB8.281', 'DB5.200', 'DB8.118', 'DB8.179', 'DB4.191', 'DB7.009', 'DB4.164', 'DB5.253', 'DB5.129', 'DB7.182', 'DB7.092', 'DB7.189', 'DB7.068', 'DB4.079', 'DB6.198', 'DB8.250', 'DB8.229', 'DB5.191', 'DB6.247', 'DB8.226', 'DB8.195', 'DB8.031', 'DB8.049', 'DB4.234', 'DB7.070', 'DB6.201', 'DB5.175', 'DB4.082', 'DB8.014', 'DB5.283', 'DB4.109', 'DB6.129', 'DB7.226', 'DB6.252', 'DB4.123', 'DB5.285', 'DB5.055', 'DB8.192', 'DB5.228', 'DB8.271', 'DB5.261', 'DB4.051', 'DB5.201', 'DB8.217', 'DB4.171', 'DB6.074', 'DB4.205', 'DB6.086', 'DB5.178', 'DB4.147', 'DB8.285', 'DB7.096', 'DB5.049', 'DB5.236', 'DB6.041', 'DB5.260', 'DB6.062', 'DB6.206', 'DB7.241', 'DB8.199', 'DB8.046', 'DB7.082', 'DB4.299', 'DB8.081', 'DB4.198', 'DB8.209', 'DB7.249', 'DB4.127', 'DB7.193', 'DB5.090', 'DB6.273', 'DB8.207', 'DB4.112', 'DB8.291', 'DB5.243', 'DB4.232', 'DB5.174', 'DB4.011', 'DB4.241', 'DB4.027', 'DB8.006', 'DB5.163', 'DB8.021', 'DB5.249', 'DB4.290', 'DB8.205', 'DB5.001', 'DB8.236', 'DB5.205', 'DB7.156', 'DB5.124', 'DB4.201', 'DB4.032', 'DB7.264', 'DB8.113', 'DB6.220', 'DB5.138', 'DB5.198', 'DB7.075', 'DB7.188', 'DB4.115', 'DB6.025', 'DB8.208', 'DB4.219', 'DB6.229', 'DB6.183', 'DB5.076', 'DB5.147', 'DB8.043', 'DB4.263', 'DB4.237', 'DB4.190', 'DB6.017', 'DB8.121', 'DB8.180', 'DB8.074', 'DB7.177', 'DB7.090', 'DB7.124', 'DB5.035', 'DB5.262', 'DB5.259', 'DB5.225', 'DB6.013', 'DB5.150', 'DB5.210', 'DB7.209', 'DB5.075', 'DB5.258', 'DB6.268', 'DB5.222', 'DB6.168', 'DB8.238', 'DB5.203', 'DB5.128', 'DB4.133', 'DB6.014', 'DB4.095', 'DB8.283', 'DB8.138', 'DB6.117', 'DB4.074', 'DB4.153', 'DB8.269', 'DB5.255', 'DB7.138', 'DB5.063', 'DB5.199', 'DB5.107', 'DB5.059', 'DB4.254', 'DB7.284', 'DB6.134', 'DB4.168', 'DB5.219', 'DB4.248', 'DB8.183', 'DB7.271', 'DB5.226', 'DB4.242', 'DB8.298', 'DB6.073', 'DB7.273', 'DB7.242', 'DB5.271', 'DB5.105', 'DB8.044', 'DB8.299', 'DB6.249', 'DB5.132', 'DB7.127', 'DB5.165', 'DB8.262', 'DB6.184', 'DB8.190', 'DB8.010', 'DB6.102', 'DB6.000', 'DB8.215', 'DB8.210', 'DB6.103', 'DB7.259', 'DB4.043', 'DB6.015', 'DB4.148', 'DB8.194', 'DB4.206', 'DB4.130', 'DB4.016', 'DB7.137', 'DB7.001', 'DB5.130', 'DB5.154', 'DB7.144', 'DB6.095', 'DB5.051', 'DB7.233', 'DB8.211', 'DB5.206', 'DB8.134', 'DB7.184', 'DB5.039', 'DB4.284', 'DB4.045', 'DB7.239', 'DB8.253', 'DB4.209', 'DB8.225', 'DB5.218', 'DB5.121', 'DB4.163', 'DB6.077', 'DB5.195', 'DB5.151', 'DB7.072', 'DB7.243', 'DB4.272', 'DB7.250', 'DB8.088', 'DB4.083', 'DB4.113', 'DB5.104', 'DB6.161', 'DB4.058', 'DB5.265', 'DB7.108', 'DB6.044', 'DB8.150', 'DB6.286', 'DB6.066', 'DB5.069', 'DB6.292', 'DB8.297', 'DB5.065', 'DB8.012', 'DB4.250', 'DB6.060', 'DB7.021', 'DB5.278', 'DB5.251', 'DB5.177', 'DB7.269', 'DB8.243', 'DB7.267', 'DB4.124', 'DB6.049', 'DB7.288', 'DB5.045', 'DB6.004', 'DB8.108', 'DB4.220', 'DB7.022', 'DB6.112', 'DB6.096', 'DB7.097', 'DB4.063', 'DB7.175', 'DB5.114', 'DB5.208', 'DB7.225', 'DB7.178', 'DB7.050', 'DB7.168', 'DB6.031', 'DB6.016', 'DB7.008', 'DB7.069', 'DB6.234', 'DB6.203', 'DB5.007', 'DB8.242', 'DB5.242', 'DB6.083', 'DB8.069', 'DB7.109', 'DB5.112', 'DB6.215', 'DB5.126', 'DB6.296', 'DB5.217', 'DB5.182', 'DB4.255', 'DB4.192', 'DB7.161', 'DB4.030', 'DB6.128', 'DB8.193', 'DB7.100', 'DB5.291', 'DB7.130', 'DB6.253', 'DB7.293', 'DB8.168', 'DB4.224', 'DB7.149', 'DB8.294', 'DB5.153', 'DB4.160', 'DB8.007', 'DB6.072', 'DB6.194', 'DB8.170', 'DB7.160', 'DB7.140', 'DB8.106', 'DB7.291', 'DB5.042', 'DB8.261', 'DB6.121', 'DB6.294', 'DB6.012', 'DB7.167', 'DB8.287', 'DB4.204', 'DB7.206', 'DB5.089', 'DB6.222', 'DB4.090', 'DB4.239', 'DB7.062', 'DB6.265', 'DB8.164', 'DB4.207', 'DB8.187', 'DB4.104', 'DB6.132', 'DB5.159', 'DB7.056', 'DB6.108', 'DB6.048', 'DB8.111', 'DB6.209', 'DB5.146', 'DB4.048', 'DB7.227', 'DB5.072', 'DB7.066', 'DB4.111', 'DB8.158', 'DB5.108', 'DB4.054', 'DB6.242', 'DB5.186', 'DB8.127', 'DB7.278', 'DB7.058', 'DB8.107', 'DB4.021', 'DB6.050', 'DB7.159', 'DB8.148', 'DB8.292', 'DB8.232', 'DB5.077', 'DB8.152', 'DB8.085', 'DB8.132', 'DB8.040', 'DB5.238', 'DB4.217', 'DB8.196', 'DB6.235', 'DB6.019', 'DB8.013', 'DB5.220', 'DB8.234', 'DB6.023', 'DB6.040', 'DB4.038', 'DB4.055', 'DB4.098', 'DB7.276', 'DB5.277', 'DB5.202', 'DB5.248', 'DB6.297', 'DB6.212', 'DB6.188', 'DB5.030', 'DB7.024', 'DB7.238', 'DB7.030', 'DB6.071', 'DB7.065', 'DB8.098', 'DB4.141', 'DB7.289', 'DB7.217', 'DB6.024', 'DB6.261', 'DB8.016', 'DB6.097', 'DB7.088', 'DB8.029', 'DB7.230', 'DB8.066', 'DB4.165', 'DB6.170', 'DB5.141', 'DB6.008', 'DB8.279', 'DB6.274', 'DB6.263', 'DB5.190', 'DB4.086', 'DB4.260', 'DB6.005', 'DB6.093', 'DB8.123', 'DB6.199', 'DB8.091', 'DB6.204', 'DB7.283', 'DB6.116', 'DB6.288', 'DB5.022', 'DB7.235', 'DB5.091', 'DB8.188', 'DB6.133', 'DB7.262', 'DB4.275', 'DB6.298', 'DB7.265', 'DB8.182', 'DB7.128', 'DB4.093', 'DB4.149', 'DB7.059', 'DB5.109', 'DB8.186', 'DB8.032', 'DB5.188', 'DB5.274', 'DB8.184', 'DB4.072', 'DB8.104', 'DB4.018', 'DB5.095', 'DB6.080', 'DB6.248', 'DB5.004', 'DB8.122', 'DB8.257', 'DB7.142', 'DB6.139', 'DB7.281', 'DB6.266', 'DB8.079', 'DB7.240', 'DB8.293', 'DB8.268', 'DB7.222', 'DB7.183', 'DB8.212', 'DB7.078', 'DB8.191', 'DB7.154', 'DB8.041', 'DB4.087', 'DB4.159', 'DB5.298', 'DB4.270', 'DB5.118', 'DB4.059', 'DB4.251', 'DB8.052', 'DB7.135', 'DB8.093', 'DB7.094', 'DB4.257', 'DB8.221', 'DB5.117', 'DB6.178', 'DB6.020', 'DB6.197', 'DB4.287', 'DB6.088', 'DB6.053', 'DB4.274', 'DB6.195', 'DB6.267', 'DB5.014', 'DB8.147', 'DB4.252', 'DB7.031', 'DB7.214', 'DB8.136', 'DB6.118', 'DB6.283', 'DB4.273', 'DB7.114', 'DB6.281', 'DB5.056', 'DB6.138', 'DB6.259', 'DB5.269', 'DB5.134', 'DB5.085', 'DB6.241', 'DB7.146', 'DB8.254', 'DB5.288', 'DB5.011', 'DB5.142', 'DB8.048', 'DB5.280', 'DB8.109', 'DB6.055', 'DB4.150', 'DB4.024', 'DB8.100', 'DB7.263', 'DB8.008', 'DB7.060', 'DB8.230', 'DB8.092', 'DB8.009', 'DB8.175', 'DB6.176', 'DB6.006', 'DB8.063', 'DB7.277', 'DB7.270', 'DB4.157', 'DB8.263', 'DB5.212', 'DB5.083', 'DB4.134', 'DB7.260', 'DB7.013', 'DB5.209', 'DB5.156', 'DB8.003', 'DB8.023', 'DB8.176', 'DB8.101', 'DB7.026', 'DB7.234', 'DB6.033', 'DB4.129', 'DB8.135', 'DB5.093', 'DB5.125', 'DB6.262', 'DB4.174', 'DB4.158', 'DB5.194', 'DB6.136', 'DB8.200', 'DB8.084', 'DB7.020', 'DB4.034', 'DB6.238', 'DB7.073', 'DB4.042', 'DB6.200', 'DB8.220', 'DB5.137', 'DB8.153', 'DB7.180', 'DB8.131', 'DB6.010', 'DB6.045', 'DB8.214', 'DB5.029', 'DB7.282', 'DB6.091', 'DB4.269', 'DB7.229', 'DB8.110', 'DB7.197', 'DB5.270', 'DB7.215', 'DB5.062', 'DB8.156', 'DB4.218', 'DB8.161', 'DB4.046', 'DB5.068', 'DB7.292', 'DB7.166', 'DB5.015', 'DB7.122', 'DB4.128', 'DB4.089', 'DB8.086', 'DB8.139', 'DB8.114', 'DB6.190', 'DB5.082', 'DB5.050', 'DB7.203', 'DB7.257', 'DB7.019', 'DB8.015', 'DB7.191', 'DB6.223', 'DB8.160', 'DB5.043', 'DB5.196', 'DB4.175', 'DB8.288', 'DB4.285', 'DB6.233', 'DB8.149', 'DB8.090', 'DB4.238', 'DB8.265', 'DB6.163', 'DB8.028', 'DB8.203', 'DB8.245', 'DB5.102', 'DB4.085', 'DB5.250', 'DB6.119', 'DB4.091', 'DB7.103', 'DB4.227', 'DB7.105', 'DB4.022', 'DB6.018', 'DB7.258', 'DB7.044', 'DB5.204', 'DB4.146', 'DB6.106', 'DB5.275', 'DB7.272', 'DB5.044', 'DB8.216', 'DB5.229', 'DB5.036', 'DB5.155', 'DB5.100', 'DB6.191', 'DB7.162', 'DB5.164', 'DB8.154', 'DB7.261', 'DB7.157', 'DB5.215', 'DB4.023', 'DB4.125', 'DB7.104', 'DB5.060', 'DB5.079', 'DB7.298', 'DB6.205', 'DB7.198', 'DB7.190', 'DB7.045', 'DB7.003', 'DB4.298', 'DB5.211', 'DB8.246', 'DB5.272', 'DB6.162', 'DB4.245', 'DB8.286', 'DB5.123', 'DB8.017', 'DB8.198', 'DB7.064', 'DB5.235', 'DB4.047', 'DB6.181', 'DB4.066', 'DB6.214', 'DB5.187', 'DB4.126', 'DB7.287', 'DB8.124', 'DB8.078', 'DB5.080', 'DB5.149', 'DB7.131', 'DB5.145', 'DB5.169', 'DB7.117', 'DB7.080', 'DB6.165', 'DB4.152', 'DB5.161', 'DB7.121', 'DB7.256', 'DB5.057', 'DB8.273', 'DB8.126', 'DB5.214', 'DB8.239', 'DB7.169', 'DB6.032', 'DB5.197', 'DB5.171', 'DB8.102', 'DB7.170', 'DB8.116', 'DB5.087', 'DB8.178', 'DB5.185', 'DB4.145', 'DB7.084', 'DB4.081', 'DB8.145', 'DB7.126', 'DB8.026', 'DB6.067', 'DB7.120', 'DB8.089', 'DB6.157', 'DB6.216', 'DB6.135', 'DB7.210', 'DB6.270', 'DB7.042', 'DB6.174', 'DB5.263', 'DB4.213', 'DB4.167', 'DB8.080', 'DB8.146', 'DB5.135', 'DB5.027', 'DB7.141', 'DB6.087', 'DB6.187', 'DB4.025', 'DB7.110', 'DB8.030', 'DB6.057', 'DB6.126', 'DB5.127', 'DB5.074', 'DB7.095', 'DB7.038', 'DB8.062', 'DB7.054', 'DB4.099', 'DB6.113', 'DB8.099', 'DB6.089', 'DB7.098', 'DB8.295', 'DB7.195', 'DB6.196', 'DB5.246', 'DB6.034', 'DB5.086', 'DB7.006', 'DB4.026', 'DB4.056', 'DB8.274', 'DB5.094', 'DB6.243', 'DB7.071', 'DB7.251', 'DB7.046', 'DB4.069', 'DB4.155', 'DB7.053', 'DB8.005', 'DB5.078', 'DB4.084', 'DB4.223', 'DB7.047', 'DB7.223', 'DB7.268', 'DB7.018', 'DB6.036', 'DB8.112', 'DB7.049', 'DB8.249', 'DB6.124', 'DB6.257', 'DB8.231', 'DB8.201', 'DB7.000', 'DB4.228', 'DB5.247', 'DB8.019', 'DB4.132', 'DB8.267', 'DB5.256', 'DB8.218', 'DB7.204', 'DB5.286', 'DB7.052', 'DB7.057', 'DB4.004', 'DB7.248', 'DB6.155', 'DB4.078', 'DB7.118', 'DB5.010', 'DB4.267', 'DB7.221', 'DB4.041', 'DB8.213', 'DB4.268', 'DB7.011', 'DB6.211', 'DB5.071', 'DB7.116', 'DB4.057', 'DB5.297', 'DB7.208', 'DB6.173', 'DB6.123', 'DB6.226', 'DB5.111', 'DB8.240', 'DB8.233', 'DB7.232', 'DB6.177', 'DB4.060', 'DB5.136', 'DB4.244', 'DB5.013', 'DB6.225', 'DB7.300', 'DB5.103', 'DB7.106', 'DB8.173', 'DB7.113', 'DB6.051', 'DB8.171', 'DB8.137', 'DB6.105', 'DB8.266', 'DB4.235', 'DB5.033', 'DB8.119', 'DB8.202', 'DB4.107', 'DB4.036', 'DB5.144', 'DB6.002', 'DB5.284', 'DB7.139', 'DB4.281', 'DB5.073', 'DB5.268', 'DB4.221', 'DB5.148', 'DB8.082', 'DB5.008', 'DB4.208', 'DB5.240', 'DB4.001', 'DB5.040', 'DB7.093', 'DB7.213', 'DB6.109', 'DB4.097', 'DB8.103', 'DB5.180', 'DB8.039', 'DB7.055', 'DB7.245', 'DB4.215', 'DB5.273', 'DB7.192', 'DB7.041', 'DB6.202', 'DB4.259', 'DB4.075', 'DB5.097', 'DB8.035', 'DB8.260', 'DB5.232', 'DB8.219', 'DB7.218', 'DB5.181', 'DB4.177', 'DB7.034', 'DB7.244', 'DB6.043', 'DB5.067', 'DB7.152', 'DB7.035', 'DB5.157', 'DB6.056', 'DB8.115', 'DB6.228', 'DB8.181', 'DB7.176', 'DB4.135', 'DB4.172', 'DB7.017', 'DB4.283', 'DB7.205', 'DB5.237', 'DB5.189', 'DB6.120', 'DB5.216', 'DB5.183', 'DB8.042', 'DB5.230', 'DB6.030', 'DB8.002', 'DB8.159', 'DB7.164', 'DB4.142', 'DB6.290', 'DB7.089', 'DB4.077', 'DB6.104', 'DB7.294', 'DB5.287', 'DB4.173', 'DB7.091', 'DB6.068', 'DB6.065', 'DB6.164', 'DB5.276', 'DB7.083', 'DB6.082', 'DB4.225', 'DB7.112', 'DB8.289', 'DB6.231', 'DB5.267', 'DB6.035', 'DB7.147', 'DB5.081', 'DB4.071', 'DB4.122', 'DB6.039', 'DB5.122', 'DB7.005', 'DB7.043', 'DB6.052', 'DB6.185', 'DB5.058', 'DB8.036', 'DB7.101', 'DB4.039', 'DB6.029', 'DB5.052', 'DB8.142'}\n",
      "2021-04-05 19:59:26 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:59:26 INFO After filtering, 1172 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 1172 samples, 11834 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 25 samples, 11834 features\n",
      "**********\n",
      "processing ../studies/8 (29)\n",
      "['../studies/8/all.biom']\n",
      "2021-04-05 19:59:58 INFO loaded 33 samples, 780 features\n",
      "2021-04-05 19:59:58 WARNING These have metadata but do not have data - dropped (1): {'ERR315306'}\n",
      "2021-04-05 19:59:58 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 19:59:58 INFO After filtering, 33 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 33 samples, 780 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 33 samples, 780 features\n",
      "**********\n",
      "processing ../studies/21 (30)\n",
      "['../studies/21/all.biom']\n",
      "2021-04-05 20:00:30 INFO loaded 178 samples, 2597 features\n",
      "2021-04-05 20:00:30 WARNING These have metadata but do not have data - dropped (1): {'SRR6147218'}\n",
      "2021-04-05 20:00:30 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:00:30 INFO After filtering, 178 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 178 samples, 2597 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 178 samples, 2597 features\n",
      "**********\n",
      "processing ../studies/44 (31)\n",
      "['../studies/44/all.biom']\n",
      "2021-04-05 20:01:04 INFO loaded 875 samples, 11915 features\n",
      "2021-04-05 20:01:04 WARNING These have metadata but do not have data - dropped (64): {'ERR1545911', 'ERR1160673', 'ERR1545891', 'ERR1090161', 'ERR1545962', 'ERR1545986', 'ERR1545978', 'ERR1091354', 'ERR1089679', 'ERR1545802', 'ERR1074779', 'ERR1091155', 'ERR1545907', 'ERR1841813', 'ERR1075185', 'ERR1545774', 'ERR1074050', 'ERR1074092', 'ERR1545804', 'ERR1074007', 'ERR1160680', 'ERR1546016', 'ERR1072778', 'ERR1841525', 'ERR1597274', 'ERR1073198', 'ERR1089814', 'ERR1545780', 'ERR1074974', 'ERR1075896', 'ERR1073883', 'ERR1073330', 'ERR1075069', 'ERR1842199', 'ERR1545963', 'ERR1075272', 'ERR1546010', 'ERR2239717', 'ERR1074978', 'ERR1545939', 'ERR1545936', 'ERR1074005', 'ERR1072979', 'ERR1091147', 'ERR1545877', 'ERR2242158', 'ERR1080384', 'ERR1545747', 'ERR1545781', 'ERR1160475', 'ERR1074955', 'ERR1389624', 'ERR1545838', 'ERR1072794', 'ERR1545833', 'ERR1545990', 'ERR1545702', 'ERR1545803', 'ERR1546057', 'ERR1597126', 'ERR1389819', 'ERR1075071', 'ERR1072781', 'ERR1075286'}\n",
      "2021-04-05 20:01:04 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:01:04 INFO After filtering, 835 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 835 samples, 11915 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 835 samples, 11915 features\n",
      "**********\n",
      "processing ../studies/43 (32)\n",
      "['../studies/43/all.biom']\n",
      "2021-04-05 20:01:52 INFO loaded 91 samples, 3426 features\n",
      "2021-04-05 20:01:52 WARNING These have metadata but do not have data - dropped (9): {'ERR1545914', 'ERR1545813', 'ERR1075067', 'ERR1545849', 'ERR1841752', 'ERR1545876', 'ERR2239660', 'ERR1841803', 'ERR1073734'}\n",
      "2021-04-05 20:01:52 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:01:52 INFO After filtering, 89 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 89 samples, 3426 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 89 samples, 3426 features\n",
      "**********\n",
      "processing ../studies/36 (33)\n",
      "['../studies/36/all.biom']\n",
      "2021-04-05 20:02:39 INFO loaded 16 samples, 1191 features\n",
      "2021-04-05 20:02:39 WARNING These have metadata but do not have data - dropped (1): {'ERR1089974'}\n",
      "2021-04-05 20:02:39 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:02:39 INFO After filtering, 15 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 15 samples, 1191 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 15 samples, 1191 features\n",
      "**********\n",
      "processing ../studies/31 (34)\n",
      "['../studies/31/all.biom']\n",
      "2021-04-05 20:03:24 INFO loaded 70 samples, 1929 features\n",
      "2021-04-05 20:03:24 WARNING These have metadata but do not have data - dropped (18): {'ERR1331868', 'ERR1331858', 'ERR1331856', 'ERR1331867', 'ERR1331866', 'ERR1331871', 'ERR1331872', 'ERR1331865', 'ERR1331860', 'ERR1331869', 'ERR1331870', 'ERR1331864', 'ERR1331863', 'ERR1331862', 'ERR1331859', 'ERR1331861', 'ERR1331814', 'ERR1331857'}\n",
      "2021-04-05 20:03:24 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:03:24 INFO After filtering, 70 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 70 samples, 1929 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 70 samples, 1929 features\n",
      "**********\n",
      "processing ../studies/62 (35)\n",
      "['../studies/62/all.biom']\n",
      "2021-04-05 20:04:07 INFO loaded 196 samples, 5813 features\n",
      "2021-04-05 20:04:07 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:04:07 INFO After filtering, 196 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 196 samples, 5813 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 196 samples, 5813 features\n",
      "**********\n",
      "processing ../studies/54 (36)\n",
      "['../studies/54/all.biom']\n",
      "2021-04-05 20:04:54 INFO loaded 63 samples, 3657 features\n",
      "2021-04-05 20:04:54 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:04:54 INFO After filtering, 63 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 63 samples, 3657 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 63 samples, 3657 features\n",
      "**********\n",
      "processing ../studies/53 (37)\n",
      "['../studies/53/all.biom']\n",
      "2021-04-05 20:05:42 INFO loaded 115 samples, 4878 features\n",
      "2021-04-05 20:05:43 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:05:43 INFO After filtering, 115 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 115 samples, 4878 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 115 samples, 4878 features\n",
      "**********\n",
      "processing ../studies/37 (38)\n",
      "['../studies/37/all.biom']\n",
      "2021-04-05 20:06:32 INFO loaded 757 samples, 12055 features\n",
      "2021-04-05 20:06:32 WARNING These have metadata but do not have data - dropped (63): {'ERR1072865', 'ERR1545954', 'ERR1545899', 'ERR1841696', 'ERR1545922', 'ERR1459193', 'ERR1458945', 'ERR2580081', 'ERR1545999', 'ERR1316091', 'ERR1091311', 'ERR1072872', 'ERR1545923', 'ERR1091687', 'ERR2239311', 'ERR1545837', 'ERR2239796', 'ERR1075799', 'ERR1545979', 'ERR1091094', 'ERR1545863', 'ERR1077624', 'ERR1080064', 'ERR2239648', 'ERR1075259', 'ERR1091073', 'ERR1091610', 'ERR1545856', 'ERR1315992', 'ERR1072691', 'ERR1075115', 'ERR1080269', 'ERR1545860', 'ERR1545721', 'ERR1545995', 'ERR1545931', 'ERR1073095', 'ERR1073219', 'ERR1458825', 'ERR2239672', 'ERR2239737', 'ERR1842276', 'ERR1841719', 'ERR1841811', 'ERR1091136', 'ERR1546066', 'ERR1545835', 'ERR1389679', 'ERR1546000', 'ERR1075296', 'ERR1545811', 'ERR1090142', 'ERR1389829', 'ERR2238737', 'ERR1091704', 'ERR2239190', 'ERR1076845', 'ERR1091456', 'ERR1074977', 'ERR1545934', 'ERR1841820', 'ERR1545779', 'ERR1072933'}\n",
      "2021-04-05 20:06:32 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:06:32 INFO After filtering, 727 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 727 samples, 12055 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 727 samples, 12055 features\n",
      "**********\n",
      "processing ../studies/39 (39)\n",
      "['../studies/39/all.biom']\n",
      "2021-04-05 20:07:29 INFO loaded 625 samples, 10422 features\n",
      "2021-04-05 20:07:29 WARNING These have metadata but do not have data - dropped (35): {'ERR2238658', 'ERR1074953', 'ERR1841821', 'ERR1074855', 'ERR2239697', 'ERR1545875', 'ERR1854699', 'ERR2239793', 'ERR1080069', 'ERR2239393', 'ERR1073258', 'ERR2239770', 'ERR1841495', 'ERR1075897', 'ERR2318018', 'ERR1072703', 'ERR1545942', 'ERR1073829', 'ERR1545903', 'ERR1545815', 'ERR1842590', 'ERR2238750', 'ERR1545969', 'ERR1077488', 'ERR1089998', 'ERR1389924', 'ERR1074160', 'ERR1842514', 'ERR2239308', 'ERR1072835', 'ERR1841812', 'ERR1545985', 'ERR1545842', 'ERR1389959', 'ERR1076837'}\n",
      "2021-04-05 20:07:30 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:07:30 INFO After filtering, 587 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 587 samples, 10422 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 587 samples, 10422 features\n",
      "**********\n",
      "processing ../studies/52 (40)\n",
      "['../studies/52/all.biom']\n",
      "2021-04-05 20:08:31 INFO loaded 30 samples, 2580 features\n",
      "2021-04-05 20:08:31 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:08:31 INFO After filtering, 30 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 30 samples, 2580 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 30 samples, 2580 features\n",
      "**********\n",
      "processing ../studies/55 (41)\n",
      "['../studies/55/all.biom']\n",
      "2021-04-05 20:09:31 INFO loaded 74 samples, 3939 features\n",
      "2021-04-05 20:09:31 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:09:31 INFO After filtering, 74 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 74 samples, 3939 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 74 samples, 3939 features\n",
      "**********\n",
      "processing ../studies/46 (42)\n",
      "['../studies/46/all.biom']\n",
      "2021-04-05 20:10:31 INFO loaded 584 samples, 10980 features\n",
      "2021-04-05 20:10:31 WARNING These have metadata but do not have data - dropped (41): {'ERR1545904', 'ERR2238718', 'ERR1075338', 'ERR2239702', 'ERR1073073', 'ERR1545701', 'ERR1077699', 'ERR1546003', 'ERR1089788', 'ERR1545968', 'ERR1459007', 'ERR1546040', 'ERR1854779', 'ERR1075243', 'ERR2304178', 'ERR1841755', 'ERR1074261', 'ERR1074885', 'ERR1841818', 'ERR1074071', 'ERR1080238', 'ERR1545694', 'ERR1841546', 'ERR1089939', 'ERR1075226', 'ERR1545678', 'ERR1545890', 'ERR1545910', 'ERR1073933', 'ERR1545902', 'ERR1545916', 'ERR2239804', 'ERR1073802', 'ERR1546027', 'ERR1073744', 'ERR1545859', 'ERR1389664', 'ERR1545965', 'ERR2242536', 'ERR2239582', 'ERR1545961'}\n",
      "2021-04-05 20:10:31 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:10:31 INFO After filtering, 554 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 554 samples, 10980 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 554 samples, 10980 features\n",
      "**********\n",
      "processing ../studies/41 (43)\n",
      "['../studies/41/all.biom']\n",
      "2021-04-05 20:11:42 INFO loaded 84 samples, 3557 features\n",
      "2021-04-05 20:11:42 WARNING These have metadata but do not have data - dropped (5): {'ERR1075045', 'ERR2238709', 'ERR1545772', 'ERR1160360', 'ERR2238703'}\n",
      "2021-04-05 20:11:42 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:11:42 INFO After filtering, 80 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 80 samples, 3557 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 80 samples, 3557 features\n",
      "**********\n",
      "processing ../studies/48 (44)\n",
      "['../studies/48/all.biom']\n",
      "2021-04-05 20:12:46 INFO loaded 293 samples, 6184 features\n",
      "2021-04-05 20:12:46 WARNING These have metadata but do not have data - dropped (21): {'ERR2239290', 'ERR1075875', 'ERR1090826', 'ERR1545775', 'ERR1545869', 'ERR2308577', 'ERR1545878', 'ERR1160560', 'ERR2238646', 'ERR1841718', 'ERR1075817', 'ERR1089851', 'ERR1091689', 'ERR1546021', 'ERR1546022', 'ERR1089874', 'ERR1073280', 'ERR1089880', 'ERR2033565', 'ERR1091390', 'ERR1841798'}\n",
      "2021-04-05 20:12:47 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:12:47 INFO After filtering, 280 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 280 samples, 6184 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 280 samples, 6184 features\n",
      "**********\n",
      "processing ../studies/24 (45)\n",
      "['../studies/24/all.biom']\n",
      "2021-04-05 20:13:56 INFO loaded 25 samples, 999 features\n",
      "2021-04-05 20:13:56 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:13:56 INFO After filtering, 25 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 25 samples, 999 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 25 samples, 999 features\n",
      "**********\n",
      "processing ../studies/23 (46)\n",
      "['../studies/23/all.biom']\n",
      "2021-04-05 20:15:02 INFO loaded 114 samples, 1867 features\n",
      "2021-04-05 20:15:02 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:15:02 INFO After filtering, 114 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 114 samples, 1867 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 114 samples, 1867 features\n",
      "**********\n",
      "processing ../studies/4 (47)\n",
      "['../studies/4/all.biom']\n",
      "2021-04-05 20:16:06 INFO loaded 83 samples, 1652 features\n",
      "2021-04-05 20:16:06 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:16:06 INFO After filtering, 82 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 82 samples, 1652 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 82 samples, 1652 features\n",
      "**********\n",
      "processing ../studies/15 (48)\n",
      "['../studies/15/all.biom']\n",
      "2021-04-05 20:17:10 INFO loaded 174 samples, 2103 features\n",
      "2021-04-05 20:17:10 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:17:10 INFO After filtering, 162 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 162 samples, 2103 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 162 samples, 2103 features\n",
      "**********\n",
      "processing ../studies/3 (49)\n",
      "['../studies/3/all.biom']\n",
      "2021-04-05 20:18:15 INFO loaded 337 samples, 3021 features\n",
      "2021-04-05 20:18:15 WARNING These have metadata but do not have data - dropped (11): {'ERR1513792', 'ERR1513697', 'ERR1513809', 'ERR1513852', 'ERR1513897', 'ERR1513791', 'ERR1513823', 'ERR1513668', 'ERR1513848', 'ERR1513720', 'ERR1513893'}\n",
      "2021-04-05 20:18:15 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:18:15 INFO After filtering, 334 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 334 samples, 3021 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 334 samples, 3021 features\n",
      "**********\n",
      "processing ../studies/12 (50)\n",
      "['../studies/12/all.biom']\n",
      "2021-04-05 20:19:21 INFO loaded 2664 samples, 13732 features\n",
      "2021-04-05 20:19:21 WARNING These have data but do not have metadata: {'DB7.163', 'DB9.089', 'DB3.063', 'DB3.117', 'DB4.253', 'DB6.279', 'DB10.263', 'DB2.134', 'DB4.062', 'DB8.235', 'DB10.105', 'DB4.233', 'DB3.053', 'DB9.253', 'DB4.053', 'DB8.206', 'DB8.143', 'DB6.282', 'DB9.110', 'DB5.289', 'DB2.092', 'DB3.107', 'DB7.158', 'DB8.174', 'DB4.049', 'DB2.030', 'DB5.028', 'DB1.270', 'DB8.004', 'DB6.246', 'DB10.269', 'DB7.079', 'DB2.122', 'DB4.067', 'DB9.091', 'DB5.003', 'DB8.001', 'DB3.026', 'DB5.143', 'DB6.218', 'DB9.188', 'DB6.213', 'DB9.283', 'DB8.155', 'DB10.014', 'DB4.050', 'DB6.193', 'DB9.291', 'DB2.222', 'DB8.247', 'DB5.227', 'DB2.025', 'DB10.170', 'DB3.004', 'DB8.018', 'DB10.053', 'DB2.009', 'DB9.144', 'DB2.149', 'DB3.116', 'DB7.074', 'DB7.253', 'DB10.132', 'DB6.230', 'DB3.157', 'DB10.282', 'DB6.029', 'DB9.194', 'DB9.154', 'DB3.166', 'DB9.216', 'DB6.063', 'DB6.236', 'DB6.237', 'DB8.280', 'DB7.133', 'DB2.183', 'DB4.052', 'DB9.118', 'DB9.038', 'DB4.037', 'DB2.096', 'DB4.064', 'DB9.117', 'DB9.251', 'DB7.173', 'DB8.120', 'DB6.293', 'DB8.177', 'DB1.129', 'DB8.083', 'DB4.028', 'DB4.061', 'DB3.064', 'DB7.247', 'DB4.247', 'DB3.091', 'DB4.211', 'DB4.068', 'DB4.222', 'DB6.284', 'DB3.122', 'DB6.054', 'DB8.255', 'DB2.044', 'DB5.031', 'DB3.206', 'DB9.102', 'DB7.081', 'DB3.033', 'DB6.289', 'DB4.031', 'DB4.131', 'DB3.209', 'DB2.101', 'DB2.148', 'DB4.110', 'DB6.100', 'DB7.286', 'DB3.013', 'DB3.054', 'DB2.036', 'DB9.098', 'DB4.103', 'DB3.110', 'DB9.095', 'DB4.013', 'DB2.212', 'DB9.059', 'DB2.139', 'DB6.094', 'DB5.116', 'DB5.098', 'DB9.207', 'DB2.067', 'DB9.058', 'DB3.096', 'DB6.260', 'DB9.139', 'DB7.048', 'DB7.252', 'DB5.096', 'DB2.240', 'DB2.068', 'DB7.201', 'DB7.014', 'DB9.056', 'DB2.089', 'DB6.011', 'DB5.158', 'DB4.140', 'DB3.260', 'DB8.141', 'DB9.167', 'DB6.098', 'DB6.219', 'DB6.110', 'DB9.244', 'DB4.162', 'DB8.237', 'DB6.092', 'DB7.010', 'DB7.016', 'DB6.240', 'DB9.074', 'DB3.102', 'DB10.094', 'DB6.131', 'DB8.117', 'DB3.086', 'DB7.151', 'DB7.254', 'DB10.104', 'DB10.069', 'DB4.076', 'DB9.163', 'DB8.256', 'DB9.147', 'DB10.131', 'DB8.165', 'DB2.076', 'DB3.022', 'DB5.160', 'DB2.205', 'DB7.155', 'DB2.028', 'DB5.241', 'DB9.276', 'DB9.195', 'DB6.064', 'DB9.077', 'DB9.048', 'DB10.055', 'DB3.095', 'DB10.286', 'DB6.169', 'DB10.089', 'DB2.186', 'DB4.040', 'DB5.046', 'DB2.180', 'DB2.074', 'DB4.216', 'DB1.299', 'DB6.217', 'DB7.266', 'DB8.020', 'DB9.099', 'DB6.038', 'DB6.264', 'DB9.158', 'DB3.159', 'DB9.015', 'DB4.138', 'DB9.247', 'DB7.115', 'DB7.280', 'DB4.105', 'DB9.019', 'DB2.083', 'DB5.221', 'DB2.191', 'DB5.254', 'DB5.066', 'DB6.179', 'DB4.271', 'DB9.192', 'DB4.035', 'DB8.133', 'DB6.127', 'DB4.236', 'DB8.125', 'DB8.094', 'DB8.282', 'DB1.064', 'DB8.087', 'DB4.088', 'DB8.185', 'DB1.278', 'DB2.174', 'DB8.151', 'DB6.047', 'DB6.037', 'DB1.271', 'DB9.042', 'DB9.006', 'DB4.020', 'DB9.014', 'DB5.041', 'DB5.034', 'DB5.084', 'DB9.222', 'DB6.042', 'DB7.274', 'DB10.020', 'DB1.096', 'DB8.259', 'DB9.219', 'DB3.043', 'DB1.275', 'DB5.024', 'DB6.007', 'DB9.119', 'DB2.201', 'DB5.131', 'DB6.291', 'DB1.078', 'DB9.070', 'DB3.149', 'DB9.023', 'DB10.211', 'DB5.023', 'DB9.069', 'DB2.120', 'DB3.202', 'DB3.097', 'DB2.061', 'DB7.153', 'DB3.008', 'DB9.106', 'DB5.006', 'DB6.058', 'DB7.145', 'DB1.065', 'DB9.295', 'DB3.098', 'DB9.108', 'DB10.035', 'DB2.199', 'DB8.140', 'DB7.185', 'DB9.241', 'DB7.102', 'DB7.228', 'DB3.245', 'DB7.171', 'DB2.192', 'DB9.169', 'DB3.171', 'DB6.137', 'DB8.157', 'DB7.216', 'DB9.109', 'DB10.183', 'DB2.102', 'DB5.064', 'DB8.241', 'DB9.135', 'DB6.107', 'DB7.027', 'DB7.236', 'DB7.224', 'DB8.144', 'DB1.273', 'DB3.267', 'DB9.126', 'DB1.276', 'DB3.153', 'DB2.161', 'DB2.112', 'DB2.095', 'DB7.165', 'DB4.151', 'DB2.170', 'DB6.186', 'DB2.159', 'DB4.096', 'DB10.206', 'DB1.193', 'DB2.220', 'DB8.095', 'DB10.276', 'DB5.184', 'DB5.038', 'DB7.150', 'DB4.144', 'DB3.016', 'DB10.060', 'DB4.070', 'DB5.239', 'DB6.258', 'DB7.220', 'DB3.129', 'DB4.120', 'DB3.135', 'DB7.231', 'DB6.224', 'DB10.191', 'DB8.270', 'DB10.159', 'DB1.150', 'DB7.279', 'DB2.082', 'DB10.057', 'DB8.227', 'DB2.049', 'DB3.228', 'DB9.239', 'DB10.215', 'DB6.140', 'DB5.037', 'DB2.173', 'DB9.181', 'DB10.086', 'DB2.014', 'DB2.077', 'DB3.126', 'DB3.029', 'DB9.094', 'DB6.022', 'DB4.256', 'DB4.210', 'DB10.294', 'DB10.289', 'DB2.225', 'DB3.192', 'DB4.212', 'DB4.226', 'DB2.164', 'DB2.178', 'DB10.037', 'DB6.084', 'DB3.188', 'DB4.229', 'DB10.242', 'DB2.206', 'DB8.244', 'DB7.087', 'DB3.067', 'DB10.180', 'DB6.250', 'DB8.118', 'DB8.179', 'DB9.200', 'DB8.050', 'DB9.004', 'DB10.048', 'DB3.136', 'DB5.253', 'DB7.092', 'DB7.189', 'DB7.068', 'DB3.007', 'DB6.198', 'DB8.250', 'DB3.078', 'DB10.298', 'DB5.191', 'DB3.061', 'DB10.237', 'DB10.220', 'DB8.031', 'DB7.070', 'DB6.201', 'DB2.109', 'DB8.014', 'DB10.277', 'DB3.165', 'DB9.137', 'DB10.107', 'DB3.094', 'DB2.034', 'DB2.023', 'DB8.192', 'DB5.228', 'DB5.261', 'DB4.051', 'DB10.250', 'DB10.139', 'DB10.067', 'DB10.205', 'DB5.201', 'DB4.171', 'DB8.217', 'DB6.074', 'DB2.187', 'DB4.205', 'DB3.115', 'DB2.202', 'DB5.178', 'DB4.147', 'DB7.096', 'DB10.066', 'DB5.236', 'DB10.052', 'DB1.282', 'DB5.260', 'DB7.241', 'DB9.084', 'DB2.210', 'DB8.046', 'DB7.082', 'DB4.299', 'DB8.081', 'DB4.127', 'DB10.003', 'DB3.154', 'DB2.153', 'DB7.193', 'DB4.112', 'DB9.243', 'DB4.011', 'DB2.163', 'DB9.124', 'DB4.241', 'DB9.196', 'DB8.006', 'DB9.114', 'DB3.263', 'DB10.221', 'DB2.029', 'DB8.205', 'DB8.236', 'DB2.228', 'DB7.156', 'DB9.025', 'DB4.201', 'DB4.032', 'DB3.197', 'DB8.113', 'DB6.220', 'DB5.138', 'DB5.198', 'DB10.054', 'DB6.025', 'DB10.200', 'DB2.003', 'DB2.105', 'DB5.076', 'DB9.231', 'DB4.263', 'DB8.043', 'DB9.017', 'DB3.241', 'DB4.190', 'DB9.278', 'DB6.017', 'DB8.121', 'DB9.087', 'DB7.177', 'DB7.124', 'DB9.249', 'DB6.013', 'DB9.068', 'DB5.150', 'DB10.280', 'DB3.001', 'DB9.285', 'DB3.168', 'DB5.075', 'DB5.258', 'DB6.268', 'DB5.222', 'DB9.010', 'DB9.172', 'DB8.238', 'DB6.014', 'DB10.151', 'DB8.138', 'DB10.295', 'DB9.229', 'DB2.143', 'DB5.063', 'DB7.138', 'DB5.059', 'DB3.243', 'DB2.097', 'DB4.248', 'DB8.183', 'DB5.226', 'DB7.271', 'DB4.242', 'DB10.296', 'DB6.073', 'DB7.273', 'DB9.213', 'DB7.242', 'DB8.044', 'DB2.235', 'DB6.249', 'DB8.299', 'DB5.132', 'DB7.127', 'DB5.165', 'DB8.073', 'DB6.184', 'DB2.080', 'DB3.123', 'DB9.300', 'DB6.000', 'DB9.073', 'DB9.179', 'DB6.103', 'DB10.077', 'DB10.266', 'DB8.194', 'DB10.059', 'DB4.130', 'DB2.001', 'DB7.137', 'DB7.001', 'DB5.130', 'DB5.154', 'DB9.046', 'DB5.051', 'DB5.206', 'DB7.233', 'DB8.211', 'DB10.091', 'DB7.184', 'DB8.134', 'DB9.013', 'DB7.239', 'DB10.051', 'DB9.125', 'DB6.077', 'DB2.004', 'DB5.151', 'DB7.243', 'DB4.272', 'DB1.152', 'DB2.078', 'DB9.237', 'DB10.265', 'DB8.088', 'DB4.113', 'DB9.036', 'DB9.208', 'DB5.265', 'DB7.108', 'DB6.044', 'DB6.286', 'DB6.066', 'DB3.031', 'DB5.069', 'DB6.292', 'DB10.006', 'DB10.026', 'DB1.269', 'DB8.297', 'DB2.019', 'DB3.169', 'DB3.089', 'DB8.012', 'DB10.071', 'DB4.250', 'DB9.142', 'DB1.074', 'DB10.134', 'DB7.021', 'DB5.278', 'DB9.078', 'DB3.208', 'DB9.178', 'DB8.243', 'DB9.293', 'DB5.045', 'DB8.108', 'DB2.211', 'DB9.141', 'DB7.097', 'DB3.182', 'DB2.175', 'DB10.192', 'DB4.063', 'DB5.114', 'DB5.208', 'DB10.074', 'DB7.050', 'DB7.168', 'DB1.190', 'DB9.088', 'DB6.031', 'DB10.152', 'DB7.069', 'DB6.203', 'DB8.242', 'DB5.242', 'DB2.213', 'DB3.244', 'DB6.083', 'DB2.155', 'DB7.109', 'DB10.162', 'DB9.150', 'DB5.112', 'DB9.133', 'DB9.152', 'DB5.126', 'DB10.082', 'DB4.030', 'DB7.100', 'DB8.193', 'DB3.014', 'DB5.291', 'DB3.266', 'DB7.130', 'DB6.253', 'DB3.265', 'DB10.166', 'DB9.160', 'DB9.153', 'DB3.048', 'DB3.065', 'DB7.149', 'DB3.070', 'DB4.160', 'DB5.153', 'DB2.162', 'DB2.032', 'DB7.160', 'DB2.209', 'DB7.140', 'DB8.106', 'DB2.056', 'DB8.261', 'DB2.145', 'DB6.294', 'DB8.287', 'DB2.166', 'DB4.204', 'DB5.089', 'DB9.248', 'DB2.197', 'DB7.062', 'DB9.204', 'DB4.207', 'DB4.104', 'DB8.187', 'DB10.224', 'DB6.108', 'DB7.056', 'DB8.111', 'DB6.209', 'DB4.048', 'DB5.146', 'DB1.144', 'DB3.199', 'DB5.072', 'DB4.111', 'DB3.158', 'DB5.108', 'DB8.158', 'DB6.242', 'DB9.166', 'DB5.186', 'DB10.033', 'DB7.278', 'DB1.261', 'DB9.203', 'DB10.112', 'DB6.050', 'DB3.146', 'DB8.148', 'DB8.292', 'DB10.081', 'DB10.217', 'DB1.068', 'DB9.045', 'DB8.132', 'DB8.040', 'DB10.120', 'DB8.196', 'DB6.235', 'DB6.019', 'DB5.220', 'DB8.234', 'DB6.023', 'DB3.246', 'DB4.038', 'DB4.055', 'DB7.276', 'DB5.202', 'DB10.238', 'DB2.107', 'DB6.297', 'DB6.212', 'DB5.030', 'DB7.024', 'DB2.118', 'DB10.126', 'DB10.175', 'DB2.198', 'DB3.242', 'DB8.098', 'DB9.155', 'DB6.024', 'DB7.217', 'DB2.142', 'DB3.073', 'DB9.274', 'DB8.016', 'DB6.097', 'DB3.071', 'DB7.088', 'DB10.279', 'DB7.230', 'DB4.165', 'DB6.170', 'DB10.155', 'DB5.141', 'DB6.008', 'DB8.279', 'DB6.263', 'DB3.138', 'DB4.086', 'DB4.260', 'DB6.005', 'DB6.093', 'DB2.156', 'DB7.283', 'DB2.133', 'DB8.188', 'DB9.107', 'DB6.133', 'DB10.096', 'DB9.234', 'DB2.167', 'DB4.275', 'DB9.005', 'DB1.274', 'DB3.148', 'DB8.054', 'DB4.093', 'DB7.128', 'DB4.149', 'DB7.059', 'DB8.186', 'DB5.188', 'DB8.032', 'DB3.035', 'DB8.184', 'DB4.072', 'DB1.077', 'DB8.104', 'DB9.185', 'DB10.016', 'DB5.095', 'DB9.082', 'DB9.275', 'DB8.122', 'DB8.257', 'DB10.267', 'DB7.142', 'DB6.139', 'DB9.044', 'DB2.045', 'DB2.079', 'DB6.266', 'DB10.243', 'DB8.293', 'DB8.268', 'DB10.229', 'DB3.066', 'DB7.222', 'DB10.190', 'DB8.212', 'DB3.024', 'DB7.078', 'DB8.191', 'DB8.041', 'DB4.159', 'DB4.270', 'DB4.251', 'DB2.125', 'DB3.124', 'DB7.094', 'DB6.197', 'DB10.230', 'DB9.193', 'DB3.006', 'DB2.048', 'DB6.267', 'DB9.217', 'DB3.099', 'DB9.273', 'DB4.252', 'DB7.214', 'DB9.258', 'DB6.118', 'DB4.273', 'DB2.236', 'DB5.056', 'DB1.197', 'DB1.226', 'DB9.097', 'DB10.176', 'DB5.269', 'DB10.187', 'DB2.013', 'DB3.174', 'DB10.193', 'DB10.163', 'DB10.271', 'DB5.134', 'DB2.179', 'DB9.051', 'DB6.241', 'DB3.204', 'DB8.254', 'DB1.132', 'DB5.280', 'DB8.109', 'DB4.150', 'DB4.024', 'DB9.202', 'DB7.263', 'DB10.136', 'DB10.218', 'DB8.008', 'DB3.072', 'DB7.060', 'DB2.204', 'DB2.073', 'DB8.230', 'DB3.002', 'DB6.006', 'DB4.157', 'DB7.277', 'DB7.270', 'DB10.178', 'DB8.263', 'DB5.212', 'DB3.111', 'DB10.058', 'DB3.023', 'DB8.003', 'DB8.023', 'DB8.176', 'DB2.237', 'DB7.026', 'DB6.033', 'DB7.234', 'DB3.167', 'DB4.129', 'DB3.088', 'DB5.125', 'DB9.252', 'DB2.035', 'DB3.057', 'DB10.255', 'DB2.031', 'DB4.174', 'DB4.158', 'DB10.099', 'DB9.079', 'DB7.020', 'DB8.084', 'DB8.200', 'DB4.034', 'DB10.185', 'DB10.174', 'DB1.066', 'DB6.238', 'DB2.099', 'DB1.280', 'DB3.090', 'DB8.220', 'DB1.296', 'DB5.137', 'DB1.277', 'DB9.286', 'DB6.045', 'DB5.029', 'DB10.116', 'DB2.151', 'DB7.229', 'DB9.047', 'DB3.180', 'DB9.064', 'DB2.063', 'DB8.161', 'DB5.068', 'DB3.151', 'DB7.166', 'DB2.223', 'DB2.114', 'DB7.122', 'DB3.010', 'DB8.086', 'DB8.139', 'DB10.194', 'DB5.020', 'DB1.265', 'DB6.190', 'DB9.072', 'DB5.082', 'DB9.265', 'DB7.203', 'DB7.257', 'DB8.015', 'DB7.191', 'DB6.223', 'DB8.160', 'DB3.068', 'DB5.196', 'DB4.175', 'DB10.153', 'DB8.149', 'DB8.090', 'DB8.265', 'DB5.102', 'DB8.245', 'DB5.250', 'DB6.119', 'DB3.005', 'DB2.123', 'DB9.037', 'DB9.261', 'DB7.105', 'DB10.144', 'DB6.018', 'DB2.157', 'DB10.142', 'DB5.204', 'DB2.086', 'DB3.083', 'DB9.187', 'DB9.029', 'DB9.171', 'DB5.275', 'DB5.044', 'DB8.216', 'DB5.229', 'DB10.041', 'DB3.042', 'DB10.001', 'DB3.184', 'DB6.191', 'DB9.224', 'DB5.164', 'DB7.162', 'DB8.154', 'DB10.034', 'DB3.181', 'DB9.272', 'DB5.215', 'DB5.060', 'DB7.104', 'DB2.188', 'DB5.079', 'DB2.136', 'DB10.244', 'DB1.266', 'DB5.272', 'DB3.003', 'DB10.119', 'DB10.291', 'DB9.197', 'DB10.248', 'DB8.198', 'DB7.064', 'DB3.170', 'DB4.047', 'DB4.066', 'DB6.181', 'DB10.090', 'DB6.214', 'DB4.126', 'DB10.161', 'DB8.124', 'DB3.201', 'DB5.080', 'DB5.149', 'DB2.091', 'DB4.152', 'DB7.256', 'DB5.214', 'DB6.032', 'DB7.169', 'DB5.197', 'DB7.170', 'DB9.132', 'DB5.087', 'DB3.100', 'DB5.185', 'DB10.288', 'DB4.145', 'DB9.101', 'DB4.081', 'DB10.028', 'DB8.145', 'DB7.126', 'DB6.067', 'DB7.120', 'DB1.251', 'DB9.092', 'DB8.089', 'DB6.135', 'DB2.189', 'DB7.210', 'DB7.042', 'DB10.213', 'DB1.070', 'DB4.213', 'DB4.167', 'DB9.024', 'DB7.141', 'DB4.025', 'DB6.187', 'DB8.030', 'DB6.126', 'DB3.144', 'DB7.095', 'DB9.033', 'DB6.113', 'DB6.089', 'DB10.150', 'DB10.186', 'DB7.098', 'DB7.195', 'DB5.246', 'DB9.190', 'DB10.044', 'DB5.086', 'DB7.006', 'DB4.056', 'DB8.274', 'DB5.094', 'DB1.154', 'DB7.071', 'DB7.046', 'DB10.179', 'DB9.021', 'DB5.078', 'DB4.223', 'DB7.047', 'DB9.052', 'DB10.212', 'DB3.234', 'DB3.164', 'DB3.075', 'DB10.203', 'DB7.268', 'DB6.036', 'DB7.018', 'DB2.058', 'DB10.275', 'DB9.294', 'DB10.157', 'DB8.231', 'DB10.184', 'DB5.247', 'DB9.233', 'DB8.019', 'DB4.132', 'DB9.210', 'DB5.256', 'DB9.138', 'DB7.204', 'DB5.286', 'DB7.052', 'DB7.057', 'DB10.088', 'DB4.004', 'DB7.248', 'DB3.120', 'DB7.118', 'DB5.010', 'DB2.100', 'DB7.221', 'DB7.011', 'DB9.031', 'DB2.071', 'DB10.141', 'DB9.145', 'DB6.226', 'DB5.111', 'DB3.032', 'DB2.141', 'DB4.060', 'DB4.244', 'DB1.288', 'DB5.013', 'DB9.246', 'DB6.225', 'DB9.083', 'DB3.047', 'DB10.245', 'DB2.138', 'DB7.113', 'DB5.033', 'DB10.118', 'DB8.119', 'DB10.201', 'DB8.202', 'DB2.171', 'DB4.107', 'DB4.036', 'DB9.270', 'DB6.002', 'DB3.058', 'DB10.039', 'DB10.253', 'DB3.012', 'DB2.190', 'DB3.045', 'DB5.268', 'DB4.221', 'DB10.029', 'DB9.016', 'DB8.068', 'DB8.082', 'DB10.087', 'DB10.247', 'DB10.098', 'DB4.001', 'DB3.211', 'DB8.039', 'DB9.180', 'DB1.284', 'DB9.225', 'DB3.251', 'DB10.272', 'DB5.273', 'DB10.083', 'DB7.041', 'DB4.075', 'DB10.114', 'DB5.097', 'DB9.146', 'DB5.181', 'DB7.218', 'DB8.219', 'DB3.059', 'DB7.034', 'DB3.077', 'DB5.067', 'DB3.019', 'DB7.152', 'DB7.035', 'DB5.157', 'DB2.150', 'DB10.102', 'DB9.115', 'DB10.073', 'DB9.060', 'DB7.176', 'DB4.172', 'DB9.100', 'DB7.017', 'DB3.050', 'DB4.283', 'DB10.092', 'DB5.189', 'DB5.183', 'DB8.042', 'DB10.084', 'DB8.002', 'DB7.164', 'DB9.080', 'DB10.123', 'DB4.142', 'DB9.131', 'DB2.104', 'DB10.036', 'DB6.104', 'DB4.173', 'DB7.091', 'DB5.276', 'DB6.068', 'DB6.164', 'DB6.065', 'DB4.225', 'DB7.112', 'DB5.267', 'DB3.175', 'DB10.208', 'DB10.021', 'DB3.104', 'DB5.081', 'DB3.080', 'DB4.071', 'DB6.039', 'DB9.227', 'DB7.005', 'DB7.043', 'DB10.121', 'DB6.052', 'DB5.058', 'DB7.101', 'DB2.059', 'DB8.142', 'DB8.224', 'DB3.218', 'DB8.075', 'DB3.040', 'DB9.220', 'DB7.134', 'DB5.281', 'DB9.065', 'DB5.167', 'DB2.040', 'DB7.007', 'DB8.027', 'DB10.017', 'DB3.092', 'DB10.042', 'DB9.076', 'DB7.194', 'DB6.114', 'DB6.101', 'DB7.202', 'DB7.107', 'DB10.135', 'DB4.005', 'DB5.115', 'DB7.023', 'DB4.100', 'DB5.026', 'DB3.021', 'DB8.105', 'DB2.094', 'DB5.119', 'DB9.039', 'DB10.093', 'DB7.196', 'DB6.115', 'DB7.172', 'DB9.184', 'DB1.298', 'DB9.112', 'DB4.202', 'DB5.139', 'DB1.285', 'DB8.248', 'DB10.226', 'DB2.042', 'DB6.078', 'DB5.053', 'DB5.070', 'DB1.223', 'DB9.264', 'DB10.108', 'DB10.169', 'DB8.045', 'DB8.204', 'DB7.037', 'DB10.097', 'DB5.266', 'DB9.182', 'DB6.070', 'DB5.252', 'DB8.022', 'DB10.268', 'DB7.004', 'DB8.034', 'DB4.262', 'DB7.237', 'DB1.158', 'DB3.081', 'DB8.077', 'DB7.181', 'DB2.168', 'DB4.143', 'DB4.108', 'DB10.293', 'DB10.214', 'DB7.033', 'DB3.079', 'DB8.284', 'DB8.258', 'DB8.228', 'DB5.292', 'DB5.047', 'DB7.063', 'DB9.161', 'DB7.255', 'DB9.292', 'DB10.111', 'DB8.275', 'DB9.218', 'DB6.172', 'DB3.052', 'DB9.183', 'DB3.195', 'DB8.128', 'DB10.283', 'DB2.000', 'DB2.087', 'DB7.211', 'DB6.182', 'DB5.244', 'DB9.120', 'DB3.198', 'DB7.136', 'DB9.287', 'DB6.046', 'DB6.171', 'DB7.015', 'DB2.144', 'DB2.113', 'DB6.009', 'DB8.076', 'DB9.000', 'DB10.209', 'DB6.227', 'DB4.249', 'DB6.075', 'DB10.189', 'DB7.219', 'DB6.280', 'DB2.055', 'DB8.172', 'DB10.270', 'DB3.207', 'DB7.076', 'DB10.260', 'DB7.119', 'DB7.212', 'DB5.133', 'DB7.132', 'DB5.223', 'DB4.195', 'DB3.213', 'DB5.101', 'DB9.086', 'DB6.244', 'DB2.015', 'DB6.166', 'DB5.179', 'DB8.300', 'DB3.268', 'DB7.002', 'DB5.106', 'DB4.137', 'DB10.165', 'DB10.101', 'DB8.272', 'DB10.154', 'DB8.163', 'DB6.221', 'DB10.103', 'DB8.278', 'DB5.233', 'DB10.259', 'DB3.084', 'DB6.028', 'DB3.027', 'DB8.296', 'DB7.061', 'DB6.027', 'DB5.168', 'DB4.009', 'DB5.224', 'DB2.051', 'DB10.143', 'DB9.170', 'DB3.189', 'DB9.041', 'DB4.044', 'DB9.257', 'DB2.172', 'DB6.245', 'DB3.132', 'DB6.090', 'DB6.125', 'DB9.259', 'DB7.295', 'DB7.179', 'DB4.101', 'DB8.024', 'DB4.003', 'DB2.098', 'DB5.054', 'DB10.115', 'DB6.207', 'DB2.115', 'DB2.037', 'DB9.148', 'DB4.243', 'DB8.037', 'DB1.163', 'DB4.017', 'DB6.099', 'DB5.264', 'DB10.197', 'DB2.135', 'DB3.217', 'DB7.275', 'DB10.258', 'DB8.072', 'DB10.133', 'DB10.080', 'DB1.279', 'DB7.285', 'DB8.097', 'DB9.177', 'DB3.127', 'DB7.040', 'DB10.137', 'DB4.033', 'DB3.049', 'DB5.061', 'DB9.254', 'DB10.129', 'DB4.169', 'DB4.193', 'DB9.289', 'DB4.073', 'DB1.262', 'DB7.199', 'DB3.009', 'DB4.156', 'DB5.110', 'DB9.140', 'DB9.032', 'DB10.290', 'DB1.250', 'DB7.246', 'DB7.186', 'DB9.277', 'DB5.245', 'DB4.300', 'DB9.043', 'DB2.203', 'DB8.129', 'DB6.076', 'DB6.300', 'DB2.121', 'DB9.209', 'DB2.081', 'DB6.079', 'DB4.154', 'DB6.285', 'DB9.143', 'DB8.000', 'DB7.200', 'DB2.117', 'DB9.297', 'DB3.093', 'DB7.148', 'DB7.143', 'DB8.251', 'DB1.281', 'DB5.193', 'DB4.121', 'DB2.088', 'DB5.025', 'DB6.192', 'DB3.055', 'DB4.194', 'DB7.039', 'DB1.290', 'DB6.111', 'DB9.212', 'DB4.196', 'DB7.099', 'DB4.203', 'DB6.069', 'DB8.025', 'DB6.003', 'DB8.169', 'DB6.255', 'DB2.060', 'DB4.161', 'DB5.231', 'DB5.048', 'DB6.239', 'DB6.299', 'DB8.053', 'DB2.154', 'DB10.156', 'DB6.189', 'DB3.130', 'DB7.296', 'DB3.131', 'DB5.099', 'DB3.186', 'DB7.187', 'DB1.242', 'DB10.284', 'DB5.213', 'DB3.118', 'DB7.174', 'DB9.103', 'DB5.120', 'DB10.222', 'DB7.207', 'DB6.251', 'DB10.297', 'DB9.022', 'DB9.034', 'DB10.138', 'DB3.172', 'DB10.072', 'DB6.254', 'DB4.094', 'DB3.037', 'DB10.046', 'DB8.252', 'DB5.092', 'DB10.070', 'DB8.038', 'DB10.047', 'DB7.067', 'DB5.290', 'DB5.279', 'DB10.124', 'DB7.129', 'DB2.047', 'DB3.194', 'DB7.051', 'DB5.176', 'DB5.113', 'DB10.056', 'DB4.261', 'DB7.290', 'DB7.025', 'DB5.192', 'DB6.130', 'DB10.232', 'DB10.223', 'DB9.151', 'DB10.285', 'DB7.036', 'DB3.101', 'DB5.207', 'DB7.111', 'DB5.257', 'DB4.266', 'DB5.282', 'DB7.125', 'DB7.123', 'DB8.290', 'DB8.070', 'DB3.103', 'DB10.064', 'DB4.170', 'DB9.280', 'DB6.295', 'DB3.114', 'DB4.065', 'DB5.152', 'DB2.130', 'DB3.264', 'DB4.214', 'DB2.129', 'DB3.162', 'DB6.167', 'DB6.208', 'DB2.239', 'DB2.039', 'DB6.175', 'DB8.011', 'DB10.040', 'DB8.281', 'DB5.200', 'DB10.147', 'DB4.191', 'DB7.009', 'DB3.152', 'DB3.250', 'DB4.164', 'DB5.129', 'DB7.182', 'DB2.018', 'DB4.079', 'DB8.229', 'DB3.018', 'DB9.130', 'DB6.247', 'DB10.236', 'DB3.106', 'DB8.226', 'DB9.228', 'DB9.175', 'DB8.195', 'DB3.076', 'DB8.049', 'DB2.195', 'DB4.234', 'DB5.175', 'DB9.168', 'DB10.231', 'DB2.233', 'DB9.268', 'DB4.082', 'DB3.190', 'DB10.261', 'DB5.283', 'DB4.109', 'DB10.100', 'DB6.129', 'DB7.226', 'DB6.252', 'DB4.123', 'DB9.211', 'DB5.285', 'DB2.038', 'DB5.055', 'DB9.198', 'DB2.010', 'DB10.177', 'DB8.271', 'DB3.196', 'DB1.283', 'DB8.051', 'DB6.086', 'DB8.285', 'DB1.067', 'DB3.060', 'DB5.049', 'DB9.127', 'DB2.219', 'DB6.041', 'DB6.062', 'DB6.206', 'DB8.199', 'DB8.209', 'DB7.249', 'DB1.267', 'DB9.290', 'DB9.156', 'DB5.090', 'DB8.207', 'DB8.291', 'DB5.243', 'DB9.018', 'DB9.071', 'DB4.232', 'DB5.174', 'DB9.271', 'DB9.221', 'DB4.027', 'DB5.163', 'DB8.021', 'DB5.018', 'DB5.249', 'DB5.205', 'DB5.124', 'DB10.196', 'DB7.264', 'DB8.060', 'DB2.147', 'DB3.133', 'DB7.075', 'DB7.188', 'DB8.208', 'DB4.219', 'DB10.011', 'DB6.229', 'DB6.183', 'DB9.001', 'DB5.147', 'DB4.237', 'DB3.128', 'DB3.119', 'DB8.180', 'DB10.239', 'DB8.074', 'DB7.090', 'DB5.035', 'DB5.262', 'DB5.259', 'DB5.225', 'DB9.121', 'DB5.210', 'DB7.209', 'DB4.133', 'DB5.203', 'DB5.128', 'DB6.168', 'DB4.095', 'DB8.283', 'DB3.108', 'DB4.074', 'DB6.117', 'DB3.147', 'DB4.153', 'DB8.269', 'DB2.110', 'DB5.255', 'DB5.199', 'DB5.107', 'DB2.006', 'DB4.254', 'DB2.017', 'DB9.066', 'DB7.284', 'DB6.134', 'DB4.168', 'DB5.219', 'DB9.067', 'DB9.096', 'DB10.188', 'DB2.158', 'DB8.298', 'DB5.271', 'DB10.273', 'DB5.105', 'DB9.242', 'DB9.226', 'DB8.262', 'DB9.235', 'DB8.190', 'DB2.221', 'DB2.229', 'DB8.010', 'DB6.102', 'DB3.193', 'DB8.215', 'DB8.210', 'DB7.259', 'DB10.262', 'DB4.043', 'DB6.015', 'DB1.142', 'DB9.191', 'DB4.148', 'DB2.242', 'DB4.206', 'DB4.016', 'DB10.013', 'DB7.144', 'DB6.095', 'DB10.274', 'DB5.039', 'DB4.045', 'DB8.253', 'DB4.209', 'DB3.017', 'DB8.225', 'DB5.218', 'DB4.163', 'DB5.121', 'DB5.195', 'DB9.260', 'DB9.090', 'DB7.072', 'DB10.234', 'DB4.083', 'DB7.250', 'DB9.240', 'DB5.104', 'DB3.062', 'DB6.161', 'DB4.058', 'DB3.230', 'DB8.150', 'DB10.210', 'DB10.117', 'DB10.207', 'DB8.056', 'DB5.065', 'DB2.072', 'DB3.038', 'DB10.204', 'DB6.060', 'DB3.044', 'DB5.251', 'DB5.177', 'DB7.269', 'DB7.267', 'DB4.124', 'DB6.049', 'DB1.076', 'DB7.288', 'DB9.111', 'DB4.220', 'DB6.004', 'DB9.027', 'DB9.262', 'DB3.177', 'DB7.022', 'DB6.112', 'DB6.096', 'DB2.169', 'DB10.085', 'DB3.191', 'DB5.017', 'DB3.112', 'DB7.175', 'DB9.057', 'DB7.225', 'DB7.178', 'DB3.020', 'DB6.016', 'DB1.184', 'DB6.234', 'DB7.008', 'DB9.215', 'DB10.233', 'DB9.162', 'DB6.215', 'DB4.255', 'DB5.217', 'DB3.121', 'DB4.192', 'DB5.182', 'DB6.296', 'DB7.161', 'DB9.003', 'DB6.128', 'DB9.053', 'DB9.230', 'DB2.227', 'DB7.293', 'DB8.168', 'DB4.224', 'DB3.229', 'DB8.294', 'DB8.007', 'DB6.072', 'DB6.194', 'DB8.170', 'DB10.235', 'DB7.291', 'DB5.042', 'DB9.205', 'DB2.116', 'DB2.165', 'DB6.012', 'DB7.167', 'DB1.072', 'DB9.062', 'DB9.116', 'DB6.222', 'DB7.206', 'DB10.008', 'DB2.245', 'DB3.254', 'DB4.090', 'DB10.031', 'DB4.239', 'DB6.265', 'DB3.179', 'DB10.252', 'DB9.238', 'DB9.263', 'DB2.160', 'DB8.164', 'DB10.068', 'DB5.159', 'DB6.132', 'DB6.048', 'DB10.164', 'DB2.043', 'DB9.236', 'DB7.227', 'DB7.066', 'DB10.012', 'DB4.054', 'DB3.028', 'DB7.058', 'DB8.127', 'DB8.107', 'DB3.160', 'DB9.282', 'DB4.021', 'DB10.125', 'DB7.159', 'DB5.077', 'DB8.232', 'DB8.152', 'DB8.085', 'DB10.278', 'DB2.232', 'DB10.241', 'DB10.002', 'DB9.007', 'DB10.167', 'DB10.256', 'DB10.264', 'DB5.238', 'DB4.217', 'DB10.148', 'DB9.281', 'DB8.013', 'DB6.040', 'DB4.098', 'DB5.277', 'DB5.248', 'DB6.188', 'DB7.238', 'DB2.062', 'DB6.071', 'DB7.065', 'DB10.172', 'DB3.025', 'DB2.127', 'DB10.181', 'DB9.214', 'DB4.141', 'DB7.289', 'DB10.219', 'DB2.026', 'DB2.053', 'DB2.052', 'DB6.261', 'DB9.093', 'DB8.029', 'DB10.292', 'DB3.240', 'DB9.176', 'DB2.152', 'DB9.085', 'DB5.190', 'DB8.123', 'DB6.199', 'DB6.204', 'DB8.091', 'DB6.116', 'DB6.288', 'DB7.235', 'DB5.091', 'DB6.298', 'DB7.262', 'DB3.225', 'DB2.215', 'DB7.265', 'DB8.182', 'DB9.054', 'DB1.075', 'DB5.109', 'DB3.269', 'DB9.061', 'DB9.008', 'DB3.155', 'DB5.274', 'DB2.046', 'DB3.015', 'DB4.018', 'DB10.199', 'DB2.241', 'DB6.080', 'DB6.248', 'DB2.093', 'DB10.168', 'DB7.281', 'DB7.240', 'DB8.079', 'DB7.183', 'DB7.154', 'DB9.134', 'DB3.220', 'DB3.113', 'DB4.087', 'DB5.118', 'DB4.059', 'DB9.279', 'DB8.052', 'DB7.135', 'DB4.257', 'DB8.093', 'DB5.117', 'DB8.221', 'DB4.200', 'DB10.109', 'DB6.178', 'DB6.020', 'DB2.024', 'DB3.247', 'DB6.088', 'DB6.053', 'DB1.286', 'DB4.274', 'DB9.011', 'DB6.195', 'DB10.160', 'DB8.147', 'DB10.106', 'DB3.210', 'DB5.019', 'DB8.136', 'DB6.283', 'DB10.216', 'DB9.136', 'DB7.114', 'DB6.281', 'DB6.138', 'DB6.259', 'DB2.007', 'DB5.085', 'DB9.128', 'DB2.132', 'DB2.146', 'DB7.146', 'DB5.288', 'DB2.069', 'DB5.011', 'DB2.090', 'DB5.142', 'DB2.108', 'DB2.070', 'DB8.048', 'DB9.269', 'DB6.055', 'DB8.100', 'DB10.075', 'DB9.020', 'DB10.149', 'DB8.092', 'DB8.009', 'DB8.175', 'DB6.176', 'DB9.255', 'DB10.079', 'DB10.173', 'DB2.106', 'DB4.134', 'DB5.083', 'DB7.260', 'DB9.159', 'DB7.013', 'DB5.209', 'DB5.156', 'DB2.021', 'DB3.105', 'DB9.050', 'DB8.101', 'DB5.093', 'DB8.135', 'DB9.026', 'DB2.124', 'DB9.223', 'DB3.156', 'DB6.262', 'DB5.194', 'DB4.258', 'DB10.128', 'DB9.298', 'DB3.039', 'DB10.249', 'DB10.122', 'DB4.042', 'DB7.073', 'DB9.129', 'DB9.035', 'DB6.200', 'DB8.153', 'DB9.232', 'DB7.180', 'DB8.131', 'DB6.010', 'DB3.109', 'DB6.091', 'DB7.282', 'DB8.214', 'DB4.269', 'DB9.105', 'DB8.110', 'DB5.270', 'DB7.197', 'DB9.049', 'DB7.215', 'DB3.231', 'DB5.062', 'DB8.156', 'DB4.046', 'DB7.292', 'DB3.137', 'DB4.128', 'DB4.089', 'DB8.114', 'DB10.022', 'DB3.051', 'DB10.240', 'DB5.050', 'DB2.131', 'DB7.019', 'DB2.226', 'DB2.176', 'DB2.103', 'DB5.043', 'DB2.011', 'DB10.225', 'DB10.009', 'DB8.288', 'DB10.063', 'DB9.002', 'DB10.127', 'DB2.016', 'DB3.212', 'DB4.238', 'DB6.233', 'DB4.085', 'DB6.163', 'DB8.028', 'DB8.203', 'DB10.198', 'DB4.091', 'DB7.077', 'DB7.103', 'DB4.227', 'DB2.194', 'DB4.022', 'DB7.258', 'DB7.044', 'DB3.183', 'DB9.012', 'DB4.146', 'DB3.219', 'DB6.106', 'DB3.011', 'DB7.272', 'DB9.245', 'DB5.036', 'DB10.158', 'DB5.155', 'DB3.056', 'DB4.246', 'DB5.100', 'DB9.186', 'DB3.036', 'DB7.261', 'DB2.050', 'DB10.032', 'DB7.157', 'DB10.195', 'DB4.023', 'DB4.125', 'DB6.205', 'DB2.181', 'DB1.069', 'DB7.198', 'DB1.272', 'DB7.190', 'DB7.298', 'DB7.045', 'DB7.003', 'DB4.298', 'DB5.211', 'DB8.246', 'DB10.145', 'DB5.021', 'DB6.162', 'DB10.065', 'DB4.245', 'DB8.286', 'DB5.123', 'DB8.017', 'DB2.207', 'DB3.150', 'DB5.235', 'DB10.257', 'DB5.187', 'DB7.287', 'DB10.049', 'DB8.078', 'DB3.074', 'DB3.145', 'DB3.069', 'DB7.131', 'DB3.125', 'DB5.145', 'DB5.169', 'DB7.117', 'DB6.165', 'DB7.080', 'DB5.161', 'DB7.121', 'DB5.057', 'DB8.273', 'DB2.140', 'DB8.126', 'DB2.119', 'DB5.002', 'DB8.239', 'DB5.171', 'DB2.084', 'DB8.102', 'DB8.116', 'DB1.297', 'DB8.178', 'DB10.005', 'DB7.084', 'DB2.111', 'DB10.110', 'DB2.005', 'DB8.026', 'DB6.216', 'DB3.187', 'DB9.149', 'DB1.268', 'DB6.174', 'DB3.030', 'DB5.263', 'DB10.246', 'DB8.146', 'DB5.135', 'DB8.080', 'DB5.027', 'DB10.050', 'DB3.139', 'DB6.087', 'DB7.110', 'DB10.007', 'DB6.057', 'DB8.065', 'DB10.140', 'DB5.127', 'DB9.063', 'DB2.193', 'DB5.074', 'DB9.299', 'DB9.122', 'DB7.038', 'DB2.008', 'DB7.054', 'DB4.099', 'DB8.099', 'DB6.196', 'DB8.295', 'DB6.034', 'DB4.026', 'DB3.185', 'DB9.055', 'DB6.243', 'DB10.146', 'DB3.161', 'DB7.251', 'DB4.069', 'DB4.155', 'DB7.053', 'DB3.046', 'DB3.176', 'DB8.005', 'DB2.064', 'DB10.043', 'DB4.084', 'DB2.085', 'DB7.223', 'DB8.112', 'DB7.049', 'DB8.249', 'DB6.257', 'DB10.076', 'DB8.201', 'DB4.228', 'DB2.126', 'DB9.173', 'DB10.030', 'DB3.087', 'DB8.267', 'DB8.218', 'DB10.171', 'DB9.104', 'DB4.078', 'DB4.041', 'DB8.213', 'DB4.268', 'DB6.211', 'DB5.071', 'DB7.116', 'DB4.057', 'DB2.185', 'DB9.284', 'DB7.208', 'DB6.173', 'DB8.240', 'DB8.233', 'DB10.228', 'DB7.232', 'DB10.202', 'DB5.136', 'DB6.177', 'DB3.178', 'DB5.103', 'DB7.106', 'DB8.173', 'DB9.123', 'DB6.051', 'DB8.171', 'DB3.239', 'DB8.137', 'DB6.105', 'DB8.266', 'DB4.235', 'DB10.023', 'DB9.009', 'DB5.144', 'DB5.284', 'DB7.139', 'DB4.281', 'DB5.073', 'DB5.148', 'DB9.030', 'DB3.041', 'DB4.208', 'DB3.082', 'DB5.240', 'DB9.267', 'DB10.010', 'DB5.040', 'DB6.109', 'DB4.097', 'DB7.093', 'DB7.213', 'DB8.103', 'DB2.234', 'DB9.081', 'DB5.180', 'DB10.004', 'DB3.223', 'DB7.055', 'DB9.189', 'DB7.245', 'DB4.215', 'DB10.254', 'DB7.192', 'DB9.296', 'DB9.174', 'DB9.040', 'DB6.202', 'DB4.259', 'DB10.027', 'DB10.227', 'DB9.250', 'DB8.035', 'DB9.164', 'DB8.260', 'DB5.232', 'DB7.244', 'DB6.043', 'DB9.157', 'DB10.251', 'DB6.056', 'DB6.228', 'DB8.115', 'DB8.181', 'DB4.135', 'DB2.184', 'DB9.266', 'DB7.205', 'DB10.061', 'DB5.237', 'DB1.141', 'DB5.216', 'DB6.120', 'DB5.230', 'DB6.030', 'DB8.159', 'DB3.034', 'DB3.200', 'DB10.018', 'DB6.290', 'DB7.089', 'DB4.077', 'DB2.218', 'DB5.005', 'DB9.075', 'DB2.238', 'DB2.002', 'DB7.294', 'DB5.287', 'DB3.134', 'DB10.045', 'DB10.281', 'DB10.113', 'DB10.095', 'DB7.083', 'DB4.240', 'DB6.082', 'DB2.128', 'DB8.289', 'DB6.231', 'DB6.035', 'DB9.028', 'DB10.130', 'DB7.147', 'DB4.122', 'DB5.122', 'DB1.071', 'DB6.185', 'DB9.206', 'DB1.073', 'DB8.036', 'DB4.039', 'DB5.052'}\n",
      "2021-04-05 20:19:22 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:19:22 INFO After filtering, 2373 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 2373 samples, 13732 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 224 samples, 13732 features\n",
      "**********\n",
      "processing ../studies/49 (52)\n",
      "['../studies/49/all.biom']\n",
      "2021-04-05 20:20:38 INFO loaded 273 samples, 8112 features\n",
      "2021-04-05 20:20:38 WARNING These have metadata but do not have data - dropped (28): {'ERR1545766', 'ERR1545737', 'ERR1545712', 'ERR1074598', 'ERR1545980', 'ERR1074963', 'ERR1545898', 'ERR1091143', 'ERR1841753', 'ERR1546004', 'ERR1841517', 'ERR2032154', 'ERR1080264', 'ERR1545971', 'ERR1090164', 'ERR1545713', 'ERR1091007', 'ERR1090991', 'ERR2238662', 'ERR1546037', 'ERR1545736', 'ERR1091581', 'ERR1091017', 'ERR1090999', 'ERR1090029', 'ERR1545777', 'ERR1841808', 'ERR1075006'}\n",
      "2021-04-05 20:20:38 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:20:38 INFO After filtering, 263 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 263 samples, 8112 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 263 samples, 8112 features\n",
      "**********\n",
      "processing ../studies/40 (53)\n",
      "['../studies/40/all.biom']\n",
      "2021-04-05 20:21:52 INFO loaded 86 samples, 3917 features\n",
      "2021-04-05 20:21:52 WARNING These have metadata but do not have data - dropped (7): {'ERR1073218', 'ERR2580024', 'ERR1091444', 'ERR1091053', 'ERR1841756', 'ERR1090007', 'ERR1075021'}\n",
      "2021-04-05 20:21:52 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:21:52 INFO After filtering, 85 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 85 samples, 3917 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 85 samples, 3917 features\n",
      "**********\n",
      "processing ../studies/47 (54)\n",
      "['../studies/47/all.biom']\n",
      "2021-04-05 20:23:05 INFO loaded 257 samples, 7437 features\n",
      "2021-04-05 20:23:05 WARNING These have metadata but do not have data - dropped (17): {'ERR1074882', 'ERR1545836', 'ERR1545948', 'ERR1545812', 'ERR1841629', 'ERR1091158', 'ERR1073703', 'ERR1458779', 'ERR1160308', 'ERR1841809', 'ERR1545953', 'ERR1073252', 'ERR1091085', 'ERR1091044', 'ERR1841777', 'ERR1841802', 'ERR1546002'}\n",
      "2021-04-05 20:23:05 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:23:05 INFO After filtering, 247 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 247 samples, 7437 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 247 samples, 7437 features\n",
      "**********\n",
      "processing ../studies/2 (55)\n",
      "['../studies/2/all.biom']\n",
      "2021-04-05 20:24:24 INFO loaded 179 samples, 2528 features\n",
      "2021-04-05 20:24:24 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:24:24 INFO After filtering, 179 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 179 samples, 2528 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 179 samples, 2528 features\n",
      "**********\n",
      "processing ../studies/13 (56)\n",
      "['../studies/13/all.biom']\n",
      "2021-04-05 20:25:40 INFO loaded 31 samples, 925 features\n",
      "2021-04-05 20:25:40 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:25:40 INFO After filtering, 31 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 31 samples, 925 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 31 samples, 925 features\n",
      "**********\n",
      "processing ../studies/5 (57)\n",
      "['../studies/5/all.biom']\n",
      "2021-04-05 20:26:54 INFO loaded 333 samples, 4664 features\n",
      "2021-04-05 20:26:54 WARNING These have metadata but do not have data - dropped (507): {'SRR10911493', 'SRR10911928', 'SRR10911696', 'SRR10911171', 'SRR10911808', 'SRR10911335', 'SRR10911445', 'SRR10911244', 'SRR10911330', 'SRR10911418', 'SRR10911953', 'SRR10911447', 'SRR10911900', 'SRR10911889', 'SRR10911349', 'SRR10911239', 'SRR10911870', 'SRR10911222', 'SRR10911963', 'SRR10911891', 'SRR10911734', 'SRR10911887', 'SRR10911365', 'SRR10911745', 'SRR10911880', 'SRR10911196', 'SRR10911710', 'SRR10911862', 'SRR10911368', 'SRR10911958', 'SRR10911163', 'SRR10911726', 'SRR10911829', 'SRR10911911', 'SRR10911816', 'SRR10911852', 'SRR10911243', 'SRR10911820', 'SRR10911678', 'SRR10911159', 'SRR10911814', 'SRR10911181', 'SRR10911697', 'SRR10911861', 'SRR10911962', 'SRR10911850', 'SRR10911276', 'SRR10911853', 'SRR10911338', 'SRR10911192', 'SRR10911446', 'SRR10911707', 'SRR10911310', 'SRR10911947', 'SRR10911441', 'SRR10911823', 'SRR10911448', 'SRR10911687', 'SRR10911226', 'SRR10911322', 'SRR10911705', 'SRR10911832', 'SRR10911860', 'SRR10911193', 'SRR10911875', 'SRR10911763', 'SRR10911921', 'SRR10911149', 'SRR10911806', 'SRR10911692', 'SRR10911712', 'SRR10911956', 'SRR10911798', 'SRR10911346', 'SRR10911835', 'SRR10911162', 'SRR10911156', 'SRR10911158', 'SRR10911700', 'SRR10911709', 'SRR10911505', 'SRR10911782', 'SRR10911896', 'SRR10911848', 'SRR10911164', 'SRR10911800', 'SRR10911905', 'SRR10911680', 'SRR10911248', 'SRR10911918', 'SRR10911320', 'SRR10911948', 'SRR10911144', 'SRR10911242', 'SRR10911298', 'SRR10911755', 'SRR10911922', 'SRR10911735', 'SRR10911357', 'SRR10911919', 'SRR10911356', 'SRR10911373', 'SRR10911685', 'SRR10911197', 'SRR10911555', 'SRR10911359', 'SRR10911247', 'SRR10911926', 'SRR10911874', 'SRR10911528', 'SRR10911878', 'SRR10911175', 'SRR10911160', 'SRR10911440', 'SRR10911730', 'SRR10911788', 'SRR10911792', 'SRR10911336', 'SRR10911888', 'SRR10911229', 'SRR10911831', 'SRR10911733', 'SRR10911610', 'SRR10911199', 'SRR10911287', 'SRR10911797', 'SRR10911885', 'SRR10911177', 'SRR10911779', 'SRR10911824', 'SRR10911769', 'SRR10911382', 'SRR10911371', 'SRR10911224', 'SRR10911701', 'SRR10911718', 'SRR10911315', 'SRR10911343', 'SRR10911327', 'SRR10911698', 'SRR10911954', 'SRR10911153', 'SRR10911183', 'SRR10911746', 'SRR10911172', 'SRR10911369', 'SRR10911145', 'SRR10911378', 'SRR10911810', 'SRR10911451', 'SRR10911713', 'SRR10911754', 'SRR10911693', 'SRR10911377', 'SRR10911790', 'SRR10911347', 'SRR10911326', 'SRR10911703', 'SRR10911457', 'SRR10911711', 'SRR10911787', 'SRR10911185', 'SRR10911951', 'SRR10911873', 'SRR10911187', 'SRR10911688', 'SRR10911304', 'SRR10911339', 'SRR10911238', 'SRR10911150', 'SRR10911844', 'SRR10911864', 'SRR10911912', 'SRR10911348', 'SRR10911902', 'SRR10911311', 'SRR10911950', 'SRR10911813', 'SRR10911727', 'SRR10911210', 'SRR10911753', 'SRR10911331', 'SRR10911191', 'SRR10911764', 'SRR10911748', 'SRR10911883', 'SRR10911795', 'SRR10911722', 'SRR10911716', 'SRR10911694', 'SRR10911316', 'SRR10911186', 'SRR10911328', 'SRR10911817', 'SRR10911704', 'SRR10911245', 'SRR10911842', 'SRR10911367', 'SRR10911925', 'SRR10911857', 'SRR10911720', 'SRR10911897', 'SRR10911442', 'SRR10911775', 'SRR10911194', 'SRR10911855', 'SRR10911146', 'SRR10911964', 'SRR10911309', 'SRR10911344', 'SRR10911908', 'SRR10911354', 'SRR10911453', 'SRR10911578', 'SRR10911742', 'SRR10911773', 'SRR10911838', 'SRR10911325', 'SRR10911751', 'SRR10911223', 'SRR10911961', 'SRR10911246', 'SRR10911783', 'SRR10911913', 'SRR10911772', 'SRR10911924', 'SRR10911628', 'SRR10911907', 'SRR10911165', 'SRR10911719', 'SRR10911379', 'SRR10911872', 'SRR10911740', 'SRR10911750', 'SRR10911854', 'SRR10911909', 'SRR10911943', 'SRR10911372', 'SRR10911825', 'SRR10911271', 'SRR10911351', 'SRR10911195', 'SRR10911681', 'SRR10911154', 'SRR10911353', 'SRR10911937', 'SRR10911731', 'SRR10911736', 'SRR10911826', 'SRR10911933', 'SRR10911270', 'SRR10911566', 'SRR10911847', 'SRR10911780', 'SRR10911312', 'SRR10911168', 'SRR10911952', 'SRR10911767', 'SRR10911173', 'SRR10911725', 'SRR10911946', 'SRR10911895', 'SRR10911863', 'SRR10911932', 'SRR10911840', 'SRR10911757', 'SRR10911358', 'SRR10911714', 'SRR10911939', 'SRR10911821', 'SRR10911375', 'SRR10911957', 'SRR10911805', 'SRR10911308', 'SRR10911686', 'SRR10911809', 'SRR10911599', 'SRR10911148', 'SRR10911869', 'SRR10911184', 'SRR10911777', 'SRR10911364', 'SRR10911230', 'SRR10911879', 'SRR10911721', 'SRR10911917', 'SRR10911960', 'SRR10911152', 'SRR10911241', 'SRR10911319', 'SRR10911147', 'SRR10911233', 'SRR10911884', 'SRR10911898', 'SRR10911724', 'SRR10911455', 'SRR10911589', 'SRR10911801', 'SRR10911449', 'SRR10911760', 'SRR10911965', 'SRR10911765', 'SRR10911929', 'SRR10911942', 'SRR10911577', 'SRR10911143', 'SRR10911352', 'SRR10911770', 'SRR10911915', 'SRR10911708', 'SRR10911739', 'SRR10911340', 'SRR10911232', 'SRR10911324', 'SRR10911901', 'SRR10911881', 'SRR10911231', 'SRR10911717', 'SRR10911332', 'SRR10911389', 'SRR10911737', 'SRR10911945', 'SRR10911738', 'SRR10911967', 'SRR10911802', 'SRR10911400', 'SRR10911867', 'SRR10911450', 'SRR10911454', 'SRR10911314', 'SRR10911227', 'SRR10911968', 'SRR10911342', 'SRR10911151', 'SRR10911456', 'SRR10911689', 'SRR10911851', 'SRR10911894', 'SRR10911899', 'SRR10911240', 'SRR10911856', 'SRR10911944', 'SRR10911836', 'SRR10911812', 'SRR10911176', 'SRR10911381', 'SRR10911360', 'SRR10911333', 'SRR10911236', 'SRR10911906', 'SRR10911752', 'SRR10911892', 'SRR10911161', 'SRR10911762', 'SRR10911935', 'SRR10911741', 'SRR10911776', 'SRR10911904', 'SRR10911789', 'SRR10911683', 'SRR10911793', 'SRR10911667', 'SRR10911732', 'SRR10911914', 'SRR10911656', 'SRR10911866', 'SRR10911188', 'SRR10911786', 'SRR10911539', 'SRR10911756', 'SRR10911182', 'SRR10911843', 'SRR10911916', 'SRR10911682', 'SRR10911715', 'SRR10911744', 'SRR10911699', 'SRR10911749', 'SRR10911791', 'SRR10911934', 'SRR10911362', 'SRR10911329', 'SRR10911376', 'SRR10911804', 'SRR10911318', 'SRR10911178', 'SRR10911702', 'SRR10911868', 'SRR10911370', 'SRR10911691', 'SRR10911466', 'SRR10911729', 'SRR10911621', 'SRR10911940', 'SRR10911822', 'SRR10911859', 'SRR10911174', 'SRR10911827', 'SRR10911305', 'SRR10911794', 'SRR10911761', 'SRR10911955', 'SRR10911876', 'SRR10911941', 'SRR10911361', 'SRR10911157', 'SRR10911771', 'SRR10911849', 'SRR10911679', 'SRR10911366', 'SRR10911949', 'SRR10911341', 'SRR10911818', 'SRR10911306', 'SRR10911796', 'SRR10911759', 'SRR10911768', 'SRR10911834', 'SRR10911684', 'SRR10911452', 'SRR10911350', 'SRR10911927', 'SRR10911355', 'SRR10911690', 'SRR10911846', 'SRR10911190', 'SRR10911516', 'SRR10911234', 'SRR10911334', 'SRR10911170', 'SRR10911766', 'SRR10911429', 'SRR10911966', 'SRR10911920', 'SRR10911363', 'SRR10911841', 'SRR10911228', 'SRR10911189', 'SRR10911815', 'SRR10911337', 'SRR10911167', 'SRR10911313', 'SRR10911865', 'SRR10911443', 'SRR10911133', 'SRR10911166', 'SRR10911858', 'SRR10911845', 'SRR10911910', 'SRR10911936', 'SRR10911237', 'SRR10911728', 'SRR10911155', 'SRR10911169', 'SRR10911540', 'SRR10911706', 'SRR10911758', 'SRR10911828', 'SRR10911959', 'SRR10911321', 'SRR10911833', 'SRR10911903', 'SRR10911871', 'SRR10911890', 'SRR10911471', 'SRR10911785', 'SRR10911886', 'SRR10911345', 'SRR10911317', 'SRR10911747', 'SRR10911839', 'SRR10911225', 'SRR10911307', 'SRR10911743', 'SRR10911923', 'SRR10911931', 'SRR10911374', 'SRR10911380', 'SRR10911781', 'SRR10911179', 'SRR10911235', 'SRR10911811', 'SRR10911180', 'SRR10911411', 'SRR10911639', 'SRR10911383', 'SRR10911444', 'SRR10911221', 'SRR10911938', 'SRR10911723', 'SRR10911819', 'SRR10911837', 'SRR10911893', 'SRR10911930', 'SRR10911799', 'SRR10911323', 'SRR10911695', 'SRR10911803', 'SRR10911882', 'SRR10911259', 'SRR10911784', 'SRR10911807', 'SRR10911774', 'SRR10911482', 'SRR10911830', 'SRR10911877', 'SRR10911778'}\n",
      "2021-04-05 20:26:54 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:26:54 INFO After filtering, 333 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 333 samples, 4664 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 333 samples, 4664 features\n",
      "**********\n",
      "processing ../studies/14 (58)\n",
      "['../studies/14/all.biom']\n",
      "2021-04-05 20:28:22 INFO loaded 123 samples, 1271 features\n",
      "2021-04-05 20:28:22 WARNING These have metadata but do not have data - dropped (57): {'SRR6804656', 'SRR6307733', 'SRR6804663', 'SRR6804657', 'SRR6307712', 'SRR6307779', 'SRR6307775', 'SRR6307789', 'SRR6307795', 'SRR6307780', 'SRR6804660', 'SRR6307756', 'SRR6307788', 'SRR6307794', 'SRR6307777', 'SRR6307759', 'SRR6307781', 'SRR6307790', 'SRR6804661', 'SRR6307774', 'SRR6804665', 'SRR6307796', 'SRR6307792', 'SRR6307734', 'SRR6307714', 'SRR6804653', 'SRR6804654', 'SRR6804655', 'SRR6307716', 'SRR6307797', 'SRR6307721', 'SRR6307778', 'SRR6307791', 'SRR6804666', 'SRR6307755', 'SRR6307776', 'SRR6804652', 'SRR6307750', 'SRR6307757', 'SRR6307717', 'SRR6307718', 'SRR6804664', 'SRR6804658', 'SRR6307713', 'SRR6307758', 'SRR6307719', 'SRR6307773', 'SRR6804659', 'SRR6307715', 'SRR6307760', 'SRR6307793', 'SRR6307753', 'SRR6804662', 'SRR6307720', 'SRR6307782', 'SRR6307754', 'SRR6307751'}\n",
      "2021-04-05 20:28:22 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:28:22 INFO After filtering, 123 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 123 samples, 1271 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 123 samples, 1271 features\n",
      "**********\n",
      "processing ../studies/22 (59)\n",
      "['../studies/22/all.biom']\n",
      "2021-04-05 20:29:54 INFO loaded 144 samples, 2687 features\n",
      "2021-04-05 20:29:54 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:29:54 INFO After filtering, 144 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 144 samples, 2687 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 144 samples, 2687 features\n",
      "**********\n",
      "processing ../studies/25 (61)\n",
      "['../studies/25/all.biom']\n",
      "2021-04-05 20:31:29 INFO loaded 92 samples, 1781 features\n",
      "2021-04-05 20:31:29 WARNING These have metadata but do not have data - dropped (1): {'ERR1776941'}\n",
      "2021-04-05 20:31:29 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 20:31:29 INFO After filtering, 89 remain.\n",
      "-------------\n",
      "AmpliconExperiment with 89 samples, 1781 features\n",
      "filtering for HC and disease\n",
      "AmpliconExperiment with 89 samples, 1781 features\n",
      "finished. Total processed 62. Total ok 59. Not enough samples 0\n"
     ]
    }
   ],
   "source": [
    "expall=join_all_exps(subset_size=None, random_seed=2020)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AmpliconExperiment (\"merged no subsampling\") with 11459 samples, 44537 features"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expall.description='merged no subsampling'\n",
    "expall"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2021-04-05 20:33:17 INFO Metadata field taxonomy not found. Saving biom table without metadata\n"
     ]
    }
   ],
   "source": [
    "expall.save('../allsamples/allsamples-no-subsampling')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "../studies/45    1043\n",
       "../studies/44     835\n",
       "../studies/37     727\n",
       "../studies/6      612\n",
       "../studies/29     594\n",
       "../studies/39     587\n",
       "../studies/46     554\n",
       "../studies/9      451\n",
       "../studies/7      441\n",
       "../studies/3      334\n",
       "../studies/5      333\n",
       "../studies/48     280\n",
       "../studies/49     263\n",
       "../studies/47     247\n",
       "../studies/27     233\n",
       "../studies/12     224\n",
       "../studies/62     196\n",
       "../studies/2      179\n",
       "../studies/21     178\n",
       "../studies/10     174\n",
       "../studies/51     164\n",
       "../studies/15     162\n",
       "../studies/20     151\n",
       "../studies/22     144\n",
       "../studies/28     135\n",
       "../studies/14     123\n",
       "../studies/16     119\n",
       "../studies/53     115\n",
       "../studies/23     114\n",
       "../studies/11     109\n",
       "../studies/42      96\n",
       "../studies/25      89\n",
       "../studies/43      89\n",
       "../studies/57      85\n",
       "../studies/40      85\n",
       "../studies/18      84\n",
       "../studies/4       82\n",
       "../studies/41      80\n",
       "../studies/55      74\n",
       "../studies/60      73\n",
       "../studies/31      70\n",
       "../studies/19      68\n",
       "../studies/54      63\n",
       "../studies/33      58\n",
       "../studies/50      58\n",
       "../studies/17      50\n",
       "../studies/58      45\n",
       "../studies/56      43\n",
       "../studies/32      43\n",
       "../studies/61      41\n",
       "../studies/1       38\n",
       "../studies/8       33\n",
       "../studies/59      33\n",
       "../studies/34      32\n",
       "../studies/13      31\n",
       "../studies/52      30\n",
       "../studies/24      25\n",
       "../studies/26      25\n",
       "../studies/36      15\n",
       "Name: exp, dtype: int64"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expall.sample_metadata.exp.value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Also save after filtering for blooiming bacteria"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "expall=expall.filter_by_fasta('../allsamples/blooming-bacteria.fa',negate=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2021-04-05 20:34:01 INFO Metadata field taxonomy not found. Saving biom table without metadata\n"
     ]
    }
   ],
   "source": [
    "expall.save('../allsamples/allsamples-no-subsampling-no-bloom')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load the data\n",
    "(up to 23 control and 23 cases in each study)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2021-04-05 22:16:40 INFO loaded 2533 samples, 44526 features\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/amnon/miniconda3/envs/calour/lib/python3.7/site-packages/ipykernel_launcher.py:3: DtypeWarning: Columns (7,8,9,11,12,13,14,21,23,28,36,40,43,48,50,52,54,55,56,57,58,59,61,63,64,65,66,69,70,71,72,73,75,77,79,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,103,106,107,108,110,113,114,115,118,120,121,122,123,124,125,127,129,131,132,133,134,135,137,138,140,145,146,147,148,152,153,154,155,156,157,158,159,160,162,163,164,165,166,167,168,169,170,171,185,186,189,190,191,192,193,194,195,196,197,199,200,201,202,203,205,206,207,208,209,211,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228,229,230,231,232,233,234,235,236,237,239,240,242,243,260,261,263,264,265,266,269,270,272,332,333,336,337,341,342,343) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  This is separate from the ipykernel package so we can avoid doing imports until\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2021-04-05 22:16:45 WARNING Do you forget to normalize your data? It is required before running this function\n",
      "2021-04-05 22:16:45 INFO After filtering, 2514 remain.\n"
     ]
    }
   ],
   "source": [
    "exp=ca.read_amplicon('../allsamples/allsamples-no-bloom.biom',\n",
    "                     '../allsamples/allsamples-no-bloom_sample.txt',\n",
    "                     min_reads=1000, normalize=10000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AmpliconExperiment with 2514 samples, 44526 features"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exp"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Keep only features present in at least 10 samples\n",
    "(so we filter min_prevalence 9.5/num_of_samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2021-04-05 22:16:50 INFO After filtering, 2692 remain.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "AmpliconExperiment with 2514 samples, 2692 features"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tt=exp.filter_prevalence(9.5/len(exp.sample_metadata))\n",
    "tt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### merge by sample type for taxonomy plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "tt=exp.join_metadata_fields('exp','type','exptype',axis=0,inplace=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "tt=tt.aggregate_by_metadata('exptype', agg='mean')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "tt=tt.sort_samples('exp')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "creating logger\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<calour.heatmap.plotgui_qt5.PlotGUI_QT5 at 0x7f911ba8fe10>"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cu.splot(tt,'type')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2021-04-05 22:17:05 INFO Metadata field taxonomy not found. Saving biom table without metadata\n"
     ]
    }
   ],
   "source": [
    "tt.save('../allsamples/cohort-disease-mean')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Save separately the healthy and sick groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AmpliconExperiment with 59 samples, 44526 features"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ttt=tt.filter_samples('type','HC')\n",
    "ttt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2021-04-05 22:17:08 INFO Metadata field taxonomy not found. Saving biom table without metadata\n"
     ]
    }
   ],
   "source": [
    "ttt.save('../allsamples/cohort-disease-mean-HC')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AmpliconExperiment with 59 samples, 44526 features"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ttt=tt.filter_samples('type','disease')\n",
    "ttt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2021-04-05 22:17:10 INFO Metadata field taxonomy not found. Saving biom table without metadata\n"
     ]
    }
   ],
   "source": [
    "ttt.save('../allsamples/cohort-disease-mean-disease')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Save single combined healthy/sick (for all cohorts together)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AmpliconExperiment with 2 samples, 44526 features"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ttt=tt.aggregate_by_metadata('type', agg='mean')\n",
    "ttt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2021-04-05 22:17:12 INFO Metadata field taxonomy not found. Saving biom table without metadata\n"
     ]
    }
   ],
   "source": [
    "ttt.save('../allsamples/mean-type-combined')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
